rm(list=ls())

path="D:/Document/Serieux/Travail/Data_analysis_and_papers/GTEX_analysis_Fsusie"
source(paste0(path,"/code/plot_all_effect_log.R"))
source(paste0(path,"/code/plot_log.R"))
library(fsusieR)
library(ggplot2)

load(paste0(path,"/output/local/largest_effect/ENSG00000168710.csv.gz.RData"))
plot_susiF_pip(out$res, pos_SNP = as.numeric(out$info_SNP$POS))+
  
  geom_rect(aes(xmin = out$locus[1], 
                xmax =out$locus[2],
                ymin = -0.01,
                ymax = 1.02), 
            alpha = 0.0, color = "red")
 
#lead SNP for the 3 CS
#CS 1 rs376197939
#CS 2 rs1400511103
#CS 3 rs1557742436


obj= out$res
Y=log1p(as.matrix(out$Y/out$size_factor_local) )
X=as.matrix(out$X) 

  
 out1=out
 obj1=out$res
 obj=out$res
 chr= paste0("chr",out1$chr)
 pos0 =out1$locus[1]
 pos1=out1$locus[2]
 
 snp_info=out1$info_SNP
 cs = 1
 log1p_count=TRUE
 data_splice=NULL
 plot_cred_band=TRUE
 type_data="p"
  
  # Extract the relevant genes and exons in the specified region
  region_genes <- genes(txdb,columns = c("tx_id","gene_id"))
  
  # Subset the genes and exons to the region of interest.
  region_genes <- subsetByOverlaps(region_genes,
                                   GRanges(seqnames = chr,
                                           ranges = IRanges(pos0,pos1)))
  
  # Generate a sequence of positions with a length of 1,024.
  positions <- seq(pos0,pos1,length.out = 1024)
  
  markers <- obj$cs[[cs]]
  j       <- which.max(obj$pip[markers])
  marker  <- markers[j]
  x       <- X[,marker]
  
  
  
  
   
      read_counts <- rbind(colMeans(log1p(Y[x == 0,])),
                           colMeans(log1p(Y[x == 1,])) )
    
  ### CS1 -----
  
  uni_res= univariate_functional_regression(Y= Y  ,
                                            X=as.matrix(X[,obj$cs[[cs]][1]],
                                                        ncol=1),
                                            method="TI"
  )
  
  
  if (plot_cred_band){
    
    effect=rbind (uni_res$effect_estimate ,
                  uni_res$cred_band ,
                  rep(0, length(obj$fitted_func[[cs]])))
    
    group_cred= c(1:3,0)
    group_colors <- c("black" ,"royalblue","royalblue","royalblue" )
  }else{
    effect=obj$fitted_func[[cs]]
    
    
  }

  
  
  # Create a "data track" to show the CS effect.
  cex <- 1
  
  group_lwd= c(1,2,1,1)
  group_lty= c(1,1,2,2)
  effect_track <-
    DataTrack(range = GRanges(seqnames = chr,
                              ranges = IRanges(start = positions,
                                               end = positions + 1)),
              data = effect, genome = "hg38",
              groups= group_cred,
              lwd=group_lwd,
              lty=group_lty,
              name = paste("Effect CS",cs),type = "l",col = group_colors,
              track.margin = 0.05,cex.title = cex,cex.axis = cex,
              col.axis = "black",col.title = "black",
              fontface = "plain",background.title = "white",
              fontface.title = 1,,
              legend = FALSE )
  
  
  plotTracks(effect_track)
  # Create another "data track" to show the read counts.
  
  
  
  n0  <- sum(x == 0)
  n1  <- sum(x == 1)
  n2  <- sum(x == 2)
  id  <- "rs376197939" #snp_info[marker,"ID"]
  ref <- "AA"#snp_info[marker,"REF"]
  alt <- "AT" #snp_info[marker,"ALT"]
  
  if (! length(which(x==2))>1){
    groups <- c(sprintf("\t    %s %s (n = %d)",id, ref,n0),
                sprintf("\t    %s %s (n = %d)",id, alt,n1) )
    geno_colors <- c("navyblue","turquoise" )
  }else{
    groups <- c(sprintf("%s %s%s (n = %d)",id,ref,ref,n0),
                sprintf("%s %s%s (n = %d)",id,ref,alt,n1),
                sprintf("%s %s%s (n = %d)",id,alt,alt,n2))
    geno_colors <- c("navyblue","turquoise","darkorange")
  }
  
  
  if (mean(effect) > 0) {
    groups <- factor(groups,rev(groups))
    geno_colors <- rev(geno_colors)
  } else {
    groups <- factor(groups,groups)
  }
  
 
  lab_y =ifelse(log1p_count, "Avg. log1p count","Avg. count")
  data_track <- DataTrack(range = GRanges(seqnames = chr,
                                          ranges = IRanges(start = positions,
                                                           end = positions + 1)),
                          data = read_counts,genome = "hg38",
                          groups = groups,
                          name = lab_y  , type = type_data, #"p",#type = "l",
                          col = geno_colors  ,
                          track.margin = 0.05,cex.title = cex,cex.axis = cex,
                         
                          col.axis = "black",col.title = "black",
                          fontface = "plain",background.title = "white",
                          fontface.title = 1,cex.legend = cex, cex=2)
  
  data_track <-DataTrack(range = GRanges(seqnames = chr,
                                       ranges = IRanges(start = positions,
                                                        end = positions + 1)),
                       data = read_counts,genome = "hg38",
                       groups = groups,
                       name = lab_y , type = type_data,  col = geno_colors ,track.margin = 0.05,cex.title = cex,cex.axis = cex,
                       
                       col.axis = "black",col.title = "black",
                       fontface = "plain",background.title = "white", 
                       fontface.title = 1, cex= .6,cex.legend = 1.1)
             
  
  plotTracks(data_track)
  
  # Create an "ideogram" track.
  ideo_track <- IdeogramTrack(genome = "hg38",chromosome = chr)
  
  # Create a "genome axis" track.
  genome_track <- GenomeAxisTrack(col.axis = "black",col.title = "black")
  
  # Create a "gene region" track.
  gene_track <- GeneRegionTrack(txdb,genome = "hg38",chromosome = chr,
                                pos0 = pos0,pos1 = pos1,name = "",
                                showId = TRUE,geneSymbol = TRUE,
                                col.axis = "black",col.title = "black",
                                transcriptAnnotation = "symbol",
                                rotation.title = 0,cex.title = 2,
                                col = "salmon",fill = "salmon",
                                background.title = "white")
  
  # Map gene IDs to gene symbols.
  gene_ids <- unique(unlist(region_genes$gene_id))
  
  # Map to gene symbols using org.Hs.eg.db
  gene_symbols <- AnnotationDbi::select(org.Hs.eg.db,keys = gene_ids,
                                        columns = "SYMBOL",
                                        keytype = "ENTREZID")
  n <- nrow(gene_symbols)
  if (n > 0) {
    for (i in 1:n) {
      j <- which(gene_track@range@elementMetadata@listData$gene ==
                   gene_symbols$ENTREZID[i])
      gene_track@range@elementMetadata@listData$id[j]     <- gene_symbols$SYMBOL[i]
      gene_track@range@elementMetadata@listData$symbol[j] <- gene_symbols$SYMBOL[i]
    }
  }
  
   
    # Combine all tracks into a single plot.
    tracks <- c(ideo_track,
                genome_track,
                effect_track,
                data_track,
                gene_track)
  
  
  plotTracks(tracks,from = pos0,to = pos1,sizes = c(1,1.75,2,4,2)) 
  
  folder_path=  paste0(getwd(),
                       "/plot/"
  )
  file_path <- file.path(folder_path, "AHCYL1_cs1.pdf")
  pdf(file_path, width =16.69, height = 8.27 )  # A4 in inches
  
  
  
  plotTracks(tracks,from = pos0-200,
             to = pos1+200,
             sizes = c(1,1 ,4,4,2),
             cex.main=1.2, cex.title = 1.
             ) 
 dev.off()
 
 
 
 
 ## CS 2 -----
 cs=2
 Y=log1p(as.matrix(out$Y/out$size_factor_local) )
 X=as.matrix(out$X)
 
 markers <- obj$cs[[cs]]
 j       <- which.max(obj$pip[markers])
 marker  <- markers[j]
 x       <- X[,marker]
 n0  <- sum(x == 0)
 n1  <- sum(x == 1)
 n2  <- sum(x == 2)
 id  <- "rs1400511103" #snp_info[marker,"ID"]
 ref <- "CC"#snp_info[marker,"REF"]
 alt <- "CT" #snp_info[marker,"ALT"]
 
 
 groups <- c(sprintf("    %s %s (n = %d)",id, ref,n0),
             sprintf("    %s %s (n = %d)",id, alt,n1) )
 geno_colors <- c("turquoise","navyblue" )
 
 
 read_counts <- rbind(colMeans(log1p(Y[x == 0,])),
                      colMeans(log1p(Y[x == 1,])) )
 
 which(x==2
       )
 uni_res= univariate_functional_regression(Y=log1p(Y[-65,]/out$size_factor_local[-65] ),
                                           X=as.matrix(X[-65,obj$cs[[cs]][1]],
                                                       ncol=1),
                                           method="TI"
 )
 uni_res= smash_regression(obj, Y= log1p(Y[-65,]/out$size_factor_local[-65])  ,
                                           X=  X  [-65,]
 ) 
 plot( uni_res$fitted_func[[2]], type="l")
 plot( uni_res$effect_estimate, type="l")
 
 lines( uni_res$cred_band[1,], lty=2)
 lines( uni_res$cred_band[2,], lty=2)
 
 
 
 if (plot_cred_band){
   
   effect=rbind (uni_res$fitted_func[[2]] ,
                 uni_res$cred_band[[2]] ,
                 rep(0, length(obj$fitted_func[[cs]])))
   
   group_cred= c(1:3,0)
   group_colors <- c("black" ,"royalblue","royalblue","royalblue" )
 }else{
   effect=obj$fitted_func[[cs]]
   
   
 }
 
 
 
 # Create a "data track" to show the CS effect.
 cex <- 1
 
 group_lwd= c(1,2,1,1)
 group_lty= c(1,1,2,2)
 effect_track <-
   DataTrack(range = GRanges(seqnames = chr,
                             ranges = IRanges(start = positions,
                                              end = positions + 1)),
             data = effect, genome = "hg38",
             groups= group_cred,
             lwd=group_lwd,
             lty=group_lty,
             name = paste("Effect CS",cs),type = "l",col = group_colors,
             track.margin = 0.05,cex.title = cex,cex.axis = cex,
             col.axis = "black",col.title = "black",
             fontface = "plain",background.title = "white",
             fontface.title = 1,,
             legend = FALSE )
 
 
 plotTracks(effect_track)
 # Create another "data track" to show the read counts.
 
 
   groups <- factor(groups,rev(groups))
  # geno_colors <- rev(geno_colors)
 
 
 
 lab_y =ifelse(log1p_count, "Avg. log1p count","Avg. count")
 
 data_track <-DataTrack(range = GRanges(seqnames = chr,
                                        ranges = IRanges(start = positions,
                                                         end = positions + 1)),
                        data = read_counts,genome = "hg38",
                        groups = groups,
                        name = lab_y , type = type_data,  col = geno_colors ,
                        track.margin = 0.05,cex.title = cex,cex.axis = cex,
                        
                        col.axis = "black",col.title = "black",
                        fontface = "plain",background.title = "white", 
                        fontface.title = 1, cex= .6,cex.legend = 1.1)
 
 
 plotTracks(data_track)
  
 # Combine all tracks into a single plot.
 tracks <- c(ideo_track,
             genome_track,
             effect_track,
             data_track,
             gene_track)
 
 
 plotTracks(tracks,from = pos0,to = pos1,sizes = c(1,1.75,2,4,2)) 
 
 folder_path=  paste0(getwd(),
                      "/plot/"
 )
 file_path <- file.path(folder_path, "AHCYL1_cs2.pdf")
 pdf(file_path, width =16.69, height = 8.27 )  # A4 in inches
 
 
 
 plotTracks(tracks,from = pos0-200,
            to = pos1+200,
            sizes = c(1,1 ,4,4,2),
            cex.main=1.2, cex.title = 1.
 ) 
 dev.off()
 
 
 
 ## CS 3 -----
 cs=3
 
 
 uni_res= univariate_functional_regression(Y=log1p(Y ),
                                           X=as.matrix(X[,obj$cs[[cs]][1]],
                                                       ncol=1),
                                           method="TI"
 )
 if (plot_cred_band){
   
   effect=rbind (uni_res$effect_estimate ,
                 uni_res$cred_band ,
                 rep(0, length(obj$fitted_func[[cs]])))
   
   group_cred= c(1:3,0)
   group_colors <- c("black" ,"royalblue","royalblue","royalblue" )
 }else{
   effect=obj$fitted_func[[cs]]
   
   
 }
 
 
 
 # Create a "data track" to show the CS effect.
 cex <- 1
 
 group_lwd= c(1,2,1,1)
 group_lty= c(1,1,2,2)
 effect_track <-
   DataTrack(range = GRanges(seqnames = chr,
                             ranges = IRanges(start = positions,
                                              end = positions + 1)),
             data = effect, genome = "hg38",
             groups= group_cred,
             lwd=group_lwd,
             lty=group_lty,
             name = paste("Effect CS",cs),type = "l",col = group_colors,
             track.margin = 0.05,cex.title = cex,cex.axis = cex,
             col.axis = "black",col.title = "black",
             fontface = "plain",background.title = "white",
             fontface.title = 1,,
             legend = FALSE )
 
 
 plotTracks(effect_track)
 # Create another "data track" to show the read counts.
 
 n0  <- sum(x == 0)
 n1  <- sum(x == 1)
 n2  <- sum(x == 2)
 id  <- "rs376197939" #snp_info[marker,"ID"]
 ref <- "AA"#snp_info[marker,"REF"]
 alt <- "AT" #snp_info[marker,"ALT"]
 
 if (! length(which(x==2))>1){
   groups <- c(sprintf("\t    %s %s (n = %d)",id, ref,n0),
               sprintf("\t    %s %s (n = %d)",id, alt,n1) )
   geno_colors <- c("navyblue","turquoise" )
 }else{
   groups <- c(sprintf("%s %s%s (n = %d)",id,ref,ref,n0),
               sprintf("%s %s%s (n = %d)",id,ref,alt,n1),
               sprintf("%s %s%s (n = %d)",id,alt,alt,n2))
   geno_colors <- c("navyblue","turquoise","darkorange")
 }
 
 
 if (mean(effect) > 0) {
   groups <- factor(groups,rev(groups))
   geno_colors <- rev(geno_colors)
 } else {
   groups <- factor(groups,groups)
 }
 
 
 lab_y =ifelse(log1p_count, "Avg. log1p count","Avg. count")
 data_track <- DataTrack(range = GRanges(seqnames = chr,
                                         ranges = IRanges(start = positions,
                                                          end = positions + 1)),
                         data = read_counts,genome = "hg38",
                         groups = groups,
                         name = lab_y  , type = type_data, #"p",#type = "l",
                         col = geno_colors  ,
                         track.margin = 0.05,cex.title = cex,cex.axis = cex,
                         
                         col.axis = "black",col.title = "black",
                         fontface = "plain",background.title = "white",
                         fontface.title = 1,cex.legend = cex, cex=2)
 
 data_track <-DataTrack(range = GRanges(seqnames = chr,
                                        ranges = IRanges(start = positions,
                                                         end = positions + 1)),
                        data = read_counts,genome = "hg38",
                        groups = groups,
                        name = lab_y , type = type_data,  col = geno_colors ,track.margin = 0.05,cex.title = cex,cex.axis = cex,
                        
                        col.axis = "black",col.title = "black",
                        fontface = "plain",background.title = "white", 
                        fontface.title = 1, cex= .6,cex.legend = 1.1)
 
 
 plotTracks(data_track)
 
 # Create an "ideogram" track.
 ideo_track <- IdeogramTrack(genome = "hg38",chromosome = chr)
 
 # Create a "genome axis" track.
 genome_track <- GenomeAxisTrack(col.axis = "black",col.title = "black")
 
 # Create a "gene region" track.
 gene_track <- GeneRegionTrack(txdb,genome = "hg38",chromosome = chr,
                               pos0 = pos0,pos1 = pos1,name = "",
                               showId = TRUE,geneSymbol = TRUE,
                               col.axis = "black",col.title = "black",
                               transcriptAnnotation = "symbol",
                               rotation.title = 0,cex.title = 2,
                               col = "salmon",fill = "salmon",
                               background.title = "white")
 
 # Map gene IDs to gene symbols.
 gene_ids <- unique(unlist(region_genes$gene_id))
 
 # Map to gene symbols using org.Hs.eg.db
 gene_symbols <- AnnotationDbi::select(org.Hs.eg.db,keys = gene_ids,
                                       columns = "SYMBOL",
                                       keytype = "ENTREZID")
 n <- nrow(gene_symbols)
 if (n > 0) {
   for (i in 1:n) {
     j <- which(gene_track@range@elementMetadata@listData$gene ==
                  gene_symbols$ENTREZID[i])
     gene_track@range@elementMetadata@listData$id[j]     <- gene_symbols$SYMBOL[i]
     gene_track@range@elementMetadata@listData$symbol[j] <- gene_symbols$SYMBOL[i]
   }
 }
 
 
 # Combine all tracks into a single plot.
 tracks <- c(ideo_track,
             genome_track,
             effect_track,
             data_track,
             gene_track)
 
 
 plotTracks(tracks,from = pos0,to = pos1,sizes = c(1,1.75,2,4,2)) 
 
 folder_path=  paste0(getwd(),
                      "/plot/"
 )
 file_path <- file.path(folder_path, "AHCYL1_cs3.pdf")
 pdf(file_path, width =16.69, height = 8.27 )  # A4 in inches
 
 
 
 plotTracks(tracks,from = pos0-200,
            to = pos1+200,
            sizes = c(1,1 ,4,4,2),
            cex.main=1.2, cex.title = 1.
 ) 
 dev.off()
 
 
 