<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="William Denault, Hao Sun, Angjing Liu, Peter Carbonetto, Gao Wang" />


<title>Examine ROSMAP methylation fine-mapping results</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fsusie experiments</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/fsusie-experiments">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Examine ROSMAP methylation fine-mapping
results</h1>
<h4 class="author">William Denault, Hao Sun, Angjing Liu, Peter
Carbonetto, Gao Wang</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>fsusie-experiments/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
flag-duplicated-cs
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>rosmap_overview_cache</code> and re-run the analysis. To have
workflowr automatically delete the cache directory prior to building the
file, set <code>delete_cache = TRUE</code> when running
<code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfsusieexperimentstree14bc90169522c19e724f0ef84008a6b9b8475168targetblank14bc901a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/stephenslab/fsusie-experiments/tree/14bc90169522c19e724f0ef84008a6b9b8475168" target="_blank">14bc901</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfsusieexperimentstree14bc90169522c19e724f0ef84008a6b9b8475168targetblank14bc901a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/stephenslab/fsusie-experiments/tree/14bc90169522c19e724f0ef84008a6b9b8475168" target="_blank">14bc901</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/rosmap_h3k27ac_cache/
    Untracked:  analysis/rosmap_overview_cache/
    Untracked:  data/afreq.RData
    Untracked:  data/analysis_result/Fungen_xQTL.ENSG00000163808.cis_results_db.export.rds
    Untracked:  data/analysis_result/ROSMAP_haQTL.chr3_43915257_48413435.fsusie_mixture_normal_top_pc_weights.rds
    Untracked:  data/analysis_result/ROSMAP_mQTL.chr3_43915257_48413435.fsusie_mixture_normal_top_pc_weights.rds
    Untracked:  outputs/CASS4_all_effects.RData
    Untracked:  outputs/CASS4_obj.RData
    Untracked:  outputs/CD2AP_all_effects.RData
    Untracked:  outputs/CD2AP_obj.RData
    Untracked:  outputs/CR1_CR2_all_effects.RData
    Untracked:  outputs/CR1_CR2_obj.RData
    Untracked:  outputs/ROSMAP_DLPFC_mega_eQTL.cs_only.tsv.gz
    Untracked:  outputs/ROSMAP_DLPFC_pQTL.cs_only.tsv.gz
    Untracked:  outputs/ROSMAP_haQTL_cs_effect_ha_peak_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_haQTL_cs_snp_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_haQTL_cs_snp_toppc1_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_haQTL_qtl_snp_qval0.05.tsv.gz
    Untracked:  outputs/ROSMAP_haQTL_qtl_snp_qval0.05_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_cs_effect_cpg_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_cs_snp_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_cs_snp_toppc1_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_qtl_snp_qval0.05.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_qtl_snp_qval0.05_annotation.tsv.gz

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/rosmap_overview.Rmd</code>) and
HTML (<code>docs/rosmap_overview.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/14bc90169522c19e724f0ef84008a6b9b8475168/analysis/rosmap_overview.Rmd" target="_blank">14bc901</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-14
</td>
<td>
wflow_publish("rosmap_overview.Rmd", verbose = TRUE, view = FALSE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/387295e14f2e98bd664a982e2862fc9c0abc924d/analysis/rosmap_overview.Rmd" target="_blank">387295e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-14
</td>
<td>
A few improvements to the CpG-mSNP distance plot in the rosmap_overview
analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/fe030b047e0c8528832d831d0124dce6802d9b1d/analysis/rosmap_overview.Rmd" target="_blank">fe030b0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-14
</td>
<td>
A bunch of revisions to the rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/3e1d8ab629227c8a814f32dda9a117033ebbdd81/analysis/rosmap_overview.Rmd" target="_blank">3e1d8ab</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-14
</td>
<td>
Updated distance-to-TSS plot in rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/31d4931a5960c4618555cc5bed6998a7f2f833d3/analysis/rosmap_overview.Rmd" target="_blank">31d4931</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-14
</td>
<td>
Revised blacklisting steps in the rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/0ed2a67ceca6ff7d156c324f8998c65fc2b2ff07/analysis/rosmap_overview.Rmd" target="_blank">0ed2a67</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-14
</td>
<td>
Revised blacklisting step in rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/4a1285d70efd37eeba7440616787c42a2089c53b/analysis/rosmap_overview.Rmd" target="_blank">4a1285d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-14
</td>
<td>
Added step to rosmap_overview to filter the results by MAF.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/215a994e7c4d94ea1cf52fdb1f932cfbad07159a/analysis/rosmap_overview.Rmd" target="_blank">215a994</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Overhauling rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/392f166ff5df03c802890f110676a5dbe0c19f01/analysis/rosmap_overview.Rmd" target="_blank">392f166</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Added haSNP-peak histogram to the rosmap_h3k27ac analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/09385eecf236d16b2102d16a680c8a08fc6369c8/analysis/rosmap_overview.Rmd" target="_blank">09385ee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Working on adding affected peak results to the rosmap_h3k27ac analysis;
also added a step to remove duplicated CSs from peak-level results for
fSuSiE as well as a MAF filtering step for the SNP-peak association
testing results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/69dcefaff07902e13314d5c5d4bc8b3a9f73b9f1/analysis/rosmap_overview.Rmd" target="_blank">69dcefa</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Added steps to the rosmap_h3k27ac analysis to load the peak-level
results and apply the MAF filter to them.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/890a515df051d86fe1a21c63f7949b54cc387176/analysis/rosmap_overview.Rmd" target="_blank">890a515</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Added tables to the rosmap_h3k27ac analysis showing the distribution of
PIPs in the 1-SNP CSs.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/acd259e1fc3a4fad64ff746783d2524ccbd1573e/analysis/rosmap_overview.Rmd" target="_blank">acd259e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Added TSS plot to rosmap_h3k27ac analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/1c5ff9e2e530a281518a3a00ca0a4926ec67e145/analysis/rosmap_overview.Rmd" target="_blank">1c5ff9e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Added step to remove duplicate CSs and added CS size histograms to
rosmap_h3k27ac analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/9a0d3ab0e25a21fa35d4336073c7775e006a449d/analysis/rosmap_overview.Rmd" target="_blank">9a0d3ab</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Added plots to rosmap_h3k27ac analysis comparing discovery of CSs in
TADs.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/e6480b7147272a287af486be27f3bf95102b333a/analysis/rosmap_overview.Rmd" target="_blank">e6480b7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Added steps to rosmap_h3k27ac analysis to load gene data, allele
frequency data, and fine-mapping results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/ed237c677d5921080c66b2599e3d2855ad75a983/analysis/rosmap_overview.Rmd" target="_blank">ed237c6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-13
</td>
<td>
Started new analysis rosmap_h3k27ac.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/0920fa24e74a91cda14f8a58cd57e263e9ad66d7/analysis/rosmap_overview.Rmd" target="_blank">0920fa2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-02
</td>
<td>
Added the effect plot to pip_zoomout_cass4.pdf.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/ba346e7b4ba97ba5d755cbb29362249e3ec3fd35/analysis/rosmap_overview.Rmd" target="_blank">ba346e7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-01
</td>
<td>
Fixed a typo in rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/4c5e1a0d9d795bb9471c49ff9dc335ceefed70aa/analysis/rosmap_overview.Rmd" target="_blank">4c5e1a0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-30
</td>
<td>
Small edit to rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/4adf2dd19b17c506f500c6a17f2d3f186a3939ea/docs/rosmap_overview.html" target="_blank">4adf2dd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-29
</td>
<td>
Implemented exploratory analysis script fsusie_illumina450k.R.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/524bd60522dd5e79ce0ab40db214c982e5152ef8/analysis/rosmap_overview.Rmd" target="_blank">524bd60</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-26
</td>
<td>
Added short note to rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/d651761c2957578efbbd4af40be85c5a46abaa99/analysis/rosmap_overview.Rmd" target="_blank">d651761</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-25
</td>
<td>
A few small fixes to the rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/3b385be4e7e4d441064d45e73e228f7c7677bdc6/analysis/rosmap_overview.Rmd" target="_blank">3b385be</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-25
</td>
<td>
Fixed a couple of bugs in rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/748b8b0f5ebcc202993cc01ec6bb9ae3b468c122/analysis/rosmap_overview.Rmd" target="_blank">748b8b0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-25
</td>
<td>
Added more detailed analysis of the 1-SNP CSs for methylation in
rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/9dad9a938e354a4dfa012bc086d22e5ba8e77fe8/analysis/rosmap_overview.Rmd" target="_blank">9dad9a9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-24
</td>
<td>
Working on additional steps to identify and remove blacklisted mSNPs.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/dade7d2d7b0b7b39aa36a06a689b4ea4b96a98bf/docs/rosmap_overview.html" target="_blank">dade7d2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Ran workflowr::wflow_publish("rosmap_overview.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/14e44873b58948fbaf51301dc25b5edaee71cc2d/analysis/rosmap_overview.Rmd" target="_blank">14e4487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Updated the plot in rosmap_overview.Rmd showing num. peaks vs. num. CSs
for fSusiE H3K27ac results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/1fd676e4f2f1cf362aaad605b20ee56f1865b6d6/analysis/rosmap_overview.Rmd" target="_blank">1fd676e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Added step to rosmap_overview analysis to remove duplicate CSs for
H3K27ac.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/cdc55a21371d4e9349f9092a7d433f0b5642f20d/analysis/rosmap_overview.Rmd" target="_blank">cdc55a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Fixed histogram showing no. cpgs vs. no. CSs in rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/2c0d57446fe7bcb5a9e69d56498b5b509f173b51/analysis/rosmap_overview.Rmd" target="_blank">2c0d574</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Small fix to the CS size histograms for the methylation results in
rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/e1fe364dc6626e234e44231c8ec657e5e34b7cac/analysis/rosmap_overview.Rmd" target="_blank">e1fe364</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-04
</td>
<td>
I have the CD2AP zoom-in plot mostly done except for the panel showing
the raw data (beta values).
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/46d0b351bb74d6e04428c15090865c1e6003a996/analysis/rosmap_overview.Rmd" target="_blank">46d0b35</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-04
</td>
<td>
Removed some code in the rosmap_overview.Rmd analysis into a separate
code chunk.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/9416011b5977235a8c3c1f0e23bec16807f29e13/analysis/rosmap_overview.Rmd" target="_blank">9416011</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-04
</td>
<td>
Added function create_cs_maps() to help remove ‘duplicated’ CSs.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/618cae4fa9aadcbe31f55487e1b478d485770fba/analysis/rosmap_overview.Rmd" target="_blank">618cae4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-03
</td>
<td>
A few fixes to the CR1 eQTL panel.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/618cae4fa9aadcbe31f55487e1b478d485770fba/docs/rosmap_overview.html" target="_blank">618cae4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-03
</td>
<td>
A few fixes to the CR1 eQTL panel.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/3bb4ba0dd104f7fc4d5447b6c8548273d2e777cd/docs/rosmap_overview.html" target="_blank">3bb4ba0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-28
</td>
<td>
Ran wflow_publish("rosmap_overview.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/dcece5bb310e360ec5007a6e5d0e9d03f672748f/analysis/rosmap_overview.Rmd" target="_blank">dcece5b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-28
</td>
<td>
Added code chunks to rosmap_overview.Rmd to save some more plots to PDF.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/a70588c212c06901e4b8b90a29735c9424ad9fbc/analysis/rosmap_overview.Rmd" target="_blank">a70588c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-28
</td>
<td>
Added code chunks to rosmap_overview.Rmd to save a couple plots to PDFs.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/c07956874db10cbee93e0c44ce20bfc946db2004/docs/rosmap_overview.html" target="_blank">c079568</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
Ran wflow_publish("rosmap_overview.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/2bd0de32c1e91e3107a779a9ec9a564199d940e2/analysis/rosmap_overview.Rmd" target="_blank">2bd0de3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
wflow_publish("rosmap_overview.Rmd", verbose = TRUE, view = FALSE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/53d4a33a43b153d27ddac8618759b18f72d6942e/analysis/rosmap_overview.Rmd" target="_blank">53d4a33</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
Added code chunks to rosmap_overview.Rmd for generating PDFs of some of
the plots.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/d5c9e372a24cb6a5d91124df71f15aab6012aef7/analysis/rosmap_overview.Rmd" target="_blank">d5c9e37</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
A few fixes to rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/544235d894e6bb5e84e4e904b560ac62286fa225/analysis/rosmap_overview.Rmd" target="_blank">544235d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
Added code chunks to save some of the plots in the rosmap_overview
analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/aece910973f3984cf07ffbe6089a898f58910c41/docs/rosmap_overview.html" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Ran workflowr::wflow_publish("rosmap_overview.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/0eb82957ca28e83cf1754315d54529997e31f92b/analysis/rosmap_overview.Rmd" target="_blank">0eb8295</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
wflow_publish("rosmap_overview.Rmd", view = FALSE, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/6de898eb207e5f45002c95f841ba794d45426b4f/analysis/rosmap_overview.Rmd" target="_blank">6de898e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Added plots to the rosmap_overview analysis summarizing the recovery of
affected H3K27ac peaks.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/7c8d51d9607a08e886dd1760c9b4c48cd46caa11/analysis/rosmap_overview.Rmd" target="_blank">7c8d51d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Added TSS plot for H3K27ac in rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/8e174862fa2b5e865601f1bb0fb4cd521c67d67a/analysis/rosmap_overview.Rmd" target="_blank">8e17486</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Added code to create a plot showing the density of causal SNPs near the
closest TSS.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/18c11b56e652a26b056cd61e0eaa25fb62455841/analysis/rosmap_overview.Rmd" target="_blank">18c11b5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Added code to rosmap_overview.Rmd to load gene annotations.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/60bbe231bfe3abf2948ff7be5ab88137ffaf0715/analysis/rosmap_overview.Rmd" target="_blank">60bbe23</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-22
</td>
<td>
Added a couple notes to the rosmap_summary analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/00bb89c30a7c47d5661e91ad6ae3f3c4a116eff0/analysis/rosmap_overview.Rmd" target="_blank">00bb89c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added note to rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/7adfd4d32dae6b06ae384254c0f035ccaa1039d9/analysis/rosmap_overview.Rmd" target="_blank">7adfd4d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
A few fixes to the rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c087a58be2e3a5f3a2165fa24c6d484e85293009/analysis/rosmap_overview.Rmd" target="_blank">c087a58</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added some notes to the rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/5e9432b3f27e5f8ac453173d1b25c702e087e291/analysis/rosmap_overview.Rmd" target="_blank">5e9432b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added plots to the rosmap_overview analysis summarizing the affected
CpGs identified by fSuSiE and the association tests.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/593c7e0f140991467385ef93d92611b8060d9324/analysis/rosmap_overview.Rmd" target="_blank">593c7e0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added some code for analyzing the HA peak results in the rosmap_overview
analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/12ed0b15e3b6a2b036a6a414ce55de63104ac175/analysis/rosmap_overview.Rmd" target="_blank">12ed0b1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Small fix.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/e663ab0144e239fbda7948af02ae536cb3ddb91a/analysis/rosmap_overview.Rmd" target="_blank">e663ab0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added code to read in HA_peak results in rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/71954f15429162c72e3c8013edbb72accecc71a9/docs/rosmap_overview.html" target="_blank">71954f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added plots summarizing H3K27ac results to rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/1ad355ae94d1bb9a770faeb0ac56392549a38b4c/analysis/rosmap_overview.Rmd" target="_blank">1ad355a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
wflow_publish("rosmap_overview.Rmd", view = FALSE, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/c84e5c046a56501c872e946ab6839eac67d0a41a/docs/rosmap_overview.html" target="_blank">c84e5c0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Rebuilt the rosmap_overview analysis with the new results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/1c8eeb7f07827e141d64f671b8a38821e1ac7c86/analysis/rosmap_overview.Rmd" target="_blank">1c8eeb7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
wflow_publish("rosmap_overview.Rmd", view = FALSE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/acfadd18f631b80fc953f7c282ada7aeeba48c68/analysis/rosmap_overview.Rmd" target="_blank">acfadd1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Made a few improvements to the code and text of the rosmap_analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c102af90b142616368e65e5deddab968daea046f/analysis/rosmap_overview.Rmd" target="_blank">c102af9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added a scatterplot comparing number of CSs per TAD (susie vs. fsusie).
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/12f2fd34bb981b86976ea16a9d293b4b6a0c55e7/analysis/rosmap_overview.Rmd" target="_blank">12f2fd3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added some histograms on TAD CS sizes.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c69e187bb6e7fd10716665e4c59ebdf0fb927d84/analysis/rosmap_overview.Rmd" target="_blank">c69e187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Created plot showing TAD sizes from the methylation fine-mapping
results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/2a5c7060bfea715430e3f822a89a1ded9791f816/analysis/rosmap_overview.Rmd" target="_blank">2a5c706</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added code to the rosmap_overview analysis to load the methylation SNP
results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c3a01c7e937d5a86509a699d9e57504d6c583587/analysis/rosmap_overview.Rmd" target="_blank">c3a01c7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added link for downloading data to rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/5c446c08a11735474bd94ded6e641065aff07171/docs/rosmap_overview.html" target="_blank">5c446c0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
First build of the rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/7532908fe0418cc52f231593c2a621e8fc7719eb/analysis/rosmap_overview.Rmd" target="_blank">7532908</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
workflowr::wflow_publish("rosmap_overview.Rmd", verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/bc6d0a16f7dfe114a6a17d5cff05ab2eb457497f/analysis/rosmap_overview.Rmd" target="_blank">bc6d0a1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Started working on rosmap_overview analysis.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><strong>Note:</strong> If you would like to run this analysis on your
computer, you will first need to download the fine-mapping outputs. They
can be downloaded from <a
href="https://uchicago.box.com/s/tt1vgg7vqayfthg0vsbl8gw0sdi0f1uo">here</a>.
Once you have downloaded the files, copy each file to the “data” or
“outputs” subdirectory.</p>
<p>Load some packages and custom functions used in the code below:</p>
<pre class="r"><code>library(data.table)
library(dplyr)
library(ggplot2)
library(cowplot)
source(&quot;../code/rosmap_functions_more.R&quot;)</code></pre>
<p>Load the gene annotations used in some of the analyses below.</p>
<pre class="r"><code>gene_file &lt;-
  file.path(&quot;../data/genome_annotations&quot;,
    &quot;Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz&quot;)
genes &lt;- get_gene_annotations(gene_file)</code></pre>
<p>Load the allele frequencies computed by PLINK:</p>
<pre class="r"><code>load(&quot;../data/afreq.RData&quot;)</code></pre>
<p>Next I load methylation SNP results generated by SuSiE-topPC, fSuSiE
and the SNP-CpG association testing:</p>
<pre class="r"><code>assoc_file       &lt;- &quot;../outputs/ROSMAP_mQTL_qtl_snp_qval0.05.tsv.gz&quot;
snps_susie_file  &lt;- &quot;../outputs/ROSMAP_mQTL_cs_snp_toppc1_annotation.tsv.gz&quot;
snps_fsusie_file &lt;- &quot;../outputs/ROSMAP_mQTL_cs_snp_annotation.tsv.gz&quot;
snps_susie  &lt;- read_enrichment_results(snps_susie_file,n = 6)
snps_fsusie &lt;- read_enrichment_results(snps_fsusie_file,n = 7)
assoc       &lt;- read_enrichment_results(assoc_file,n = 8)
snps_susie  &lt;- snps_susie[1:6]
snps_fsusie &lt;- snps_fsusie[1:7]
snps_susie$region &lt;-
  sapply(strsplit(snps_susie$cs,&quot;:&quot;,fixed = TRUE),&quot;[[&quot;,2)
snps_susie  &lt;- transform(snps_susie,
                         region = factor(region),
                         cs     = factor(cs),
                         pc     = factor(pc))
snps_fsusie &lt;- transform(snps_fsusie,
                         cs     = factor(cs),
                         region = factor(region),
                         study  = factor(study))</code></pre>
<p>Add the allele frequencies to the methylation fine-mapping
results:</p>
<pre class="r"><code>ids  &lt;- with(snps_susie,paste(chr,pos,sep = &quot;_&quot;))
rows &lt;- match(ids,afreq$id)
snps_susie$maf &lt;- afreq[rows,&quot;maf&quot;]
ids  &lt;- with(snps_fsusie,paste(chr,pos,sep = &quot;_&quot;))
rows &lt;- match(ids,afreq$id)
snps_fsusie$maf &lt;- afreq[rows,&quot;maf&quot;]</code></pre>
<p>Also load the CpG-level results generated by fSuSiE:</p>
<pre class="r"><code>cpg_fsusie_file    &lt;- &quot;../outputs/ROSMAP_mQTL_cs_effect_cpg_annotation.tsv.gz&quot;
cpgs_fsusie        &lt;- read_enrichment_results(cpg_fsusie_file,n = 9)
cpgs_fsusie        &lt;- cpgs_fsusie[1:9]
cpgs_fsusie$region &lt;- sapply(strsplit(cpgs_fsusie$cs,&quot;:&quot;,fixed = TRUE),&quot;[[&quot;,2)
cpgs_fsusie &lt;- transform(cpgs_fsusie,
                         cs      = factor(cs), 
                         region  = factor(region),
                         context = factor(context))</code></pre>
<p>Keep only CSs if the MAF of the sentinel SNP is &gt;5%:</p>
<pre class="r"><code>keep        &lt;- tapply(snps_susie[c(&quot;pip&quot;,&quot;maf&quot;)],snps_susie$cs,
                      function (x) x[which.max(x$pip),&quot;maf&quot;] &gt;= 0.05)
keep_cs     &lt;- names(which(keep))
snps_susie  &lt;- subset(snps_susie,is.element(cs,keep_cs))
snps_susie  &lt;- transform(snps_susie,
                         region = factor(region),
                         cs     = factor(cs))
keep        &lt;- tapply(snps_fsusie[c(&quot;pip&quot;,&quot;maf&quot;)],snps_fsusie$cs,
                      function (x) x[which.max(x$pip),&quot;maf&quot;] &gt;= 0.05)
keep_cs     &lt;- names(which(keep))
snps_fsusie &lt;- subset(snps_fsusie,is.element(cs,keep_cs))
snps_fsusie &lt;- transform(snps_fsusie,
                         region = factor(region),
                         cs     = factor(cs))</code></pre>
<p>Apply this same MAF filter to the SNP-CpG association tests and the
fSuSiE CpG-level results:</p>
<pre class="r"><code>ids         &lt;- with(assoc,paste(chr,pos,sep = &quot;_&quot;))
rows        &lt;- match(ids,afreq$id)
assoc$maf   &lt;- afreq[rows,&quot;maf&quot;]
assoc       &lt;- subset(assoc,maf &gt;= 0.05)
cpgs_fsusie &lt;- subset(cpgs_fsusie,is.element(cs,keep_cs))
cpgs_fsusie &lt;- transform(cpgs_fsusie,
                         region = factor(region),
                         cs     = factor(cs))</code></pre>
<p>Let’s now get an overview of the TADs. This is the number of
fine-mapping regions (TADs) that contained at least one CS in each of
the analyses:</p>
<pre class="r"><code>nlevels(snps_susie$region)
nlevels(snps_fsusie$region)
# [1] 1145
# [1] 1327</code></pre>
<p>This plot summarizes the sizes of the TADs that were analyzed by
SuSiE-topPC and fSuSiE:</p>
<pre class="r"><code>plot_tad_sizes &lt;- function (tads) {
  tad_size &lt;- get_tad_sizes(tads)
  pdat &lt;- data.frame(tad_size = tad_size)
  return(ggplot(pdat,aes(x = tad_size)) +
         geom_histogram(color = &quot;white&quot;,fill = &quot;black&quot;,bins = 48) +
         labs(x = &quot;size (Mb)&quot;,y = &quot;number of TADs&quot;) +
         theme_cowplot(font_size = 10))
}
tads &lt;- levels(snps_fsusie$region)
p &lt;- plot_tad_sizes(tads) +
  scale_x_continuous(limits = c(2,9),breaks = 1:10) +
  scale_y_continuous(breaks = seq(0,100,10))
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/plot-tad-sizes-1.png" width="330" style="display: block; margin: auto;" /></p>
<p>Some other useful statistics on the TAD sizes:</p>
<pre class="r"><code>tad_size &lt;- get_tad_sizes(tads)
length(tad_size)
range(tad_size)
mean(tad_size)
median(tad_size)
sum(tad_size &gt; 9)
# [1] 1327
# [1]  2.320952 34.727189
# [1] 4.54465
# [1] 4.154266
# [1] 19</code></pre>
<p>There are some technical concerns about fine-mapping mSNPs that
overlap with CpGs, so these SNPs (and the CSs that include these SNPs)
should be removed from the results.</p>
<p>For the SNP-CpG association tests, it turns out that there are no
SNPs to “blacklist”:</p>
<pre class="r"><code>sum(with(assoc,pos == molecular_trait_id_pos))
# [1] 0</code></pre>
<p>Now identify the blacklisted mSNPs as the mSNPs that have the same
position as a CpG (I will use both “peak start” and “peak end”):</p>
<pre class="r"><code>ids_snp &lt;- c(with(snps_susie,paste(chr,pos,sep = &quot;_&quot;)),
             with(snps_fsusie,paste(chr,pos,sep = &quot;_&quot;)))
ids_cpg &lt;- with(cpgs_fsusie,
                c(paste(chr,peak_start,sep = &quot;_&quot;),
                  paste(chr,peak_end,sep = &quot;_&quot;)))
blacklisted_ids &lt;- intersect(ids_snp,ids_cpg)</code></pre>
<p>Now remove all SuSiE and fSuSiE CSs containing one or more
blacklisted SNPs:</p>
<pre class="r"><code>ids            &lt;- with(snps_susie,paste(chr,pos,sep = &quot;_&quot;))
rows           &lt;- which(is.element(ids,blacklisted_ids))
blacklisted_cs &lt;- unique(snps_susie[rows,]$cs)
snps_susie     &lt;- subset(snps_susie,!is.element(cs,blacklisted_cs))
ids            &lt;- with(snps_fsusie,paste(chr,pos,sep = &quot;_&quot;))
rows           &lt;- which(is.element(ids,blacklisted_ids))
blacklisted_cs &lt;- unique(snps_fsusie[rows,]$cs)
snps_fsusie    &lt;- subset(snps_fsusie,!is.element(cs,blacklisted_cs))
snps_susie     &lt;- transform(snps_susie,cs = factor(cs))
snps_fsusie    &lt;- transform(snps_fsusie,cs = factor(cs))</code></pre>
<p>These histograms summarize the number of CSs per TAD:</p>
<pre class="r"><code>pdat1 &lt;- get_cs_vs_tad_size(snps_susie)
pdat2 &lt;- get_cs_vs_tad_size(snps_fsusie)
pdat1 &lt;- transform(pdat1,num_cs = factor(num_cs,1:20))
pdat2 &lt;- transform(pdat2,num_cs = factor(num_cs,1:20))
p1 &lt;- ggplot(pdat1,aes(x = num_cs)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;darkblue&quot;) +
  scale_x_discrete(drop = FALSE) +
  labs(x = &quot;number of CSs&quot;,y = &quot;number of TADs&quot;,title = &quot;SuSiE-topPC&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(pdat2,aes(x = num_cs)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;darkblue&quot;) +
  scale_x_discrete(drop = FALSE) +
  labs(x = &quot;number of CSs&quot;,y = &quot;number of TADs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/num-cs-per-tad-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Compare discovery of causal SNPs (number of CSs) in the SuSiE-topPC
and fSuSiE analyses:</p>
<pre class="r"><code>dat1 &lt;- get_cs_vs_tad_size(snps_susie)
dat2 &lt;- get_cs_vs_tad_size(snps_fsusie)
dat1 &lt;- dat1[c(1,3)]
dat2 &lt;- dat2[c(1,3)]
names(dat1) &lt;- c(&quot;tad&quot;,&quot;num_cs_susie&quot;)
names(dat2) &lt;- c(&quot;tad&quot;,&quot;num_cs_fsusie&quot;)
dat &lt;- merge(dat1,dat2,all = TRUE)
rows &lt;- which(is.na(dat$num_cs_susie))
dat[rows,&quot;num_cs_susie&quot;] &lt;- 0
pdat &lt;- melt(with(dat,table(num_cs_susie,num_cs_fsusie)))
rows &lt;- which(pdat$value == 0)
pdat[rows,&quot;value&quot;] &lt;- NA
p &lt;- ggplot(pdat,aes(x = num_cs_susie,y = num_cs_fsusie,size = value)) +
  geom_point(color = &quot;white&quot;,fill = &quot;darkblue&quot;,shape = 21) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_x_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_y_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_size(breaks = c(1,5,10,50,100)) +
  labs(x = &quot;SuSiE-topPC&quot;,y = &quot;fSuSiE&quot;,size = &quot;number of TADs&quot;) +
  theme_cowplot(font_size = 10)
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/compare-num-cs-per-tad-1.png" width="450" style="display: block; margin: auto;" /></p>
<p>Flag the “duplicate” CSs:</p>
<pre class="r"><code>root_cs_susie  &lt;- create_cs_maps(snps_susie)$root
root_cs_fsusie &lt;- create_cs_maps(snps_fsusie)$root</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Then remove the duplicate CSs:</p>
<pre class="r"><code>snps_susie  &lt;- subset(snps_susie,is.element(cs,root_cs_susie))
snps_fsusie &lt;- subset(snps_fsusie,is.element(cs,root_cs_fsusie))
snps_susie  &lt;- transform(snps_susie,cs = factor(cs))
snps_fsusie &lt;- transform(snps_fsusie,cs = factor(cs))</code></pre>
<p>Also remove the duplicate CSs from the fSuSiE CpG-level results:</p>
<pre class="r"><code>nodup_cs    &lt;- levels(snps_fsusie$cs)
cpgs_fsusie &lt;- subset(cpgs_fsusie,is.element(cs,nodup_cs))
cpgs_fsusie &lt;- transform(cpgs_fsusie,cs = factor(cs))</code></pre>
<p>Compare the sizes of the CSs in the SuSiE-topPC and fSuSiE
analyses:</p>
<pre class="r"><code>bins &lt;- c(0,1,2,5,10,20,Inf)
snps_susie     &lt;- transform(snps_susie,cs = factor(cs))
snps_fsusie    &lt;- transform(snps_fsusie,cs = factor(cs))
cs_size_susie  &lt;- as.numeric(table(snps_susie$cs))
cs_size_fsusie &lt;- as.numeric(table(snps_fsusie$cs))
cs_size_susie  &lt;- cut(cs_size_susie,bins)
cs_size_fsusie &lt;- cut(cs_size_fsusie,bins)
levels(cs_size_susie) &lt;- bins[-1]
levels(cs_size_fsusie) &lt;- bins[-1]
p1 &lt;- ggplot(data.frame(cs_size = cs_size_susie),aes(x = cs_size)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;darkblue&quot;,
                 width = 0.65) +
  labs(x = &quot;CS size&quot;,y = &quot;number of CSs&quot;,title = &quot;SuSiE-topPC&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(data.frame(cs_size = cs_size_fsusie),aes(x = cs_size)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;darkblue&quot;,
                 width = 0.65) +
  scale_y_continuous(breaks = seq(0,1e4,2000)) +             
  labs(x = &quot;CS size&quot;,y = &quot;number of CSs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/cs-sizes-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>Here are the exact numbers:</p>
<pre class="r"><code>table(cs_size_susie)
table(cs_size_fsusie)
# cs_size_susie
#   1   2   5  10  20 Inf 
# 328 156 343 327 348 650 
# cs_size_fsusie
#    1    2    5   10   20  Inf 
# 6355 1683 1658  766  417  248</code></pre>
<p>We expect that most of the causal SNPs to be very close to the
nearest TSS. Let’s check this. This will involve a couple of
intermediate calculations.</p>
<pre class="r"><code>snps_assoc  &lt;- get_top_snp_per_location(assoc)
snps_assoc  &lt;- add_min_dist_to_tss(snps_assoc,genes)
snps_susie  &lt;- add_min_dist_to_tss(snps_susie,genes)
snps_fsusie &lt;- add_min_dist_to_tss(snps_fsusie,genes)</code></pre>
<p>For SuSiE-topPC and fSuSiE, we calculate “weighted” counts of SNPs,
in which the weights are given by the PIPs.</p>
<pre class="r"><code>bin_size &lt;- 10000
bins &lt;- c(-Inf,seq(-2e5,2e5,bin_size),Inf)
bins &lt;- bins + bin_size/2
counts_assoc &lt;- as.numeric(table(cut(snps_assoc$min_dist_to_tss,bins)))
counts_susie &lt;- tapply(snps_susie$pip,
                       cut(snps_susie$min_dist_to_tss,bins),
                       function (x) sum(x,na.rm = TRUE))
counts_fsusie &lt;- tapply(snps_fsusie$pip,
                        cut(snps_fsusie$min_dist_to_tss,bins),
                        function (x) sum(x,na.rm = TRUE))</code></pre>
<p>Now we can plot the result:</p>
<pre class="r"><code>n &lt;- length(bins)
i &lt;- seq(2,n-2)
bin_centers   &lt;- bins[i] + bin_size/2
counts_assoc  &lt;- counts_assoc[i]
counts_susie  &lt;- counts_susie[i]
counts_fsusie &lt;- counts_fsusie[i]
counts_assoc  &lt;- counts_assoc/sum(counts_assoc)
counts_susie  &lt;- counts_susie/sum(counts_susie)
counts_fsusie &lt;- counts_fsusie/sum(counts_fsusie)
pdat &lt;- data.frame(method = rep(c(&quot;assoc&quot;,&quot;susie&quot;,&quot;fsusie&quot;),
                                each = length(bin_centers)),
                   dist   = rep(bin_centers/1000,times = 3),
                   freq   = c(counts_assoc,counts_susie,counts_fsusie),
                   stringsAsFactors = TRUE)
p &lt;- ggplot(pdat,aes(x = dist,y = freq,color = method)) +
  geom_line(linewidth = 0.5) +
  geom_point(size = 1) +
  scale_x_continuous(breaks = seq(-200,200,50)) +
  scale_y_continuous(breaks = seq(0,1,0.05)) +
  scale_color_manual(values = c(&quot;darkblue&quot;,&quot;darkorange&quot;,&quot;dodgerblue&quot;)) +
  labs(x = &quot;distance to TSS (kb)&quot;,y = &quot;proportion of SNPs&quot;) +
  theme_cowplot(font_size = 10)
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/plot-weighted-distance-to-tss-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>Now let’s look closely at the 1-SNP CSs.</p>
<pre class="r"><code>cs1snp_susie  &lt;- table(snps_susie$cs)
cs1snp_susie  &lt;- names(cs1snp_susie)[cs1snp_susie == 1]
cs1snp_susie  &lt;- subset(snps_susie,is.element(cs,cs1snp_susie))
cs1snp_fsusie &lt;- table(snps_fsusie$cs)
cs1snp_fsusie &lt;- names(cs1snp_fsusie)[cs1snp_fsusie == 1]
cs1snp_fsusie &lt;- subset(snps_fsusie,is.element(cs,cs1snp_fsusie))</code></pre>
<p>Many of the PIPs in the 1-SNP CSs are 1 or very close to 1:</p>
<pre class="r"><code>table(cut(1 - cs1snp_susie$pip,c(0,0.0001,0.001,0.01,1)))
table(cut(1 - cs1snp_fsusie$pip,c(0,0.0001,0.001,0.01,1)))
# 
#     (0,0.0001] (0.0001,0.001]   (0.001,0.01]       (0.01,1] 
#            122             33             59             77 
# 
#     (0,0.0001] (0.0001,0.001]   (0.001,0.01]       (0.01,1] 
#           1989            477            638            638</code></pre>
<p>Now let’s turn to the recovery of affected CpGs.</p>
<p>Counting the number of CpGs per TAD is quite simple from the way the
fSuSiE results were compiled. It involves counting the number of unique
CpGs in each TAD:</p>
<pre class="r"><code>cpgs_per_tad_fsusie &lt;-
  with(cpgs_fsusie,tapply(ID,region,function (x) length(unique(x))))</code></pre>
<p>Counting the number of CpGs per TAD from the SNP-CpG association
tests is a little more complicated because the CpGs were not assigned to
TADs in the results.</p>
<pre class="r"><code>tads &lt;- get_tad_info(levels(cpgs_fsusie$region))
cpgs_per_tad_assoc &lt;- count_features_per_tad(assoc,tads)</code></pre>
<p>These plots summarize the number of affected CpGs per TAD:</p>
<pre class="r"><code>cpgs_per_tad_assoc[cpgs_per_tad_assoc == 0] &lt;- NA
cpgs_per_tad_fsusie[cpgs_per_tad_fsusie == 0] &lt;- NA
pdat1 &lt;- data.frame(x = cpgs_per_tad_assoc)
pdat2 &lt;- data.frame(x = cpgs_per_tad_fsusie)
p1 &lt;- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 64) +
  xlim(0,2000) +
  labs(x = &quot;number of CpGs&quot;,y = &quot;number of TADs&quot;,
       title = &quot;SNP-CpG association tests&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 64) +
  xlim(0,2000) +
  labs(x = &quot;number of CpGs&quot;,y = &quot;number of TADs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/plot-cpgs-per-tad-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-cpgs-per-tad-1">
Past versions of plot-cpgs-per-tad-1.png
</button>
</p>
<div id="fig-plot-cpgs-per-tad-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/plot-cpgs-per-tad-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>A small number of TADs have more than 2,000 affected CpGs:</p>
<pre class="r"><code>sum(cpgs_per_tad_assoc &gt; 2000,na.rm = TRUE)
sum(cpgs_per_tad_fsusie &gt; 2000,na.rm = TRUE)
# [1] 13
# [1] 5</code></pre>
<p>This plot compares the number of affected CpGs identified by fSuSiE
and the SNP-CpG association tests:</p>
<pre class="r"><code>pdat &lt;- data.frame(assoc  = cpgs_per_tad_assoc,
                   fsusie = cpgs_per_tad_fsusie)
p &lt;- ggplot(pdat,aes(x = assoc,y = fsusie)) +
  geom_point(color = &quot;darkblue&quot;) +
  geom_abline(intercept = 0,slope = 1,color = &quot;magenta&quot;,linetype = &quot;dotted&quot;) +
  labs(x = &quot;SNP-CpG association tests&quot;,y = &quot;fSusiE&quot;) +
  xlim(0,4100) + 
  ylim(0,4100) + 
  theme_cowplot(font_size = 10)
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/compare-cpgs-per-tad-1.png" width="300" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare-cpgs-per-tad-1">
Past versions of compare-cpgs-per-tad-1.png
</button>
</p>
<div id="fig-compare-cpgs-per-tad-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/compare-cpgs-per-tad-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>One advantage of fSuSiE is that it is able to tell us which molecular
features are affected by which SNPs, and therefore it provides a more
coherent summary of how the SNPs affect methylation levels at a locus.
To examine this quantitatively, we compare the number of affected CpGs
per CS for fSuSiE to the number of affected CpGs per SNP from the
association tests. The result is much fewer CSs and many more affected
CpGs per CS:</p>
<pre class="r"><code>x &lt;- factor(assoc$variant_id)
cpgs_per_snp_assoc &lt;- tapply(assoc$molecular_trait_id,x,
                             function (x) length(unique(x)))
rm(x)
cpgs_per_snp_fsusie &lt;-
  with(cpgs_fsusie,tapply(ID,cs,function (x) length(unique(x))))
pdat1 &lt;- data.frame(x = cpgs_per_snp_assoc)
pdat2 &lt;- data.frame(x = cpgs_per_snp_fsusie)
pdat1 &lt;- subset(pdat1,x &lt;= 25)
pdat2 &lt;- subset(pdat2,x &lt;= 150)
p1 &lt;- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 25) +
  labs(x = &quot;number of CpGs&quot;,y = &quot;number of SNPs&quot;,
       title = &quot;SNP-CpG association tests&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 25) +
  labs(x = &quot;number of CpGs&quot;,y = &quot;number of CSs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/count-cpgs-per-snp-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-count-cpgs-per-snp-1">
Past versions of count-cpgs-per-snp-1.png
</button>
</p>
<div id="fig-count-cpgs-per-snp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/4adf2dd19b17c506f500c6a17f2d3f186a3939ea/docs/figure/rosmap_overview.Rmd/count-cpgs-per-snp-1.png" target="_blank">4adf2dd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/dade7d2d7b0b7b39aa36a06a689b4ea4b96a98bf/docs/figure/rosmap_overview.Rmd/count-cpgs-per-snp-1.png" target="_blank">dade7d2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/618cae4fa9aadcbe31f55487e1b478d485770fba/docs/figure/rosmap_overview.Rmd/count-cpgs-per-snp-1.png" target="_blank">618cae4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/count-cpgs-per-snp-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>A small proportion of the SNPs and CSs are not plotted because they
have an unusually large number of CpGs:</p>
<pre class="r"><code>mean(cpgs_per_snp_assoc &gt; 25)
mean(cpgs_per_snp_fsusie &gt; 150)
# [1] 0.01052395
# [1] 0.02068903</code></pre>
<p>Finally, this plot shows the distribution of the distances between
the mSNP and the <em>nearest CpG</em> affected by that SNP:</p>
<pre class="r"><code>cpgs_fsusie_cs1snp &lt;- subset(cpgs_fsusie,
                             is.element(cs,cs1snp_fsusie$cs))
rows &lt;- match(cpgs_fsusie_cs1snp$cs,cs1snp_fsusie$cs)
cpgs_fsusie_cs1snp$variant_pos &lt;- cs1snp_fsusie[rows,&quot;pos&quot;]
cpgs_fsusie_cs1snp &lt;- transform(cpgs_fsusie_cs1snp,
                                cs = factor(cs),
                                dist = peak_start - variant_pos)
pdat &lt;- tapply(cpgs_fsusie_cs1snp$dist,cpgs_fsusie_cs1snp$cs,
               function (x) {
                 i &lt;- which.min(abs(x))
                 return(x[i])
               })
pdat &lt;- data.frame(dist = pdat/1000)
p &lt;- ggplot(subset(pdat,abs(dist) &lt; 100),aes(x = dist)) +
  geom_histogram(bins = 128,color = &quot;tomato&quot;,fill = &quot;tomato&quot;) +
  labs(x = &quot;CpG - mSNP pos. (kb)&quot;,
       y = &quot;number of mSNPs&quot;) +
  theme_cowplot(font_size = 10)
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/dist-msnp-cpg-cs1snp-1.png" width="300" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table(abs(pdat$dist) &lt; 100)
# 
# FALSE  TRUE 
#   256  6093</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS 15.4.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.1.3     ggplot2_3.5.0     dplyr_1.1.4       data.table_1.15.2
# 
# loaded via a namespace (and not attached):
#  [1] sass_0.4.9        utf8_1.2.4        generics_0.1.3    stringi_1.8.3    
#  [5] digest_0.6.34     magrittr_2.0.3    evaluate_1.0.3    grid_4.3.3       
#  [9] fastmap_1.1.1     plyr_1.8.9        R.oo_1.26.0       rprojroot_2.0.4  
# [13] workflowr_1.7.1   jsonlite_1.8.8    R.utils_2.12.3    whisker_0.4.1    
# [17] promises_1.2.1    fansi_1.0.6       scales_1.3.0      textshaping_0.3.7
# [21] jquerylib_0.1.4   cli_3.6.4         rlang_1.1.5       R.methodsS3_1.8.2
# [25] munsell_0.5.0     withr_3.0.2       cachem_1.0.8      yaml_2.3.8       
# [29] tools_4.3.3       reshape2_1.4.4    colorspace_2.1-0  httpuv_1.6.14    
# [33] vctrs_0.6.5       R6_2.5.1          lifecycle_1.0.4   git2r_0.33.0     
# [37] stringr_1.5.1     fs_1.6.5          ragg_1.2.7        pkgconfig_2.0.3  
# [41] pillar_1.9.0      bslib_0.6.1       later_1.3.2       gtable_0.3.4     
# [45] glue_1.8.0        Rcpp_1.0.12       systemfonts_1.0.6 xfun_0.42        
# [49] tibble_3.2.1      tidyselect_1.2.1  highr_0.10        knitr_1.45       
# [53] farver_2.1.1      htmltools_0.5.8.1 labeling_0.4.3    rmarkdown_2.26   
# [57] compiler_4.3.3</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
