<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="William Denault, Hao Sun, Angjing Liu, Peter Carbonetto, Gao Wang" />


<title>Examine methylation and H3K27ac fine-mapping results from the ROSMAP data</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fsusie experiments</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/fsusie-experiments">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Examine methylation and H3K27ac
fine-mapping results from the ROSMAP data</h1>
<h4 class="author">William Denault, Hao Sun, Angjing Liu, Peter
Carbonetto, Gao Wang</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong>
<code>fsusie-experiments/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
get-cpgs-per-tad-assoc
</li>
<li>
remove-duplicated-cs-ha
</li>
<li>
remove-duplicated-cs-methyl
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>rosmap_overview_cache</code> and re-run the analysis. To have
workflowr automatically delete the cache directory prior to building the
file, set <code>delete_cache = TRUE</code> when running
<code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfsusieexperimentstree748b8b0f5ebcc202993cc01ec6bb9ae3b468c122targetblank748b8b0a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/stephenslab/fsusie-experiments/tree/748b8b0f5ebcc202993cc01ec6bb9ae3b468c122" target="_blank">748b8b0</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfsusieexperimentstree748b8b0f5ebcc202993cc01ec6bb9ae3b468c122targetblank748b8b0a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/stephenslab/fsusie-experiments/tree/748b8b0f5ebcc202993cc01ec6bb9ae3b468c122" target="_blank">748b8b0</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/affected_cpgs_per_snp.pdf
    Untracked:  analysis/affected_cpgs_per_tad.pdf
    Untracked:  analysis/affected_peaks_per_snp.pdf
    Untracked:  analysis/affected_peaks_per_tad.pdf
    Untracked:  analysis/cs_per_tad_scatterplot_h3k27ac.pdf
    Untracked:  analysis/cs_per_tad_scatterplot_methyl.pdf
    Untracked:  analysis/cs_sizes_h3k27ac.pdf
    Untracked:  analysis/cs_sizes_methyl.pdf
    Untracked:  analysis/dist_to_tss_h3k27ac.pdf
    Untracked:  analysis/dist_to_tss_methyl.pdf
    Untracked:  analysis/rosmap.afreq
    Untracked:  analysis/rosmap_overview_cache/
    Untracked:  analysis/tad_sizes_histogram.pdf
    Untracked:  analysis/temp1.txt
    Untracked:  analysis/temp2.txt
    Untracked:  data/ROSMAP_NIA_WGS.leftnorm.bcftools_qc.plink_qc.afreq.gz
    Untracked:  data/analysis_result/Fungen_xQTL.ENSG00000163808.cis_results_db.export.rds
    Untracked:  data/analysis_result/ROSMAP_haQTL.chr3_43915257_48413435.fsusie_mixture_normal_top_pc_weights.rds
    Untracked:  data/analysis_result/ROSMAP_mQTL.chr3_43915257_48413435.fsusie_mixture_normal_top_pc_weights.rds
    Untracked:  outputs/CASS4_obj.RData
    Untracked:  outputs/CD2AP_all_effects.RData
    Untracked:  outputs/CD2AP_obj.RData
    Untracked:  outputs/CR1_CR2_all_effects.RData
    Untracked:  outputs/CR1_CR2_obj.RData
    Untracked:  outputs/ROSMAP_haQTL_cs_effect_ha_peak_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_haQTL_cs_snp_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_haQTL_cs_snp_toppc1_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_haQTL_qtl_snp_qval0.05.tsv.gz
    Untracked:  outputs/ROSMAP_haQTL_qtl_snp_qval0.05_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_cs_effect_cpg_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_cs_snp_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_cs_snp_toppc1_annotation.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_qtl_snp_qval0.05.tsv.gz
    Untracked:  outputs/ROSMAP_mQTL_qtl_snp_qval0.05_annotation.tsv.gz

Unstaged changes:
    Modified:   analysis/rosmap_overview.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/rosmap_overview.Rmd</code>) and
HTML (<code>docs/rosmap_overview.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/748b8b0f5ebcc202993cc01ec6bb9ae3b468c122/analysis/rosmap_overview.Rmd" target="_blank">748b8b0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-25
</td>
<td>
Added more detailed analysis of the 1-SNP CSs for methylation in
rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/9dad9a938e354a4dfa012bc086d22e5ba8e77fe8/analysis/rosmap_overview.Rmd" target="_blank">9dad9a9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-24
</td>
<td>
Working on additional steps to identify and remove blacklisted mSNPs.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/dade7d2d7b0b7b39aa36a06a689b4ea4b96a98bf/docs/rosmap_overview.html" target="_blank">dade7d2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Ran workflowr::wflow_publish("rosmap_overview.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/14e44873b58948fbaf51301dc25b5edaee71cc2d/analysis/rosmap_overview.Rmd" target="_blank">14e4487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Updated the plot in rosmap_overview.Rmd showing num. peaks vs. num. CSs
for fSusiE H3K27ac results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/1fd676e4f2f1cf362aaad605b20ee56f1865b6d6/analysis/rosmap_overview.Rmd" target="_blank">1fd676e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Added step to rosmap_overview analysis to remove duplicate CSs for
H3K27ac.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/cdc55a21371d4e9349f9092a7d433f0b5642f20d/analysis/rosmap_overview.Rmd" target="_blank">cdc55a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Fixed histogram showing no. cpgs vs. no. CSs in rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/2c0d57446fe7bcb5a9e69d56498b5b509f173b51/analysis/rosmap_overview.Rmd" target="_blank">2c0d574</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
<td>
Small fix to the CS size histograms for the methylation results in
rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/e1fe364dc6626e234e44231c8ec657e5e34b7cac/analysis/rosmap_overview.Rmd" target="_blank">e1fe364</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-04
</td>
<td>
I have the CD2AP zoom-in plot mostly done except for the panel showing
the raw data (beta values).
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/46d0b351bb74d6e04428c15090865c1e6003a996/analysis/rosmap_overview.Rmd" target="_blank">46d0b35</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-04
</td>
<td>
Removed some code in the rosmap_overview.Rmd analysis into a separate
code chunk.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/9416011b5977235a8c3c1f0e23bec16807f29e13/analysis/rosmap_overview.Rmd" target="_blank">9416011</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-04
</td>
<td>
Added function create_cs_maps() to help remove ‘duplicated’ CSs.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/618cae4fa9aadcbe31f55487e1b478d485770fba/analysis/rosmap_overview.Rmd" target="_blank">618cae4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-03
</td>
<td>
A few fixes to the CR1 eQTL panel.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/618cae4fa9aadcbe31f55487e1b478d485770fba/docs/rosmap_overview.html" target="_blank">618cae4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-03
</td>
<td>
A few fixes to the CR1 eQTL panel.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/3bb4ba0dd104f7fc4d5447b6c8548273d2e777cd/docs/rosmap_overview.html" target="_blank">3bb4ba0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-28
</td>
<td>
Ran wflow_publish("rosmap_overview.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/dcece5bb310e360ec5007a6e5d0e9d03f672748f/analysis/rosmap_overview.Rmd" target="_blank">dcece5b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-28
</td>
<td>
Added code chunks to rosmap_overview.Rmd to save some more plots to PDF.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/a70588c212c06901e4b8b90a29735c9424ad9fbc/analysis/rosmap_overview.Rmd" target="_blank">a70588c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-28
</td>
<td>
Added code chunks to rosmap_overview.Rmd to save a couple plots to PDFs.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/c07956874db10cbee93e0c44ce20bfc946db2004/docs/rosmap_overview.html" target="_blank">c079568</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
Ran wflow_publish("rosmap_overview.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/2bd0de32c1e91e3107a779a9ec9a564199d940e2/analysis/rosmap_overview.Rmd" target="_blank">2bd0de3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
wflow_publish("rosmap_overview.Rmd", verbose = TRUE, view = FALSE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/53d4a33a43b153d27ddac8618759b18f72d6942e/analysis/rosmap_overview.Rmd" target="_blank">53d4a33</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
Added code chunks to rosmap_overview.Rmd for generating PDFs of some of
the plots.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/d5c9e372a24cb6a5d91124df71f15aab6012aef7/analysis/rosmap_overview.Rmd" target="_blank">d5c9e37</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
A few fixes to rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/544235d894e6bb5e84e4e904b560ac62286fa225/analysis/rosmap_overview.Rmd" target="_blank">544235d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
<td>
Added code chunks to save some of the plots in the rosmap_overview
analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/aece910973f3984cf07ffbe6089a898f58910c41/docs/rosmap_overview.html" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Ran workflowr::wflow_publish("rosmap_overview.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/0eb82957ca28e83cf1754315d54529997e31f92b/analysis/rosmap_overview.Rmd" target="_blank">0eb8295</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
wflow_publish("rosmap_overview.Rmd", view = FALSE, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/6de898eb207e5f45002c95f841ba794d45426b4f/analysis/rosmap_overview.Rmd" target="_blank">6de898e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Added plots to the rosmap_overview analysis summarizing the recovery of
affected H3K27ac peaks.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/7c8d51d9607a08e886dd1760c9b4c48cd46caa11/analysis/rosmap_overview.Rmd" target="_blank">7c8d51d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Added TSS plot for H3K27ac in rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/8e174862fa2b5e865601f1bb0fb4cd521c67d67a/analysis/rosmap_overview.Rmd" target="_blank">8e17486</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Added code to create a plot showing the density of causal SNPs near the
closest TSS.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/18c11b56e652a26b056cd61e0eaa25fb62455841/analysis/rosmap_overview.Rmd" target="_blank">18c11b5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
<td>
Added code to rosmap_overview.Rmd to load gene annotations.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/60bbe231bfe3abf2948ff7be5ab88137ffaf0715/analysis/rosmap_overview.Rmd" target="_blank">60bbe23</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-22
</td>
<td>
Added a couple notes to the rosmap_summary analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/00bb89c30a7c47d5661e91ad6ae3f3c4a116eff0/analysis/rosmap_overview.Rmd" target="_blank">00bb89c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added note to rosmap_overview.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/7adfd4d32dae6b06ae384254c0f035ccaa1039d9/analysis/rosmap_overview.Rmd" target="_blank">7adfd4d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
A few fixes to the rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c087a58be2e3a5f3a2165fa24c6d484e85293009/analysis/rosmap_overview.Rmd" target="_blank">c087a58</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added some notes to the rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/5e9432b3f27e5f8ac453173d1b25c702e087e291/analysis/rosmap_overview.Rmd" target="_blank">5e9432b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added plots to the rosmap_overview analysis summarizing the affected
CpGs identified by fSuSiE and the association tests.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/593c7e0f140991467385ef93d92611b8060d9324/analysis/rosmap_overview.Rmd" target="_blank">593c7e0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added some code for analyzing the HA peak results in the rosmap_overview
analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/12ed0b15e3b6a2b036a6a414ce55de63104ac175/analysis/rosmap_overview.Rmd" target="_blank">12ed0b1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Small fix.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/e663ab0144e239fbda7948af02ae536cb3ddb91a/analysis/rosmap_overview.Rmd" target="_blank">e663ab0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-21
</td>
<td>
Added code to read in HA_peak results in rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/71954f15429162c72e3c8013edbb72accecc71a9/docs/rosmap_overview.html" target="_blank">71954f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added plots summarizing H3K27ac results to rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/1ad355ae94d1bb9a770faeb0ac56392549a38b4c/analysis/rosmap_overview.Rmd" target="_blank">1ad355a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
wflow_publish("rosmap_overview.Rmd", view = FALSE, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/c84e5c046a56501c872e946ab6839eac67d0a41a/docs/rosmap_overview.html" target="_blank">c84e5c0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Rebuilt the rosmap_overview analysis with the new results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/1c8eeb7f07827e141d64f671b8a38821e1ac7c86/analysis/rosmap_overview.Rmd" target="_blank">1c8eeb7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
wflow_publish("rosmap_overview.Rmd", view = FALSE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/acfadd18f631b80fc953f7c282ada7aeeba48c68/analysis/rosmap_overview.Rmd" target="_blank">acfadd1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Made a few improvements to the code and text of the rosmap_analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c102af90b142616368e65e5deddab968daea046f/analysis/rosmap_overview.Rmd" target="_blank">c102af9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added a scatterplot comparing number of CSs per TAD (susie vs. fsusie).
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/12f2fd34bb981b86976ea16a9d293b4b6a0c55e7/analysis/rosmap_overview.Rmd" target="_blank">12f2fd3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added some histograms on TAD CS sizes.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c69e187bb6e7fd10716665e4c59ebdf0fb927d84/analysis/rosmap_overview.Rmd" target="_blank">c69e187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Created plot showing TAD sizes from the methylation fine-mapping
results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/2a5c7060bfea715430e3f822a89a1ded9791f816/analysis/rosmap_overview.Rmd" target="_blank">2a5c706</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added code to the rosmap_overview analysis to load the methylation SNP
results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c3a01c7e937d5a86509a699d9e57504d6c583587/analysis/rosmap_overview.Rmd" target="_blank">c3a01c7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Added link for downloading data to rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/5c446c08a11735474bd94ded6e641065aff07171/docs/rosmap_overview.html" target="_blank">5c446c0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
First build of the rosmap_overview analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/7532908fe0418cc52f231593c2a621e8fc7719eb/analysis/rosmap_overview.Rmd" target="_blank">7532908</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
workflowr::wflow_publish("rosmap_overview.Rmd", verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/bc6d0a16f7dfe114a6a17d5cff05ab2eb457497f/analysis/rosmap_overview.Rmd" target="_blank">bc6d0a1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
<td>
Started working on rosmap_overview analysis.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><strong>Note:</strong> If you would like to run this analysis on your
computer, you will first need to download the fine-mapping outputs. They
can be downloaded from <a
href="https://uchicago.box.com/s/tt1vgg7vqayfthg0vsbl8gw0sdi0f1uo">here</a>.
Once you have downloaded the files, copy them to the “outputs”
subdirectory.</p>
<p>Load some packges and custom functions used in the code below:</p>
<pre class="r"><code>library(data.table)
library(dplyr)
library(ggplot2)
library(cowplot)
library(hexbin)
source(&quot;../code/merge_cs_sets.R&quot;)</code></pre>
<p>This is a function I use to Extract the gene annotations from the GTF
(“gene transfer format”) file. Here we keep only the annotated gene
transcripts for protein-coding genes as defined in the Ensembl/Havana
database.</p>
<pre class="r"><code>get_gene_annotations &lt;- function (gene_file) {
  out &lt;- fread(file = gene_file,sep = &quot;\t&quot;,header = FALSE,skip = 1)
  class(out) &lt;- &quot;data.frame&quot;
  names(out) &lt;- c(&quot;chromosome&quot;,&quot;source&quot;,&quot;feature&quot;,&quot;start&quot;,&quot;end&quot;,&quot;score&quot;,
                    &quot;strand&quot;,&quot;frame&quot;,&quot;attributes&quot;)
  out &lt;- out[c(&quot;chromosome&quot;,&quot;source&quot;,&quot;feature&quot;,&quot;start&quot;,&quot;end&quot;,&quot;strand&quot;,
               &quot;attributes&quot;)]
  out &lt;- transform(out,
                   chromosome = factor(chromosome),
                   source     = factor(source),
                   feature    = factor(feature),
                   strand     = factor(strand))
  out &lt;- subset(out,
                source == &quot;ensembl_havana&quot; &amp;
                feature == &quot;transcript&quot;)
  out &lt;-
    transform(out,
      ensembl   = sapply(strsplit(attributes,&quot;;&quot;),
                         function (x) substr(x[[1]],10,24)),
      gene_type = sapply(strsplit(attributes,&quot;;&quot;),
                         function (x) substr(x[[3]],13,nchar(x[[3]]) - 1)),
      gene_name = sapply(strsplit(attributes,&quot;;&quot;),
                         function (x) substr(x[[4]],13,nchar(x[[4]]) - 1)))
  out &lt;- out[-7]
  out &lt;- transform(out,gene_type = factor(gene_type))
  out &lt;- subset(out,gene_type == &quot;protein_coding&quot;)
  rownames(out) &lt;- NULL
  return(out)
}</code></pre>
<p>Load the gene annotations which will be used in some of the analyses
below. Specifically, I extract here only the annotated gene transcripts
for protein-coding genes as defined in the Ensembl/Havana database.</p>
<pre class="r"><code>gene_file &lt;-
  file.path(&quot;../data/genome_annotations&quot;,
    &quot;Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz&quot;)
genes &lt;- get_gene_annotations(gene_file)</code></pre>
<div id="methylation-snps" class="section level2">
<h2>Methylation SNPs</h2>
<p>First, I define a helper function for loading the enrichment
results:</p>
<pre class="r"><code># The &quot;n&quot; argument specifies the number of &quot;meta data&quot; columns.
# Columns after that are treated as the enrichment results. These
# columns contain only binary data (0 or 1) indicating whether or not
# the genomic feature (genetic variant or molecular trait location)
# is assigned that specific annotation.
read_enrichment_results &lt;- function (filename, n) {
  out &lt;- fread(filename,sep = &quot;\t&quot;,stringsAsFactors = FALSE,header = TRUE)
  class(out) &lt;- &quot;data.frame&quot;
  out &lt;- transform(out,chr = factor(chr))
  if (ncol(out) &gt; n) {
    cols &lt;- seq(n + 1,ncol(out))
    for (i in cols)
      out[[i]] &lt;- factor(out[[i]])
  }
  return(out)
}</code></pre>
<p>Next I load methylation SNP results generated by SuSiE-topPC, fSuSiE
and the SNP-CpG association testing:</p>
<pre class="r"><code>methyl_cpg_assoc_file  &lt;- &quot;../outputs/ROSMAP_mQTL_qtl_snp_qval0.05.tsv.gz&quot;
methyl_snps_susie_file &lt;-
  &quot;../outputs/ROSMAP_mQTL_cs_snp_toppc1_annotation.tsv.gz&quot;
methyl_snps_fsusie_file &lt;- &quot;../outputs/ROSMAP_mQTL_cs_snp_annotation.tsv.gz&quot;
methyl_snps_susie  &lt;- read_enrichment_results(methyl_snps_susie_file,n = 6)
methyl_snps_fsusie &lt;- read_enrichment_results(methyl_snps_fsusie_file,n = 7)
methyl_cpg_assoc   &lt;- read_enrichment_results(methyl_cpg_assoc_file,n = 8)
methyl_snps_susie$region &lt;-
  sapply(strsplit(methyl_snps_susie$cs,&quot;:&quot;,fixed = TRUE),&quot;[[&quot;,2)
methyl_snps_susie  &lt;- transform(methyl_snps_susie,
                                region = factor(region),
                                cs     = factor(cs),
                                pc     = factor(pc))
methyl_snps_fsusie &lt;- transform(methyl_snps_fsusie,
                                cs     = factor(cs),
                                region = factor(region),
                                study  = factor(study))</code></pre>
<p>Also, I load the allele frequencies computed by PLINK:</p>
<pre class="r"><code>afreq_file &lt;- &quot;../data/ROSMAP_NIA_WGS.leftnorm.bcftools_qc.plink_qc.afreq.gz&quot;
system(paste(&quot;gzcat&quot;,afreq_file,&quot;| cut -f 1,6 &gt; temp1.txt&quot;))
system(paste(&quot;gzcat&quot;,afreq_file,&quot;| cut -f 2 | cut -d &#39;:&#39; -f 2 |&quot;,
             &quot;cut -d &#39;_&#39; -f 1 &gt; temp2.txt&quot;))
system(paste(&quot;paste temp1.txt temp2.txt &gt; rosmap.afreq&quot;))            
afreq &lt;- fread(&quot;rosmap.afreq&quot;,sep = &quot;\t&quot;,stringsAsFactors = FALSE,
               header = TRUE,nrows = Inf)
class(afreq) &lt;- &quot;data.frame&quot;
names(afreq) &lt;- c(&quot;chr&quot;,&quot;maf&quot;,&quot;pos&quot;)
afreq &lt;- transform(afreq,
                   chr = factor(chr),
                   maf = pmin(maf,1 - maf))
afreq &lt;- subset(afreq,maf &gt; 0.001)
levels(afreq$chr) &lt;- paste0(&quot;chr&quot;,1:22)
afreq$id &lt;- sprintf(&quot;%s_%d&quot;,as.character(afreq$chr),afreq$pos)</code></pre>
<p>Add the allele frequencies to the methylation fine-mapping
results:</p>
<pre class="r"><code>ids  &lt;- with(methyl_snps_susie,paste(chr,pos,sep = &quot;_&quot;))
rows &lt;- match(ids,afreq$id)
methyl_snps_susie$maf &lt;- afreq[rows,&quot;maf&quot;]
ids  &lt;- with(methyl_snps_fsusie,paste(chr,pos,sep = &quot;_&quot;))
rows &lt;- match(ids,afreq$id)
methyl_snps_fsusie$maf &lt;- afreq[rows,&quot;maf&quot;]</code></pre>
<div id="tad-statistics" class="section level3">
<h3>TAD statistics</h3>
<p>This is the number of fine-mapping regions (TADs) that contained at
least one CS in each of the analyses:</p>
<pre class="r"><code>nlevels(methyl_snps_susie$region)
nlevels(methyl_snps_fsusie$region)
# [1] 1236
# [1] 1327</code></pre>
<p>This is a function we will use below to get the sizes of the TADs (in
Mb):</p>
<pre class="r"><code>get_tad_sizes &lt;- function (tads) {
  tads &lt;- strsplit(tads,&quot;_&quot;,fixed = TRUE)
  pos0 &lt;- as.numeric(sapply(tads,&quot;[[&quot;,2))
  pos1 &lt;- as.numeric(sapply(tads,&quot;[[&quot;,3))
  return((pos1 - pos0)/1e6)
}</code></pre>
<p>This plot summarizes the sizes of the TADs that were analyzed by
SuSiE-topPC and fSuSiE:</p>
<pre class="r"><code>plot_tad_sizes &lt;- function (tads) {
  tad_size &lt;- get_tad_sizes(tads)
  pdat &lt;- data.frame(tad_size = tad_size)
  return(ggplot(pdat,aes(x = tad_size)) +
         geom_histogram(color = &quot;white&quot;,fill = &quot;black&quot;,bins = 48) +
         labs(x = &quot;size (Mb)&quot;,y = &quot;number of TADs&quot;) +
         theme_cowplot(font_size = 10))
}
tads &lt;- levels(methyl_snps_fsusie$region)
p &lt;- plot_tad_sizes(tads) +
  scale_x_continuous(limits = c(2,9),breaks = 1:10) +
  scale_y_continuous(breaks = seq(0,100,10))
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/plot-methyl-tad-sizes-1.png" width="330" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-methyl-tad-sizes-1">
Past versions of plot-methyl-tad-sizes-1.png
</button>
</p>
<div id="fig-plot-methyl-tad-sizes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/plot-methyl-tad-sizes-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c84e5c046a56501c872e946ab6839eac67d0a41a/docs/figure/rosmap_overview.Rmd/plot-methyl-tad-sizes-1.png" target="_blank">c84e5c0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Some other useful statistics on the TAD sizes:</p>
<pre class="r"><code>tad_size &lt;- get_tad_sizes(tads)
length(tad_size)
range(tad_size)
mean(tad_size)
median(tad_size)
sum(tad_size &gt; 9)
# [1] 1327
# [1]  2.321 34.727
# [1] 4.545
# [1] 4.154
# [1] 19</code></pre>
</div>
<div id="identify-and-remove-blacklisted-msnps" class="section level3">
<h3>Identify and remove “blacklisted” mSNPs</h3>
<p>There are some technical concerns about fine-mapping mSNPs that are
also CpGs, so these SNPs (and the CSs that include these SNPs) should be
removed from the results.</p>
<p>For the SNP-CpG association tests, finding the “blacklisted” is quite
straightforward because they are just the mSNPs that have the same
positions as the CpGs:</p>
<pre class="r"><code>rows &lt;- which(with(methyl_cpg_assoc,pos == molecular_trait_id_pos))
length(rows)
ids  &lt;- with(methyl_cpg_assoc[rows,],paste(chr,pos,sep = &quot;_&quot;))
i &lt;- match(ids,afreq$id)
table(cut(afreq[i,&quot;maf&quot;],c(0,0.01,0.1,1)))
rows &lt;- which(with(methyl_cpg_assoc,pos != molecular_trait_id_pos))
methyl_cpg_assoc &lt;- methyl_cpg_assoc[rows,]
# [1] 35
# 
#   (0,0.01] (0.01,0.1]    (0.1,1] 
#         27          8          0</code></pre>
<p>For fSuSiE, it is a bit more complicated because we need to first
load the CpG results:</p>
<pre class="r"><code>methyl_cpg_fsusie_file &lt;-
  &quot;../outputs/ROSMAP_mQTL_cs_effect_cpg_annotation.tsv.gz&quot;
methyl_cpg_fsusie &lt;- read_enrichment_results(methyl_cpg_fsusie_file,n = 9)
methyl_cpg_fsusie$region &lt;-
  sapply(strsplit(methyl_cpg_fsusie$cs,&quot;:&quot;,fixed = TRUE),&quot;[[&quot;,2)
methyl_cpg_fsusie &lt;- transform(methyl_cpg_fsusie,
                               cs      = factor(cs), 
                               region  = factor(region),
                               context = factor(context))</code></pre>
<p>Now identify the blacklisted mSNPs as the mSNPs that have the same
position as a CpG (I will use both “peak start” and “peak end”):</p>
<pre class="r"><code>ids_snp &lt;- c(with(methyl_snps_susie,paste(chr,pos,sep = &quot;_&quot;)),
             with(methyl_snps_fsusie,paste(chr,pos,sep = &quot;_&quot;)))
ids_cpg &lt;- with(methyl_cpg_fsusie,
                c(paste(chr,peak_start,sep = &quot;_&quot;),
                  paste(chr,peak_end,sep = &quot;_&quot;)))
blacklisted_ids &lt;- intersect(ids_snp,ids_cpg)
length(blacklisted_ids)
# [1] 926</code></pre>
<p>Before removing the blacklisted SNPs from the results, let’s take a
quick look at them:</p>
<pre class="r"><code>ids &lt;- with(methyl_snps_fsusie,paste(chr,pos,sep = &quot;_&quot;))
rows &lt;- which(is.element(ids,blacklisted_ids))
methyl_snps_fsusie_blacklisted &lt;- methyl_snps_fsusie[rows,]
nrow(methyl_snps_fsusie_blacklisted)
sum(methyl_snps_fsusie_blacklisted$pip &gt; 0.5)
# [1] 1769
# [1] 1263</code></pre>
<p>It is interesting that a large fraction of the blacklisted SNPs have
a high PIPs. Let’s also look at the MAFs of these “blacklisted”
SNPs:</p>
<pre class="r"><code>ggplot(methyl_snps_fsusie_blacklisted,aes(x = log10(maf))) +
  geom_histogram(fill = &quot;darkblue&quot;,color = &quot;white&quot;,bins = 32) +
  labs(x = &quot;log10(MAF)&quot;,y = &quot;number of SNPs&quot;) +
  theme_cowplot(font_size = 10)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/blacklisted-snps-maf-hist-1.png" width="300" style="display: block; margin: auto;" /></p>
<p>Before moving forward with the analysis, remove all SuSiE and fSuSiE
CSs containing the blacklisted SNPs:</p>
<pre class="r"><code>ids                &lt;- with(methyl_snps_susie,paste(chr,pos,sep = &quot;_&quot;))
rows               &lt;- which(is.element(ids,blacklisted_ids))
blacklisted_cs     &lt;- unique(methyl_snps_susie[rows,]$cs)
methyl_snps_susie  &lt;- subset(methyl_snps_susie,!is.element(cs,blacklisted_cs))
ids                &lt;- with(methyl_snps_fsusie,paste(chr,pos,sep = &quot;_&quot;))
rows               &lt;- which(is.element(ids,blacklisted_ids))
blacklisted_cs     &lt;- unique(methyl_snps_fsusie[rows,]$cs)
methyl_snps_fsusie &lt;- subset(methyl_snps_fsusie,!is.element(cs,blacklisted_cs))
methyl_snps_susie  &lt;- transform(methyl_snps_susie,cs = factor(cs))
methyl_snps_fsusie &lt;- transform(methyl_snps_fsusie,cs = factor(cs))</code></pre>
</div>
<div id="compare-susie-toppc-and-fsusie-methylation-css"
class="section level3">
<h3>Compare SuSiE-topPC and fSuSiE methylation CSs</h3>
<p>These histograms summarize the number of CSs per TAD:</p>
<pre class="r"><code>get_cs_vs_tad_size &lt;- function (dat) {
  tads &lt;- levels(dat$region)
  out &lt;- data.frame(tad      = tads,
                    tad_size = get_tad_sizes(tads),
                    num_cs   = tapply(dat$cs,dat$region,
                                      function (x) length(unique(x))))
  rownames(out) &lt;- NULL
  return(out)
}
pdat1 &lt;- get_cs_vs_tad_size(methyl_snps_susie)
pdat2 &lt;- get_cs_vs_tad_size(methyl_snps_fsusie)
pdat1 &lt;- transform(pdat1,num_cs = factor(num_cs,1:20))
pdat2 &lt;- transform(pdat2,num_cs = factor(num_cs,1:20))
p1 &lt;- ggplot(pdat1,aes(x = num_cs)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;darkblue&quot;) +
  scale_x_discrete(drop = FALSE) +
  labs(x = &quot;number of CSs&quot;,y = &quot;number of TADs&quot;,title = &quot;SuSiE-topPC&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(pdat2,aes(x = num_cs)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;darkblue&quot;) +
  scale_x_discrete(drop = FALSE) +
  labs(x = &quot;number of CSs&quot;,y = &quot;number of TADs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/num-cs-per-tad-methyl-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-num-cs-per-tad-methyl-1">
Past versions of num-cs-per-tad-methyl-1.png
</button>
</p>
<div id="fig-num-cs-per-tad-methyl-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/71954f15429162c72e3c8013edbb72accecc71a9/docs/figure/rosmap_overview.Rmd/num-cs-per-tad-methyl-1.png" target="_blank">71954f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c84e5c046a56501c872e946ab6839eac67d0a41a/docs/figure/rosmap_overview.Rmd/num-cs-per-tad-methyl-1.png" target="_blank">c84e5c0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Compare discovery of causal SNPs (number of CSs) in the SuSiE-topPC
and fSuSiE analyses:</p>
<pre class="r"><code>dat1 &lt;- get_cs_vs_tad_size(methyl_snps_susie)
dat2 &lt;- get_cs_vs_tad_size(methyl_snps_fsusie)
dat1 &lt;- dat1[c(1,3)]
dat2 &lt;- dat2[c(1,3)]
names(dat1) &lt;- c(&quot;tad&quot;,&quot;num_cs_susie&quot;)
names(dat2) &lt;- c(&quot;tad&quot;,&quot;num_cs_fsusie&quot;)
dat &lt;- merge(dat1,dat2,all = TRUE)
rows &lt;- which(is.na(dat$num_cs_susie))
dat[rows,&quot;num_cs_susie&quot;] &lt;- 0
pdat &lt;- melt(with(dat,table(num_cs_susie,num_cs_fsusie)))
rows &lt;- which(pdat$value == 0)
pdat[rows,&quot;value&quot;] &lt;- NA
p &lt;- ggplot(pdat,aes(x = num_cs_susie,y = num_cs_fsusie,size = value)) +
  geom_point(color = &quot;white&quot;,fill = &quot;darkblue&quot;,shape = 21) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_x_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_y_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_size(breaks = c(1,5,10,50,100)) +
  labs(x = &quot;SuSiE-topPC&quot;,y = &quot;fSuSiE&quot;,size = &quot;number of TADs&quot;) +
  theme_cowplot(font_size = 10)
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/compare-num-cs-per-tad-methyl-1.png" width="450" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare-num-cs-per-tad-methyl-1">
Past versions of compare-num-cs-per-tad-methyl-1.png
</button>
</p>
<div id="fig-compare-num-cs-per-tad-methyl-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c84e5c046a56501c872e946ab6839eac67d0a41a/docs/figure/rosmap_overview.Rmd/compare-num-cs-per-tad-methyl-1.png" target="_blank">c84e5c0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For plots counting CSs across the genome, I need to address the issue
that the same CSs are sometimes discovered in overlapping TADs. To this
end, this function removes CSs so that no two CSs share the same
SNP:</p>
<pre class="r"><code>create_cs_maps &lt;- function (info_df) {
  info_df %&gt;% 
    arrange(cs) %&gt;%
    group_by(variant_id) %&gt;%
    mutate(cs_set = list(sort(unique(cs))),
           cs_set = sapply(cs_set,paste,collapse = &quot;,&quot;)) %&gt;%
    ungroup %&gt;%
    count(cs,cs_set) %&gt;%
    group_by(cs) %&gt;%
    summarize(
      cs_set = paste(sort(unique(unlist(strsplit(cs_set,&quot;,&quot;)))),
                     collapse = &quot;,&quot;),.groups = &quot;drop&quot;) %&gt;%
    merge_cs_sets
}</code></pre>
<p>Now let’s use this function to flag and remove the “duplicate”
CSs.</p>
<pre class="r"><code>root_cs_susie  &lt;- create_cs_maps(methyl_snps_susie)$root
root_cs_fsusie &lt;- create_cs_maps(methyl_snps_fsusie)$root
methyl_snps_susie_nodup  &lt;- subset(methyl_snps_susie,
                                   is.element(cs,root_cs_susie))
methyl_snps_fsusie_nodup &lt;- subset(methyl_snps_fsusie,
                                   is.element(cs,root_cs_fsusie))</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Compare the sizes of the CSs in the SuSiE-topPC and fSuSiE
analyses:</p>
<pre class="r"><code>bins &lt;- c(0,1,2,5,10,20,Inf)
methyl_snps_susie_nodup &lt;- transform(methyl_snps_susie_nodup,cs = factor(cs))
methyl_snps_fsusie_nodup &lt;- transform(methyl_snps_fsusie_nodup,cs = factor(cs))
cs_size_susie  &lt;- as.numeric(table(methyl_snps_susie_nodup$cs))
cs_size_fsusie &lt;- as.numeric(table(methyl_snps_fsusie_nodup$cs))
cs_size_susie  &lt;- cut(cs_size_susie,bins)
cs_size_fsusie &lt;- cut(cs_size_fsusie,bins)
levels(cs_size_susie) &lt;- bins[-1]
levels(cs_size_fsusie) &lt;- bins[-1]
p1 &lt;- ggplot(data.frame(cs_size = cs_size_susie),aes(x = cs_size)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;darkblue&quot;,
                 width = 0.65) +
  labs(x = &quot;CS size&quot;,y = &quot;number of CSs&quot;,title = &quot;SuSiE-topPC&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(data.frame(cs_size = cs_size_fsusie),aes(x = cs_size)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;darkblue&quot;,
                 width = 0.65) +
  scale_y_continuous(breaks = seq(0,1e4,2000)) +             
  labs(x = &quot;CS size&quot;,y = &quot;number of CSs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/cs-sizes-methyl-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cs-sizes-methyl-1">
Past versions of cs-sizes-methyl-1.png
</button>
</p>
<div id="fig-cs-sizes-methyl-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/dade7d2d7b0b7b39aa36a06a689b4ea4b96a98bf/docs/figure/rosmap_overview.Rmd/cs-sizes-methyl-1.png" target="_blank">dade7d2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/618cae4fa9aadcbe31f55487e1b478d485770fba/docs/figure/rosmap_overview.Rmd/cs-sizes-methyl-1.png" target="_blank">618cae4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c84e5c046a56501c872e946ab6839eac67d0a41a/docs/figure/rosmap_overview.Rmd/cs-sizes-methyl-1.png" target="_blank">c84e5c0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here are the exact numbers:</p>
<pre class="r"><code>table(cs_size_susie)
table(cs_size_fsusie)
# cs_size_susie
#   1   2   5  10  20 Inf 
# 589 250 431 366 389 762 
# cs_size_fsusie
#    1    2    5   10   20  Inf 
# 8719 2074 1922  887  502  335</code></pre>
</div>
<div id="distance-to-tss-plot" class="section level3">
<h3>Distance-to-TSS plot</h3>
<p>We expect that most of the causal SNPs to be very close to the
nearest TSS. Let’s check this. This will involve a couple of
intermediate calculations.</p>
<p>This function adds a column to the SNP results containing the minimum
distance to the nearest TSS.</p>
<pre class="r"><code>add_min_dist_to_tss &lt;- function (dat, genes) {
  n &lt;- nrow(dat)
  dat$min_dist_to_tss &lt;- rep(Inf,n)
  n &lt;- nrow(genes)
  for (i in 1:n) {
    rows &lt;- which(as.character(genes[i,&quot;chromosome&quot;]) == as.character(dat$chr))
    if (length(rows) &gt; 0) {
      if (genes[i,&quot;strand&quot;] == &quot;+&quot;)
        d &lt;- genes[i,&quot;start&quot;] - dat[rows,&quot;pos&quot;]
      else
        d &lt;- dat[rows,&quot;pos&quot;] - genes[i,&quot;end&quot;]
      i &lt;- which(abs(d) &lt; abs(dat[rows,&quot;min_dist_to_tss&quot;]))
      if (length(i) &gt; 0) {
        d    &lt;- d[i]
        rows &lt;- rows[i]
        dat[rows,&quot;min_dist_to_tss&quot;] &lt;- d
      }
    }
  }
  return(dat)
}</code></pre>
<p>This function is used to extract the top SNP per location (e.g., CpG)
from the association tests.</p>
<pre class="r"><code>get_top_snp_per_location &lt;- function (dat) {
  x &lt;- factor(dat$molecular_trait_id)
  qval &lt;- dat$qvalue
  names(qval) &lt;- with(dat,paste(chr,pos,sep = &quot;_&quot;))
  res &lt;- tapply(qval,x,function (x) names(which.min(x)))
  res &lt;- strsplit(res,&quot;_&quot;,fixed = TRUE)
  out &lt;- data.frame(chr = factor(sapply(res,&quot;[[&quot;,1)),
                    pos = as.numeric(sapply(res,&quot;[[&quot;,2)))
  out &lt;- transform(out,chr = factor(chr))
  rownames(out) &lt;- levels(x)
  return(out)
}</code></pre>
<p>With these functions, we can compile the data needed for this
plot.</p>
<pre class="r"><code>methyl_snps_assoc        &lt;- get_top_snp_per_location(methyl_cpg_assoc)
methyl_snps_assoc        &lt;- add_min_dist_to_tss(methyl_snps_assoc,genes)
methyl_snps_susie_nodup  &lt;- add_min_dist_to_tss(methyl_snps_susie_nodup,genes)
methyl_snps_fsusie_nodup &lt;- add_min_dist_to_tss(methyl_snps_fsusie_nodup,genes)</code></pre>
<p>For SuSiE-topPC and fSuSiE, we calculate “weighted” counts of SNPs,
in which the weights are given by the PIPs.</p>
<pre class="r"><code>bin_size &lt;- 10000
bins &lt;- c(-Inf,seq(-2e5,2e5,bin_size),Inf)
bins &lt;- bins + bin_size/2
counts_assoc &lt;- as.numeric(table(cut(methyl_snps_assoc$min_dist_to_tss,bins)))
counts_susie &lt;- tapply(methyl_snps_susie_nodup$pip,
                       cut(methyl_snps_susie_nodup$min_dist_to_tss,bins),
                       function (x) sum(x,na.rm = TRUE))
counts_fsusie &lt;- tapply(methyl_snps_fsusie_nodup$pip,
                        cut(methyl_snps_fsusie_nodup$min_dist_to_tss,bins),
                        function (x) sum(x,na.rm = TRUE))</code></pre>
<p>Now we can plot the result:</p>
<pre class="r"><code>n &lt;- length(bins)
i &lt;- seq(2,n-2)
bin_centers   &lt;- bins[i] + bin_size/2
counts_assoc  &lt;- counts_assoc[i]
counts_susie  &lt;- counts_susie[i]
counts_fsusie &lt;- counts_fsusie[i]
counts_assoc  &lt;- counts_assoc/sum(counts_assoc)
counts_susie  &lt;- counts_susie/sum(counts_susie)
counts_fsusie &lt;- counts_fsusie/sum(counts_fsusie)
pdat &lt;- data.frame(method = rep(c(&quot;assoc&quot;,&quot;susie&quot;,&quot;fsusie&quot;),
                                each = length(bin_centers)),
                   dist   = rep(bin_centers/1000,times = 3),
                   freq   = c(counts_assoc,counts_susie,counts_fsusie),
                   stringsAsFactors = TRUE)
p &lt;- ggplot(pdat,aes(x = dist,y = freq,color = method)) +
  geom_line(linewidth = 0.5) +
  geom_point(size = 1) +
  scale_x_continuous(breaks = seq(-200,200,50)) +
  scale_y_continuous(breaks = seq(0,1,0.05)) +
  scale_color_manual(values = c(&quot;darkblue&quot;,&quot;darkorange&quot;,&quot;dodgerblue&quot;)) +
  labs(x = &quot;distance to TSS (kb)&quot;,y = &quot;proportion of SNPs&quot;) +
  theme_cowplot(font_size = 10)
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/plot-weighted-distance-to-tss-methyl-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-weighted-distance-to-tss-methyl-1">
Past versions of plot-weighted-distance-to-tss-methyl-1.png
</button>
</p>
<div id="fig-plot-weighted-distance-to-tss-methyl-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/plot-weighted-distance-to-tss-methyl-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="affected-cpgs" class="section level3">
<h3>Affected CpGs</h3>
<p>Now let’s turn to the recovery of affected CpGs.</p>
<p>Counting the number of CpGs per TAD is quite simple from the way the
fSuSiE results were compiled. It involves counting the number of unique
CpGs in each TAD:</p>
<pre class="r"><code>cpgs_per_tad_fsusie &lt;-
  with(methyl_cpg_fsusie,tapply(ID,region,function (x) length(unique(x))))</code></pre>
<p>Counting the number of CpGs per TAD from the SNP-CpG association
tests is a little more complicated because the CpGs were not assigned to
TADs in the results.</p>
<pre class="r"><code># Extract the information about the TADs from the TAD labels.
get_tad_info &lt;- function (tads) {
  res &lt;- strsplit(tads,&quot;_&quot;)
  return(data.frame(tad   = tads,
                    chr   = factor(sapply(res,&quot;[[&quot;,1)),
                    start = as.numeric(sapply(res,&quot;[[&quot;,2)),
                    end   = as.numeric(sapply(res,&quot;[[&quot;,3))))
}

# Count the number of CpGs in each TAD.
count_features_per_tad &lt;- function (features, tads) {
  tads$chr &lt;- as.character(tads$chr)
  features$chr &lt;- as.character(features$chr)
  n &lt;- nrow(tads)
  out &lt;- rep(0,n)
  names(out) &lt;- tads$tad
  for (i in 1:n) {
    rows &lt;- which(features$chr == tads[i,&quot;chr&quot;] &amp;
                  features$pos &gt;= tads[i,&quot;start&quot;] &amp;
                  features$pos &lt;= tads[i,&quot;end&quot;])
    out[i] &lt;- length(unique(features[rows,&quot;molecular_trait_id&quot;]))

  }
  return(out)
}

tads &lt;- get_tad_info(levels(methyl_cpg_fsusie$region))
cpgs_per_tad_assoc &lt;- count_features_per_tad(methyl_cpg_assoc,tads)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>These plots summarize the number of affected CpGs per TAD:</p>
<pre class="r"><code>cpgs_per_tad_assoc[cpgs_per_tad_assoc == 0] &lt;- NA
cpgs_per_tad_fsusie[cpgs_per_tad_fsusie == 0] &lt;- NA
pdat1 &lt;- data.frame(x = cpgs_per_tad_assoc)
pdat2 &lt;- data.frame(x = cpgs_per_tad_fsusie)
p1 &lt;- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 64) +
  xlim(0,2000) +
  labs(x = &quot;number of CpGs&quot;,y = &quot;number of TADs&quot;,
       title = &quot;SNP-CpG association tests&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 64) +
  xlim(0,2000) +
  labs(x = &quot;number of CpGs&quot;,y = &quot;number of TADs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/plot-cpgs-per-tad-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-cpgs-per-tad-1">
Past versions of plot-cpgs-per-tad-1.png
</button>
</p>
<div id="fig-plot-cpgs-per-tad-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/plot-cpgs-per-tad-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>A small number of TADs have more than 2,000 affected CpGs:</p>
<pre class="r"><code>sum(cpgs_per_tad_assoc &gt; 2000,na.rm = TRUE)
sum(cpgs_per_tad_fsusie &gt; 2000,na.rm = TRUE)
# [1] 15
# [1] 6</code></pre>
<p>This plot compares the number of affected CpGs identified by fSuSiE
and the SNP-CpG association tests:</p>
<pre class="r"><code>pdat &lt;- data.frame(assoc  = cpgs_per_tad_assoc,
                   fsusie = cpgs_per_tad_fsusie)
p &lt;- ggplot(pdat,aes(x = assoc,y = fsusie)) +
  geom_point(color = &quot;darkblue&quot;) +
  geom_abline(intercept = 0,slope = 1,color = &quot;magenta&quot;,linetype = &quot;dotted&quot;) +
  labs(x = &quot;SNP-CpG association tests&quot;,y = &quot;fSusiE&quot;) +
  xlim(0,4100) + 
  ylim(0,4100) + 
  theme_cowplot(font_size = 10)
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/compare-cpgs-per-tad-1.png" width="300" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare-cpgs-per-tad-1">
Past versions of compare-cpgs-per-tad-1.png
</button>
</p>
<div id="fig-compare-cpgs-per-tad-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/compare-cpgs-per-tad-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>One advantage of fSuSiE is that it is able to tell us which molecular
features are affected by which SNPs, and therefore it provides a more
coherent summary of how the SNPs affect methylation levels at a locus.
To examine this quantitatively, we compare the number of affected CpGs
per CS for fSuSiE to the number of affected CpGs per SNP from the
association tests. The result is much fewer CSs and many more affected
CpGs per CS:</p>
<pre class="r"><code>x &lt;- factor(methyl_cpg_assoc$variant_id)
cpgs_per_snp_assoc &lt;- tapply(methyl_cpg_assoc$molecular_trait_id,x,
                             function (x) length(unique(x)))
rm(x)
nodup_cs &lt;- levels(methyl_snps_fsusie_nodup$cs)
methyl_cpg_fsusie_nodup &lt;- subset(methyl_cpg_fsusie,is.element(cs,nodup_cs))
methyl_cpg_fsusie_nodup &lt;- transform(methyl_cpg_fsusie_nodup,cs = factor(cs))
cpgs_per_snp_fsusie &lt;-
  with(methyl_cpg_fsusie_nodup,
       tapply(ID,cs,function (x) length(unique(x))))
pdat1 &lt;- data.frame(x = cpgs_per_snp_assoc)
pdat2 &lt;- data.frame(x = cpgs_per_snp_fsusie)
pdat1 &lt;- subset(pdat1,x &lt;= 25)
pdat2 &lt;- subset(pdat2,x &lt;= 150)
p1 &lt;- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 25) +
  labs(x = &quot;number of CpGs&quot;,y = &quot;number of SNPs&quot;,
       title = &quot;SNP-CpG association tests&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 25) +
  labs(x = &quot;number of CpGs&quot;,y = &quot;number of CSs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/count-cpgs-per-snp-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-count-cpgs-per-snp-1">
Past versions of count-cpgs-per-snp-1.png
</button>
</p>
<div id="fig-count-cpgs-per-snp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/dade7d2d7b0b7b39aa36a06a689b4ea4b96a98bf/docs/figure/rosmap_overview.Rmd/count-cpgs-per-snp-1.png" target="_blank">dade7d2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/618cae4fa9aadcbe31f55487e1b478d485770fba/docs/figure/rosmap_overview.Rmd/count-cpgs-per-snp-1.png" target="_blank">618cae4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/count-cpgs-per-snp-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>A small proportion of the SNPs and CSs are not plotted because they
have an unusually large number of CpGs:</p>
<pre class="r"><code>mean(cpgs_per_snp_assoc &gt; 20)
mean(cpgs_per_snp_fsusie &gt; 150)
# [1] 0.02155
# [1] 0.01679</code></pre>
</div>
<div id="properties-of-msnps-in-1-snp-css" class="section level3">
<h3>Properties of mSNPs in 1-SNP CSs</h3>
<p>Now let’s look closely at the 1-SNP CSs for methylation.</p>
<pre class="r"><code>methyl_cs1snp_susie  &lt;- table(methyl_snps_susie_nodup$cs)
methyl_cs1snp_susie  &lt;- names(methyl_cs1snp_susie)[methyl_cs1snp_susie == 1]
methyl_cs1snp_susie  &lt;- subset(methyl_snps_susie,
                               is.element(cs,methyl_cs1snp_susie))
methyl_cs1snp_fsusie &lt;- table(methyl_snps_fsusie_nodup$cs)
methyl_cs1snp_fsusie &lt;- names(methyl_cs1snp_fsusie)[methyl_cs1snp_fsusie == 1]
methyl_cs1snp_fsusie &lt;- subset(methyl_snps_fsusie,
                              is.element(cs,methyl_cs1snp_fsusie))</code></pre>
<p>Compare the MAFs of the 1-SNP CSs to the MAFs across SNPs in CSs:</p>
<pre class="r"><code>p1 &lt;- ggplot(methyl_cs1snp_susie,aes(x = log10(maf))) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 36) +
  labs(x = &quot;log10(MAF)&quot;,y = &quot;number of SNPs&quot;,
       title = &quot;SuSiE-topPC, 1-SNP CSs&quot;) +
  theme_cowplot(font_size = 10)
p2 &lt;- ggplot(methyl_cs1snp_fsusie,aes(x = log10(maf))) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 36) +
  labs(x = &quot;log10(MAF)&quot;,y = &quot;number of SNPs&quot;,
       title = &quot;fSuSiE, 1-SNP CSs&quot;) +
  theme_cowplot(font_size = 10)
p3 &lt;- ggplot(methyl_snps_susie_nodup,aes(x = log10(maf))) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 36) +
  labs(x = &quot;log10(MAF)&quot;,y = &quot;number of SNPs&quot;,
       title = &quot;SuSiE-topPC, all CSs&quot;) +
  theme_cowplot(font_size = 10)
p4 &lt;- ggplot(methyl_snps_fsusie_nodup,aes(x = log10(maf))) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 36) +
  labs(x = &quot;log10(MAF)&quot;,y = &quot;number of SNPs&quot;,
       title = &quot;fSuSiE, all CSs&quot;) +
  theme_cowplot(font_size = 10)
plot_grid(p1,p2,p3,p4,nrow = 2,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/one-snp-cs-mafs-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>What is most striking is the excess of rarer SNPs in the 1-SNP CSs,
particularly in the 1-SNP CSs identified by SuSiE-topPC.</p>
<p>ADD TEXT HERE.</p>
<pre class="r"><code>methyl_cpg_fsusie_cs1snp &lt;- subset(methyl_cpg_fsusie,
                                   is.element(cs,methyl_cs1snp_fsusie$cs))
rows &lt;- match(methyl_cpg_fsusie_cs1snp$cs,methyl_cs1snp_fsusie$cs)
methyl_cpg_fsusie_cs1snp$variant_pos &lt;- methyl_cs1snp_fsusie[rows,&quot;pos&quot;]
methyl_cpg_fsusie_cs1snp &lt;- transform(methyl_cpg_fsusie_cs1snp,
                                      cs = factor(cs),
                                      dist = peak_start - variant_pos)
pdat &lt;- tapply(methyl_cpg_fsusie_cs1snp$dist,methyl_cpg_fsusie_cs1snp$cs,
               function (x) {
                 i &lt;- which.min(abs(x))
                 return(x[i])
               })
pdat &lt;- data.frame(dist = pdat)
pdat &lt;- transform(pdat,
  dist = cut(dist,c(-Inf,-1e6,-1e5,-1e4,-1e3,-100,
                    100,1e3,1e4,1e5,1e6,+Inf)))
p &lt;- ggplot(pdat,aes(x = dist)) +
  geom_bar(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 64,width = 0.5) +
  labs(x = &quot;CpG - mSNP distance (Kb)&quot;, y = &quot;number of mSNPs&quot;) +
  theme_cowplot(font_size = 10) +
  theme(axis.text.x = element_text(angle = 45,hjust = 1)) 
# Warning in geom_bar(color = &quot;white&quot;, fill = &quot;darkblue&quot;, bins = 64, width =
# 0.5): Ignoring unknown parameters: `bins`
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/dist-msnp-cpg-cs1snp-1.png" width="300" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="h3k27ac-snps" class="section level2">
<h2>H3K27ac SNPs</h2>
<p>Load the H3K27ac SNP results generated by SuSiE-topPC, fSuSiE and the
SNP-peak association testing:</p>
<pre class="r"><code>ha_peak_assoc_file &lt;- &quot;../outputs/ROSMAP_haQTL_qtl_snp_qval0.05.tsv.gz&quot;
ha_snps_susie_file &lt;- &quot;../outputs/ROSMAP_haQTL_cs_snp_toppc1_annotation.tsv.gz&quot;
ha_snps_fsusie_file &lt;- &quot;../outputs/ROSMAP_haQTL_cs_snp_annotation.tsv.gz&quot;
ha_peak_assoc  &lt;- read_enrichment_results(ha_peak_assoc_file,n = 8)
ha_snps_susie  &lt;- read_enrichment_results(ha_snps_susie_file,n = 6)
ha_snps_fsusie &lt;- read_enrichment_results(ha_snps_fsusie_file,n = 7)
ha_snps_susie$region &lt;-
  sapply(strsplit(ha_snps_susie$cs,&quot;:&quot;,fixed = TRUE),&quot;[[&quot;,2)
ha_snps_susie  &lt;- transform(ha_snps_susie,
                            region = factor(region),
                            cs     = factor(cs),
                            pc     = factor(pc))
ha_snps_fsusie &lt;- transform(ha_snps_fsusie,
                            cs     = factor(cs),
                            region = factor(region),
                            study  = factor(study))</code></pre>
<p>There is no need to look at the TAD sizes because the same TADs were
analyzed for both molecular traits, methylation and H3K27ac.</p>
<p>These histograms summarize the number of CSs per TAD:</p>
<pre class="r"><code>pdat1 &lt;- get_cs_vs_tad_size(ha_snps_susie)
pdat2 &lt;- get_cs_vs_tad_size(ha_snps_fsusie)
pdat1 &lt;- transform(pdat1,num_cs = factor(num_cs,1:20))
pdat2 &lt;- transform(pdat2,num_cs = factor(num_cs,1:20))
p1 &lt;- ggplot(pdat1,aes(x = num_cs)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;tomato&quot;) +
  scale_x_discrete(drop = FALSE) +
  labs(x = &quot;number of CSs&quot;,y = &quot;number of TADs&quot;,title = &quot;SuSiE-topPC&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(pdat2,aes(x = num_cs)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;tomato&quot;) +
  scale_x_discrete(drop = FALSE) +
  labs(x = &quot;number of CSs&quot;,y = &quot;number of TADs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/num-cs-per-tad-ha-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-num-cs-per-tad-ha-1">
Past versions of num-cs-per-tad-ha-1.png
</button>
</p>
<div id="fig-num-cs-per-tad-ha-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/71954f15429162c72e3c8013edbb72accecc71a9/docs/figure/rosmap_overview.Rmd/num-cs-per-tad-ha-1.png" target="_blank">71954f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Compare discovery of causal SNPs (number of CSs) in the SuSiE-topPC
and fSuSiE analyses:</p>
<pre class="r"><code>dat1 &lt;- get_cs_vs_tad_size(ha_snps_susie)
dat2 &lt;- get_cs_vs_tad_size(ha_snps_fsusie)
dat1 &lt;- dat1[c(1,3)]
dat2 &lt;- dat2[c(1,3)]
names(dat1) &lt;- c(&quot;tad&quot;,&quot;num_cs_susie&quot;)
names(dat2) &lt;- c(&quot;tad&quot;,&quot;num_cs_fsusie&quot;)
dat &lt;- merge(dat1,dat2,all = TRUE)
rows &lt;- which(is.na(dat$num_cs_susie))
dat[rows,&quot;num_cs_susie&quot;] &lt;- 0
pdat &lt;- melt(with(dat,table(num_cs_susie,num_cs_fsusie)))
rows &lt;- which(pdat$value == 0)
pdat[rows,&quot;value&quot;] &lt;- NA
p &lt;- ggplot(pdat,aes(x = num_cs_susie,y = num_cs_fsusie,size = value)) +
  geom_point(color = &quot;white&quot;,fill = &quot;tomato&quot;,shape = 21) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_x_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_y_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_size(breaks = c(1,10,100)) +
  labs(x = &quot;SuSiE-topPC&quot;,y = &quot;fSuSiE&quot;,size = &quot;number of TADs&quot;) +
  theme_cowplot(font_size = 10)</code></pre>
<p>Flag and remove the “duplicate” CSs.</p>
<pre class="r"><code>root_cs_susie        &lt;- create_cs_maps(ha_snps_susie)$root
root_cs_fsusie       &lt;- create_cs_maps(ha_snps_fsusie)$root
ha_snps_susie_nodup  &lt;- subset(ha_snps_susie,is.element(cs,root_cs_susie))
ha_snps_fsusie_nodup &lt;- subset(ha_snps_fsusie,is.element(cs,root_cs_fsusie))</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Compare the sizes of the CSs in the SuSiE-topPC and fSuSiE
analyses:</p>
<pre class="r"><code>bins &lt;- c(0,1,2,5,10,20,Inf)
ha_snps_susie_nodup &lt;- transform(ha_snps_susie_nodup,cs = factor(cs))
ha_snps_fsusie_nodup &lt;- transform(ha_snps_fsusie_nodup,cs = factor(cs))
cs_size_susie  &lt;- as.numeric(table(ha_snps_susie_nodup$cs))
cs_size_fsusie &lt;- as.numeric(table(ha_snps_fsusie_nodup$cs))
cs_size_susie  &lt;- cut(cs_size_susie,bins)
cs_size_fsusie &lt;- cut(cs_size_fsusie,bins)
levels(cs_size_susie) &lt;- bins[-1]
levels(cs_size_fsusie) &lt;- bins[-1]
p1 &lt;- ggplot(data.frame(cs_size = cs_size_susie),aes(x = cs_size)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;tomato&quot;,
                 width = 0.65) +
  labs(x = &quot;CS size&quot;,y = &quot;number of CSs&quot;,title = &quot;SuSiE-topPC&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(data.frame(cs_size = cs_size_fsusie),aes(x = cs_size)) +
  geom_histogram(stat = &quot;count&quot;,color = &quot;white&quot;,fill = &quot;tomato&quot;,
                 width = 0.65) +
  labs(x = &quot;CS size&quot;,y = &quot;number of CSs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/cs-sizes-ha-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cs-sizes-ha-1">
Past versions of cs-sizes-ha-1.png
</button>
</p>
<div id="fig-cs-sizes-ha-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/dade7d2d7b0b7b39aa36a06a689b4ea4b96a98bf/docs/figure/rosmap_overview.Rmd/cs-sizes-ha-1.png" target="_blank">dade7d2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/71954f15429162c72e3c8013edbb72accecc71a9/docs/figure/rosmap_overview.Rmd/cs-sizes-ha-1.png" target="_blank">71954f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We expect that the vast majority of the causal SNPs will be very
close to the TSS. Let’s verify this empirically.</p>
<p>First, add a “min_dist_to_TSS” column to each set of results:</p>
<pre class="r"><code>ha_snps_assoc  &lt;- get_top_snp_per_location(ha_peak_assoc)
ha_snps_assoc  &lt;- add_min_dist_to_tss(ha_snps_assoc,genes)
ha_snps_susie  &lt;- add_min_dist_to_tss(ha_snps_susie,genes)
ha_snps_fsusie &lt;- add_min_dist_to_tss(ha_snps_fsusie,genes)</code></pre>
<p>This next code chunk computes the histogram for the TSS plot:</p>
<pre class="r"><code>bin_size &lt;- 5e4
bins &lt;- c(-Inf,seq(-5e5,5.5e5,bin_size),Inf)
bins &lt;- bins - bin_size/2
counts_assoc &lt;- as.numeric(table(cut(ha_snps_assoc$min_dist_to_tss,bins)))
counts_susie &lt;- tapply(ha_snps_susie$pip,
                       cut(ha_snps_susie$min_dist_to_tss,bins),
                       function (x) sum(x,na.rm = TRUE))
counts_fsusie &lt;- tapply(ha_snps_fsusie$pip,
                        cut(ha_snps_fsusie$min_dist_to_tss,bins),
                        function (x) sum(x,na.rm = TRUE))</code></pre>
<p>And now we can plot the result:</p>
<pre class="r"><code>n &lt;- length(bins)
i &lt;- seq(2,n-2)
bin_centers   &lt;- bins[i] + bin_size/2
counts_assoc  &lt;- counts_assoc[i]
counts_susie  &lt;- counts_susie[i]
counts_fsusie &lt;- counts_fsusie[i]
counts_assoc  &lt;- counts_assoc/sum(counts_assoc)
counts_susie  &lt;- counts_susie/sum(counts_susie)
counts_fsusie &lt;- counts_fsusie/sum(counts_fsusie)
pdat &lt;- data.frame(method = rep(c(&quot;assoc&quot;,&quot;susie&quot;,&quot;fsusie&quot;),
                                each = length(bin_centers)),
                   dist   = rep(bin_centers/1000,times = 3),
                   freq   = c(counts_assoc,counts_susie,counts_fsusie),
                   stringsAsFactors = TRUE)
p &lt;- ggplot(pdat,aes(x = dist,y = freq,color = method)) +
  geom_line(linewidth = 0.5) +
  geom_point(size = 1) +
  scale_x_continuous(breaks = seq(-500,500,100),limits = c(-500,500)) +
  scale_y_continuous(breaks = seq(0,1,0.1)) +
  scale_color_manual(values = c(&quot;darkblue&quot;,&quot;darkorange&quot;,&quot;dodgerblue&quot;)) +
  labs(x = &quot;distance to TSS (kb)&quot;,y = &quot;proportion of SNPs&quot;) +
  theme_cowplot(font_size = 10)
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/plot-weighted-distance-to-tss-ha-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-weighted-distance-to-tss-ha-1">
Past versions of plot-weighted-distance-to-tss-ha-1.png
</button>
</p>
<div id="fig-plot-weighted-distance-to-tss-ha-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/c07956874db10cbee93e0c44ce20bfc946db2004/docs/figure/rosmap_overview.Rmd/plot-weighted-distance-to-tss-ha-1.png" target="_blank">c079568</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/plot-weighted-distance-to-tss-ha-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="affected-h3k27ac-peaks" class="section level2">
<h2>Affected H3K27ac peaks</h2>
<p>Now let’s turn to the recovery of affected H3K27ac peaks. First load
the results generated by fSuSiE:</p>
<pre class="r"><code>ha_peak_fsusie_file &lt;-
  &quot;../outputs/ROSMAP_haQTL_cs_effect_ha_peak_annotation.tsv.gz&quot;
ha_peak_fsusie &lt;- read_enrichment_results(ha_peak_fsusie_file,n = 9)
ha_peak_fsusie$region &lt;-
  sapply(strsplit(ha_peak_fsusie$cs,&quot;:&quot;,fixed = TRUE),&quot;[[&quot;,2)
ha_peak_fsusie &lt;- transform(ha_peak_fsusie,
                            cs      = factor(cs), 
                            region  = factor(region),
                            context = factor(context))</code></pre>
<p>Count the number of affected peaks per TAD from the fSuSiE
results:</p>
<pre class="r"><code>ha_peaks_per_tad_fsusie &lt;-
  with(ha_peak_fsusie,tapply(ID,region,function (x) length(unique(x))))</code></pre>
<p>Next count the number of affected peaks per TAD from the SNP-peak
association tests, using the functions defined above:</p>
<pre class="r"><code>tads &lt;- get_tad_info(levels(ha_peak_fsusie$region))
ha_peaks_per_tad_assoc &lt;- count_features_per_tad(ha_peak_assoc,tads)</code></pre>
<p>These two plots summarize the number of affected peaks per TAD:</p>
<pre class="r"><code>ha_peaks_per_tad_assoc[ha_peaks_per_tad_assoc == 0] &lt;- NA
ha_peaks_per_tad_fsusie[ha_peaks_per_tad_fsusie == 0] &lt;- NA
pdat1 &lt;- data.frame(x = ha_peaks_per_tad_assoc)
pdat2 &lt;- data.frame(x = ha_peaks_per_tad_fsusie)
p1 &lt;- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;tomato&quot;,bins = 64) +
  xlim(0,200) +
  labs(x = &quot;number of H3K27ac peaks&quot;,y = &quot;number of TADs&quot;,
       title = &quot;SNP-peak association tests&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;tomato&quot;,bins = 64) +
  xlim(0,200) +
  labs(x = &quot;number of H3K27ac peaks&quot;,y = &quot;number of TADs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/plot-ha-peaks-per-tad-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-ha-peaks-per-tad-1">
Past versions of plot-ha-peaks-per-tad-1.png
</button>
</p>
<div id="fig-plot-ha-peaks-per-tad-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/plot-ha-peaks-per-tad-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>A very small number of TADs have more than 200 affected peaks:</p>
<pre class="r"><code>mean(ha_peaks_per_tad_assoc &gt; 200,na.rm = TRUE)
mean(ha_peaks_per_tad_fsusie &gt; 200,na.rm = TRUE)
# [1] 0.002463
# [1] 0.004926</code></pre>
<p>This plot compares the number of affected H3K27ac peaks identified by
fSuSiE and the SNP-peak association tests:</p>
<pre class="r"><code>pdat &lt;- data.frame(assoc  = ha_peaks_per_tad_assoc,
                   fsusie = ha_peaks_per_tad_fsusie)
p &lt;- ggplot(pdat,aes(x = assoc,y = fsusie)) +
  geom_point(color = &quot;tomato&quot;) +
  geom_abline(intercept = 0,slope = 1,color = &quot;magenta&quot;,linetype = &quot;dotted&quot;) +
  labs(x = &quot;SNP-peak association tests&quot;,y = &quot;fSusiE&quot;) +
  xlim(0,200) + 
  ylim(0,200) + 
  theme_cowplot(font_size = 10)
print(p)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/compare-ha-peaks-per-tad-1.png" width="300" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare-ha-peaks-per-tad-1">
Past versions of compare-ha-peaks-per-tad-1.png
</button>
</p>
<div id="fig-compare-ha-peaks-per-tad-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/dade7d2d7b0b7b39aa36a06a689b4ea4b96a98bf/docs/figure/rosmap_overview.Rmd/compare-ha-peaks-per-tad-1.png" target="_blank">dade7d2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/3bb4ba0dd104f7fc4d5447b6c8548273d2e777cd/docs/figure/rosmap_overview.Rmd/compare-ha-peaks-per-tad-1.png" target="_blank">3bb4ba0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/compare-ha-peaks-per-tad-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>These next two plots compare the number of affected peaks per CS for
fSuSiE to the number of affected peaks per SNP from the association
tests:</p>
<pre class="r"><code>x &lt;- factor(ha_peak_assoc$variant_id)
ha_peaks_per_snp_assoc &lt;- tapply(ha_peak_assoc$molecular_trait_id,x,
                                 function (x) length(unique(x)))
rm(x)
nodup_cs &lt;- levels(ha_snps_fsusie_nodup$cs)
ha_peak_fsusie_nodup &lt;- subset(ha_peak_fsusie,is.element(cs,nodup_cs))
ha_peak_fsusie_nodup &lt;- transform(ha_peak_fsusie_nodup,cs = factor(cs))
ha_peaks_per_snp_fsusie &lt;-
  with(ha_peak_fsusie_nodup,tapply(ID,cs,function (x) length(unique(x))))
pdat1 &lt;- data.frame(x = ha_peaks_per_snp_assoc)
pdat2 &lt;- data.frame(x = ha_peaks_per_snp_fsusie)
pdat1 &lt;- subset(pdat1,x &lt;= 10)
pdat2 &lt;- subset(pdat2,x &lt;= 40)
pdat1 &lt;- transform(pdat1,x = factor(x))
p1 &lt;- ggplot(pdat1,aes(x = x)) +
  geom_bar(color = &quot;white&quot;,fill = &quot;tomato&quot;) +
  labs(x = &quot;number of H3K27ac peaks&quot;,y = &quot;number of SNPs&quot;,
       title = &quot;SNP-peak association tests&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
p2 &lt;- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;tomato&quot;,bins = 25) +
  labs(x = &quot;number of H3K27ac peaks&quot;,y = &quot;number of CSs&quot;,title = &quot;fSuSiE&quot;) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = &quot;plain&quot;))
plot_grid(p1,p2,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/rosmap_overview.Rmd/count-ha-peaks-per-snp-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-count-ha-peaks-per-snp-1">
Past versions of count-ha-peaks-per-snp-1.png
</button>
</p>
<div id="fig-count-ha-peaks-per-snp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/dade7d2d7b0b7b39aa36a06a689b4ea4b96a98bf/docs/figure/rosmap_overview.Rmd/count-ha-peaks-per-snp-1.png" target="_blank">dade7d2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-04-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/aece910973f3984cf07ffbe6089a898f58910c41/docs/figure/rosmap_overview.Rmd/count-ha-peaks-per-snp-1.png" target="_blank">aece910</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>A small proportion of the SNPs and CSs were not plotted because they
had an unusually large number of affected peaks:</p>
<pre class="r"><code>mean(ha_peaks_per_snp_assoc &gt; 10)
mean(ha_peaks_per_snp_fsusie &gt; 40)
# [1] 0.009821
# [1] 0.007172</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS Sonoma 14.7.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] hexbin_1.28.5     cowplot_1.1.3     ggplot2_3.5.0     dplyr_1.1.4      
# [5] data.table_1.15.2 workflowr_1.7.1  
# 
# loaded via a namespace (and not attached):
#  [1] sass_0.4.9        utf8_1.2.4        generics_0.1.3    lattice_0.22-5   
#  [5] stringi_1.8.3     digest_0.6.34     magrittr_2.0.3    evaluate_1.0.3   
#  [9] grid_4.3.3        fastmap_1.1.1     plyr_1.8.9        R.oo_1.26.0      
# [13] rprojroot_2.0.4   jsonlite_1.8.8    R.utils_2.12.3    processx_3.8.3   
# [17] whisker_0.4.1     ps_1.7.6          promises_1.2.1    httr_1.4.7       
# [21] fansi_1.0.6       scales_1.3.0      textshaping_0.3.7 jquerylib_0.1.4  
# [25] cli_3.6.4         rlang_1.1.5       R.methodsS3_1.8.2 munsell_0.5.0    
# [29] withr_3.0.2       cachem_1.0.8      yaml_2.3.8        tools_4.3.3      
# [33] reshape2_1.4.4    colorspace_2.1-0  httpuv_1.6.14     vctrs_0.6.5      
# [37] R6_2.5.1          lifecycle_1.0.4   git2r_0.33.0      stringr_1.5.1    
# [41] fs_1.6.5          ragg_1.2.7        pkgconfig_2.0.3   callr_3.7.5      
# [45] pillar_1.9.0      bslib_0.6.1       later_1.3.2       gtable_0.3.4     
# [49] glue_1.8.0        Rcpp_1.0.12       systemfonts_1.0.6 highr_0.10       
# [53] xfun_0.42         tibble_3.2.1      tidyselect_1.2.1  rstudioapi_0.15.0
# [57] knitr_1.45        farver_2.1.1      htmltools_0.5.8.1 labeling_0.4.3   
# [61] rmarkdown_2.26    compiler_4.3.3    getPass_0.2-4</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
