<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="William Denault, Hao Sun, Peter Carbonetto, Gao Wang" />


<title>Analysis of some susie and fsusie results on the ROSMAP data</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fsusie experiments</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/fsusie-experiments">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of some susie and fsusie results
on the ROSMAP data</h1>
<h4 class="author">William Denault, Hao Sun, Peter Carbonetto, Gao
Wang</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>fsusie-experiments/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfsusieexperimentstreeb316ab60b1816cf8920c6911ac548488d84b5723targetblankb316ab6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/stephenslab/fsusie-experiments/tree/b316ab60b1816cf8920c6911ac548488d84b5723" target="_blank">b316ab6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfsusieexperimentstreeb316ab60b1816cf8920c6911ac548488d84b5723targetblankb316ab6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/stephenslab/fsusie-experiments/tree/b316ab60b1816cf8920c6911ac548488d84b5723" target="_blank">b316ab6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/rosmap.Rmd</code>) and HTML
(<code>docs/rosmap.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/b316ab60b1816cf8920c6911ac548488d84b5723/analysis/rosmap.Rmd" target="_blank">b316ab6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-26
</td>
<td>
workflowr::wflow_publish("rosmap.Rmd", verbose = TRUE, view = FALSE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/cc246fef257361cdd3620bbb70e07873d94e6871/analysis/rosmap.Rmd" target="_blank">cc246fe</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-26
</td>
<td>
workflowr::wflow_publish("rosmap.Rmd", verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/95e4ababe5f7f2435f4fb802a30f7a48360ae9b7/analysis/rosmap.Rmd" target="_blank">95e4aba</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-26
</td>
<td>
Fixed the distance-to-TSS plot in the rosmap analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/366e27c32af5d0e5a62c07f498a1dbcccbb49d97/analysis/rosmap.Rmd" target="_blank">366e27c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-26
</td>
<td>
Added a couple todos to rosmap.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/e691fcda388c92371aae2758a081aa938590cde0/analysis/rosmap.Rmd" target="_blank">e691fcd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-26
</td>
<td>
Added distance-to-TSS plot in the rosmap analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/4fb83bde12c06849332a9570a4d82e5270f836bc/analysis/rosmap.Rmd" target="_blank">4fb83bd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-25
</td>
<td>
Added step to rosmap.Rmd to get TSS for each ‘region’ in susie results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/f8799a1a498cdfa190fd550c9bda4667e9aafa70/analysis/rosmap.Rmd" target="_blank">f8799a1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-25
</td>
<td>
Added plot to rosmap.Rmd for CS sizes.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/4ac6873463fc2644d239347ad48358fcba8524b1/analysis/rosmap.Rmd" target="_blank">4ac6873</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-25
</td>
<td>
Added a couple simple plots to the rosmap analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/5ba241b922b5dd5cffa8cf234c63c4b04d030b84/analysis/rosmap.Rmd" target="_blank">5ba241b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-25
</td>
<td>
Wrote function get_gene_annotations used in the rosmap analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/b1b38f0c03cbd85c7d4bc871a422f910e2297575/analysis/rosmap.Rmd" target="_blank">b1b38f0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-25
</td>
<td>
Added steps to rosmap analysis to prepare the gene annotations into a
convenient data frame.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/fsusie-experiments/142928b3dd55c7404e0bcf09284b08b8f80cf2e0/docs/rosmap.html" target="_blank">142928b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-25
</td>
<td>
First build of rosmap analysis; added gene annotation files.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/fsusie-experiments/blob/a4e3d764cb97b405af5215c2752de7fb17f3bc12/analysis/rosmap.Rmd" target="_blank">a4e3d76</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-04-25
</td>
<td>
workflowr::wflow_publish("analysis/rosmap.Rmd", verbose = TRUE)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>To build the workflowr page, I run this:</p>
<pre><code>sinteractive -c 4 --mem=24G --time=20:00:00 -p mstephens
module load R/4.1.0-no-openblas
module load pandoc/3.0.1
R
&gt; .libPaths()[1]
# [1] &quot;/home/pcarbo/R_libs_4_10_no_openblas&quot;
&gt; workflowr::wflow_build(&quot;rosmap.Rmd&quot;,view = FALSE,verbose = TRUE)</code></pre>
<p>TO DO: GIVE OVERVIEW HERE.</p>
<p>Load the packages as well as some additional custom functions used in
the analysis below,</p>
<pre class="r"><code>library(data.table)
library(ggplot2)
library(cowplot)
source(&quot;../code/rosmap_functions.R&quot;)
setDTthreads(1)</code></pre>
<p>Load the susie fine-mapping results on the “Inh_mega_eQTL” RNA-seq
data.</p>
<pre class="r"><code>datadir &lt;- file.path(&quot;/project2/mstephens/fungen_xqtl/ftp_fgc_xqtl&quot;,
                     &quot;analysis_result/finemapping_twas/prepared_results&quot;)
load(file.path(datadir,&quot;susie_Inh_mega_eQTL.RData&quot;))                     
susie &lt;- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)</code></pre>
<p>Load the fsusie fine-mapping results on the DLPFC methylation
data.</p>
<pre class="r"><code>load(file.path(datadir,&quot;fsusie_ROSMAP_DLPFC_mQTL.RData&quot;))
fsusie &lt;- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)</code></pre>
<p>Load the gene annotations. Specifically I extract here only the
annotated gene transcripts for protein-coding genes as defined in the
Ensembl/Havana database.</p>
<pre class="r"><code>gene_file &lt;-
  file.path(&quot;../data/genome_annotations&quot;,
    &quot;Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz&quot;)
genes &lt;- get_gene_annotations(gene_file)</code></pre>
<div id="susie-fine-mapping-of-rna-seq" class="section level2">
<h2>SuSiE fine-mapping of RNA-seq</h2>
<p>Size of regions in base-pairs and SNPs:</p>
<pre class="r"><code>susie$regions$pos_min &lt;- sapply(susie$pips,function (x) min(x$pos))
susie$regions$pos_max &lt;- sapply(susie$pips,function (x) max(x$pos))
susie$regions &lt;- transform(susie$regions,size_bp = pos_max - pos_min)
p1 &lt;- ggplot(susie$regions,aes(size_bp/1e6)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 64) +
  scale_x_continuous(breaks = seq(0,50,5)) +
  labs(x = &quot;size of region (Mb)&quot;,
       y = &quot;number of regions&quot;) +
  theme_cowplot(font_size = 10)
p2 &lt;- ggplot(susie$regions,aes(num_snps)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 64) +
  scale_x_continuous(breaks = seq(0,1e5,1e4)) +
  labs(x = &quot;number of SNPs&quot;,
       y = &quot;number of regions&quot;) +
  theme_cowplot(font_size = 10)
plot_grid(p1,p2)</code></pre>
<p><img src="figure/rosmap.Rmd/susie-region-sizes-1.png" width="750" style="display: block; margin: auto;" /></p>
<p>Number of CSs per region:</p>
<pre class="r"><code>table(CSs = susie$regions$num_cs)
# CSs
#    0    1    2    3    4    5    6    7    8   12 
# 4283 1333  294   68   28    5    1    1    3    1</code></pre>
<p>CS sizes:</p>
<pre class="r"><code>n &lt;- nlevels(susie$cs$region)
cs_sizes &lt;- vector(&quot;list&quot;,n)
region_names &lt;- levels(susie$cs$region)
names(cs_sizes) &lt;- region_names
for (i in region_names)
  cs_sizes[[i]] &lt;- as.vector(table(factor(subset(susie$cs,region == i)$cs)))
cs_sizes &lt;- unlist(cs_sizes)
pdat &lt;- data.frame(cs_size = cs_sizes)
ggplot(pdat,aes(cs_size)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;darkblue&quot;,bins = 64) +
  scale_x_continuous(trans = &quot;log10&quot;,breaks = c(1:5,10,100,1000)) +
  labs(x = &quot;number of SNPs&quot;,
       y = &quot;number of CSs&quot;) +
  theme_cowplot(font_size = 10)</code></pre>
<p><img src="figure/rosmap.Rmd/susie-cs-sizes-1.png" width="360" style="display: block; margin: auto;" /></p>
<p>NEXT: Create a plot showing the distance between the susie SNP and
the gene’s TSS. First get the TSS for each gene/region. Note: this code
is based on function “gtf_to_tss_bed” <a
href="https://github.com/broadinstitute/pyqtl/blob/master/qtl/io.py">here</a>.</p>
<pre class="r"><code>region_names &lt;- susie$regions$region_name
rownames(susie$regions) &lt;- region_names
susie$regions$tss       &lt;- as.numeric(NA)
susie$regions$strand    &lt;- as.character(NA)
for (i in region_names) {
  j &lt;- which(genes$ensembl == i)
  if (length(j) == 1) {
    susie$regions[i,&quot;strand&quot;] &lt;- as.character(genes[j,&quot;strand&quot;])
    if (genes[j,&quot;strand&quot;] == &quot;+&quot;)
      susie$regions[i,&quot;tss&quot;] &lt;- genes[j,&quot;start&quot;]
    else
      susie$regions[i,&quot;tss&quot;] &lt;- genes[j,&quot;end&quot;]
  }
}
susie$regions &lt;- transform(susie$regions,strand = factor(strand))</code></pre>
<p>Now compute the distance to the TSS weighted by the PIPs:</p>
<pre class="r"><code>n      &lt;- length(susie$pips)
bins   &lt;- c(-Inf,seq(-1e6,1e6,5e4),Inf)
counts &lt;- rep(0,length(bins) - 1)
for (i in 1:n) {
  if (!is.na(susie$regions[i,&quot;tss&quot;])) {
    pips &lt;- susie$pips[[i]]
    dist_to_tss &lt;- susie$regions[i,&quot;tss&quot;] - pips$pos
    if (susie$regions[i,&quot;strand&quot;] == &quot;-&quot;) 
      dist_to_tss &lt;- -dist_to_tss
    dist_to_tss &lt;- cut(dist_to_tss,bins)
    res &lt;- tapply(pips$pip,dist_to_tss,sum)
    res[is.na(res)] &lt;- 0
    counts &lt;- counts + res
  }
}</code></pre>
<p>Plot the result:</p>
<pre class="r"><code>n      &lt;- length(counts)
bins   &lt;- bins[seq(2,n-1)]
counts &lt;- counts[seq(2,n-1)]
pdat &lt;- data.frame(pos = bins + 2.5e4,count = counts)
ggplot(pdat,aes(x = pos,y = count)) +
  geom_point(color = &quot;darkblue&quot;) +
  geom_line(color = &quot;darkblue&quot;) +
  scale_y_continuous(limits = c(0,2000)) +
  labs(x = &quot;distance from TSS&quot;,y = &quot;SNPs weighted by PIPs&quot;) +
  theme_cowplot(font_size = 10)</code></pre>
<p><img src="figure/rosmap.Rmd/plot-dist-to-tss-1.png" width="390" style="display: block; margin: auto;" /></p>
<p>TO DO NEXT:</p>
<ul>
<li><p>Load fsusie haQTL results.</p></li>
<li></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.1.0 (2021-05-18)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: CentOS Linux 7 (Core)
# 
# Matrix products: default
# BLAS:   /software/R-4.1.0-no-openblas-el7-x86_64/lib64/R/lib/libRblas.so
# LAPACK: /software/R-4.1.0-no-openblas-el7-x86_64/lib64/R/lib/libRlapack.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.1.1     ggplot2_3.3.5     data.table_1.14.0
# 
# loaded via a namespace (and not attached):
#  [1] tidyselect_1.1.1  xfun_0.24         bslib_0.4.2       purrr_0.3.4      
#  [5] colorspace_2.0-2  vctrs_0.3.8       generics_0.1.0    htmltools_0.5.5  
#  [9] yaml_2.2.1        utf8_1.2.1        rlang_1.1.1       R.oo_1.24.0      
# [13] jquerylib_0.1.4   later_1.2.0       pillar_1.6.1      glue_1.4.2       
# [17] withr_2.5.0       DBI_1.1.1         R.utils_2.10.1    lifecycle_1.0.3  
# [21] stringr_1.4.0     munsell_0.5.0     gtable_0.3.0      workflowr_1.7.1.1
# [25] R.methodsS3_1.8.1 evaluate_0.14     labeling_0.4.2    knitr_1.33       
# [29] fastmap_1.1.0     httpuv_1.6.1      fansi_0.5.0       highr_0.9        
# [33] Rcpp_1.0.6        promises_1.2.0.1  scales_1.1.1      cachem_1.0.5     
# [37] jsonlite_1.7.2    farver_2.1.0      fs_1.5.0          digest_0.6.27    
# [41] stringi_1.6.2     dplyr_1.0.7       rprojroot_2.0.2   grid_4.1.0       
# [45] cli_3.6.1         tools_4.1.0       magrittr_2.0.1    sass_0.4.0       
# [49] tibble_3.1.2      crayon_1.4.1      whisker_0.4       pkgconfig_2.0.3  
# [53] ellipsis_0.3.2    assertthat_0.2.1  rmarkdown_2.9     R6_2.5.0         
# [57] git2r_0.28.0      compiler_4.1.0</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
