---
title: Examine ROSMAP methylation fine-mapping results
author: William Denault, Hao Sun, Angjing Liu, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

**Note:** If you would like to run this analysis on your computer, you
will first need to download the fine-mapping outputs. They can be
downloaded from [here][box-download-link]. Once you have downloaded
the files, copy each file to the "data" or "outputs" subdirectory.

Load some packages and custom functions used in the code below:

```{r load-pkgs, message=FALSE}
library(data.table)
library(dplyr)
library(ggplot2)
library(cowplot)
source("../code/rosmap_functions_more.R")
```

Load the gene annotations used in some of the analyses below. 

```{r load-gene-annotations}
gene_file <-
  file.path("../data/genome_annotations",
    "Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz")
genes <- get_gene_annotations(gene_file)
```

Load the allele frequencies computed by PLINK:

```{r read-allele-freqs}
load("../data/afreq.RData")
```

Next I load methylation SNP results generated by SuSiE-topPC, fSuSiE
and the SNP-CpG association testing:

```{r read-methyl-snp-results, eval=FALSE}
methyl_cpg_assoc_file  <- "../outputs/ROSMAP_mQTL_qtl_snp_qval0.05.tsv.gz"
methyl_snps_susie_file <-
  "../outputs/ROSMAP_mQTL_cs_snp_toppc1_annotation.tsv.gz"
methyl_snps_fsusie_file <- "../outputs/ROSMAP_mQTL_cs_snp_annotation.tsv.gz"
methyl_snps_susie  <- read_enrichment_results(methyl_snps_susie_file,n = 6)
methyl_snps_fsusie <- read_enrichment_results(methyl_snps_fsusie_file,n = 7)
methyl_cpg_assoc   <- read_enrichment_results(methyl_cpg_assoc_file,n = 8)
methyl_snps_susie$region <-
  sapply(strsplit(methyl_snps_susie$cs,":",fixed = TRUE),"[[",2)
methyl_snps_susie  <- transform(methyl_snps_susie,
                                region = factor(region),
                                cs     = factor(cs),
			  				    pc     = factor(pc))
methyl_snps_fsusie <- transform(methyl_snps_fsusie,
                                cs     = factor(cs),
								region = factor(region),
								study  = factor(study))
```

Add the allele frequencies to the methylation fine-mapping results:

```{r add-allele-freqs-methyl, eval=FALSE}
ids  <- with(methyl_snps_susie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
methyl_snps_susie$maf <- afreq[rows,"maf"]
ids  <- with(methyl_snps_fsusie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
methyl_snps_fsusie$maf <- afreq[rows,"maf"]
```

This is the number of fine-mapping regions (TADs) that contained at
least one CS in each of the analyses:

```{r methyl-tad-counts, eval=FALSE}
nlevels(methyl_snps_susie$region)
nlevels(methyl_snps_fsusie$region)
```

This plot summarizes the sizes of the TADs that were analyzed by
SuSiE-topPC and fSuSiE:

```{r plot-methyl-tad-sizes, fig.height=2.75, fig.width=2.75, warning=FALSE, eval=FALSE}
plot_tad_sizes <- function (tads) {
  tad_size <- get_tad_sizes(tads)
  pdat <- data.frame(tad_size = tad_size)
  return(ggplot(pdat,aes(x = tad_size)) +
         geom_histogram(color = "white",fill = "black",bins = 48) +
         labs(x = "size (Mb)",y = "number of TADs") +
         theme_cowplot(font_size = 10))
}
tads <- levels(methyl_snps_fsusie$region)
p <- plot_tad_sizes(tads) +
  scale_x_continuous(limits = c(2,9),breaks = 1:10) +
  scale_y_continuous(breaks = seq(0,100,10))
print(p)
```

```{r save-plot-methyl-tad-sizes, echo=FALSE, warning=FALSE, eval=FALSE}
ggsave("tad_sizes_histogram.pdf",p,height = 3,width = 3)
```

Some other useful statistics on the TAD sizes:

```{r summarize-methyl-tad-sizes, eval=FALSE}
tad_size <- get_tad_sizes(tads)
length(tad_size)
range(tad_size)
mean(tad_size)
median(tad_size)
sum(tad_size > 9)
```

There are some technical concerns about fine-mapping mSNPs that are
also CpGs, so these SNPs (and the CSs that include these SNPs)
should be removed from the results.

For the SNP-CpG association tests, finding the "blacklisted" is 
quite straightforward because they are just the mSNPs that have the
same positions as the CpGs:

```{r blacklist-assoc, eval=FALSE}
rows <- which(with(methyl_cpg_assoc,pos == molecular_trait_id_pos))
length(rows)
ids  <- with(methyl_cpg_assoc[rows,],paste(chr,pos,sep = "_"))
i <- match(ids,afreq$id)
table(cut(afreq[i,"maf"],c(0,0.01,0.1,1)))
rows <- which(with(methyl_cpg_assoc,pos != molecular_trait_id_pos))
methyl_cpg_assoc <- methyl_cpg_assoc[rows,]
```

For fSuSiE, it is a bit more complicated because we need to first load
the CpG results:

```{r read-methyl-cpg-results, eval=FALSE}
methyl_cpg_fsusie_file <-
  "../outputs/ROSMAP_mQTL_cs_effect_cpg_annotation.tsv.gz"
methyl_cpg_fsusie <- read_enrichment_results(methyl_cpg_fsusie_file,n = 9)
methyl_cpg_fsusie$region <-
  sapply(strsplit(methyl_cpg_fsusie$cs,":",fixed = TRUE),"[[",2)
methyl_cpg_fsusie <- transform(methyl_cpg_fsusie,
                               cs      = factor(cs), 
 						       region  = factor(region),
						       context = factor(context))
```

Now identify the blacklisted mSNPs as the mSNPs that have the same
position as a CpG (I will  use both "peak start" and "peak end"):

```{r get-blacklisted-snps, eval=FALSE}
ids_snp <- c(with(methyl_snps_susie,paste(chr,pos,sep = "_")),
             with(methyl_snps_fsusie,paste(chr,pos,sep = "_")))
ids_cpg <- with(methyl_cpg_fsusie,
                c(paste(chr,peak_start,sep = "_"),
				  paste(chr,peak_end,sep = "_")))
blacklisted_ids <- intersect(ids_snp,ids_cpg)
length(blacklisted_ids)
```

Before removing the blacklisted SNPs from the results, let's
take a quick look at them:

```{r examine-blacklisted-snps, eval=FALSE}
ids <- with(methyl_snps_fsusie,paste(chr,pos,sep = "_"))
rows <- which(is.element(ids,blacklisted_ids))
methyl_snps_fsusie_blacklisted <- methyl_snps_fsusie[rows,]
nrow(methyl_snps_fsusie_blacklisted)
sum(methyl_snps_fsusie_blacklisted$pip > 0.5)
```

It is interesting that a large fraction of the blacklisted SNPs have a
high PIPs. Let's also look at the MAFs of these "blacklisted" SNPs:

```{r blacklisted-snps-maf-hist, fig.height=2.5, fig.width=2.5, eval=FALSE}
ggplot(methyl_snps_fsusie_blacklisted,aes(x = log10(maf))) +
  geom_histogram(fill = "darkblue",color = "white",bins = 32) +
  labs(x = "log10(MAF)",y = "number of SNPs") +
  theme_cowplot(font_size = 10)
```

Before moving forward with the analysis, remove all SuSiE and
fSuSiE CSs containing the blacklisted SNPs:

```{r remove-blacklisted-cs, eval=FALSE}
ids                <- with(methyl_snps_susie,paste(chr,pos,sep = "_"))
rows               <- which(is.element(ids,blacklisted_ids))
blacklisted_cs     <- unique(methyl_snps_susie[rows,]$cs)
methyl_snps_susie  <- subset(methyl_snps_susie,!is.element(cs,blacklisted_cs))
ids                <- with(methyl_snps_fsusie,paste(chr,pos,sep = "_"))
rows               <- which(is.element(ids,blacklisted_ids))
blacklisted_cs     <- unique(methyl_snps_fsusie[rows,]$cs)
methyl_snps_fsusie <- subset(methyl_snps_fsusie,!is.element(cs,blacklisted_cs))
methyl_snps_susie  <- transform(methyl_snps_susie,cs = factor(cs))
methyl_snps_fsusie <- transform(methyl_snps_fsusie,cs = factor(cs))
```

These histograms summarize the number of CSs per TAD:

```{r num-cs-per-tad-methyl, fig.height=2, fig.width=5, warning=FALSE, eval=FALSE}
pdat1 <- get_cs_vs_tad_size(methyl_snps_susie)
pdat2 <- get_cs_vs_tad_size(methyl_snps_fsusie)
pdat1 <- transform(pdat1,num_cs = factor(num_cs,1:20))
pdat2 <- transform(pdat2,num_cs = factor(num_cs,1:20))
p1 <- ggplot(pdat1,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

Compare discovery of causal SNPs (number of CSs) in the SuSiE-topPC
and fSuSiE analyses:

```{r compare-num-cs-per-tad-methyl, fig.height=2.75, fig.width=3.75, warning=FALSE, eval=FALSE}
dat1 <- get_cs_vs_tad_size(methyl_snps_susie)
dat2 <- get_cs_vs_tad_size(methyl_snps_fsusie)
dat1 <- dat1[c(1,3)]
dat2 <- dat2[c(1,3)]
names(dat1) <- c("tad","num_cs_susie")
names(dat2) <- c("tad","num_cs_fsusie")
dat <- merge(dat1,dat2,all = TRUE)
rows <- which(is.na(dat$num_cs_susie))
dat[rows,"num_cs_susie"] <- 0
pdat <- melt(with(dat,table(num_cs_susie,num_cs_fsusie)))
rows <- which(pdat$value == 0)
pdat[rows,"value"] <- NA
p <- ggplot(pdat,aes(x = num_cs_susie,y = num_cs_fsusie,size = value)) +
  geom_point(color = "white",fill = "darkblue",shape = 21) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_x_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_y_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_size(breaks = c(1,5,10,50,100)) +
  labs(x = "SuSiE-topPC",y = "fSuSiE",size = "number of TADs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r compare-num-cs-per-tad-methyl-ggsave, echo=FALSE, warning=FALSE, eval=FALSE}
ggsave("cs_per_tad_scatterplot_methyl.pdf",p,height = 3,width = 4)
```

Now let's use this function to flag and remove the "duplicate" CSs.

```{r remove-duplicated-cs-methyl, message=FALSE, cache=TRUE, eval=FALSE}
root_cs_susie  <- create_cs_maps(methyl_snps_susie)$root
root_cs_fsusie <- create_cs_maps(methyl_snps_fsusie)$root
methyl_snps_susie_nodup  <- subset(methyl_snps_susie,
                                   is.element(cs,root_cs_susie))
methyl_snps_fsusie_nodup <- subset(methyl_snps_fsusie,
                                   is.element(cs,root_cs_fsusie))
```

Compare the sizes of the CSs in the SuSiE-topPC and fSuSiE analyses:

```{r cs-sizes-methyl, fig.height=2, fig.width=4, warning=FALSE, eval=FALSE}
bins <- c(0,1,2,5,10,20,Inf)
methyl_snps_susie_nodup <- transform(methyl_snps_susie_nodup,cs = factor(cs))
methyl_snps_fsusie_nodup <- transform(methyl_snps_fsusie_nodup,cs = factor(cs))
cs_size_susie  <- as.numeric(table(methyl_snps_susie_nodup$cs))
cs_size_fsusie <- as.numeric(table(methyl_snps_fsusie_nodup$cs))
cs_size_susie  <- cut(cs_size_susie,bins)
cs_size_fsusie <- cut(cs_size_fsusie,bins)
levels(cs_size_susie) <- bins[-1]
levels(cs_size_fsusie) <- bins[-1]
p1 <- ggplot(data.frame(cs_size = cs_size_susie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue",
                 width = 0.65) +
  labs(x = "CS size",y = "number of CSs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(data.frame(cs_size = cs_size_fsusie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue",
                 width = 0.65) +
  scale_y_continuous(breaks = seq(0,1e4,2000)) +			 
  labs(x = "CS size",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

Here are the exact numbers:

```{r cs-sizes-table-methyl, eval=FALSE}
table(cs_size_susie)
table(cs_size_fsusie)
```

### Distance-to-TSS plot

We expect that most of the causal SNPs to be very close to the nearest
TSS. Let's check this. This will involve a couple of intermediate
calculations.

```{r cs-sizes-methyl-ggsave, echo=FALSE, warning=FALSE, eval=FALSE}
ggsave("cs_sizes_methyl.pdf",
       plot_grid(p1,p2,nrow = 1,ncol = 2),
	   height = 2,width = 4)
```

With these functions, we can compile the data needed for this plot.

```{r add-min-dist-to-tss-methyl, eval=FALSE}
methyl_snps_assoc        <- get_top_snp_per_location(methyl_cpg_assoc)
methyl_snps_assoc        <- add_min_dist_to_tss(methyl_snps_assoc,genes)
methyl_snps_susie_nodup  <- add_min_dist_to_tss(methyl_snps_susie_nodup,genes)
methyl_snps_fsusie_nodup <- add_min_dist_to_tss(methyl_snps_fsusie_nodup,genes)
```

For SuSiE-topPC and fSuSiE, we calculate "weighted" counts of SNPs,
in which the weights are given by the PIPs.

```{r compute-weighted-distance-to-tss-methyl, eval=FALSE}
bin_size <- 10000
bins <- c(-Inf,seq(-2e5,2e5,bin_size),Inf)
bins <- bins + bin_size/2
counts_assoc <- as.numeric(table(cut(methyl_snps_assoc$min_dist_to_tss,bins)))
counts_susie <- tapply(methyl_snps_susie_nodup$pip,
                       cut(methyl_snps_susie_nodup$min_dist_to_tss,bins),
					   function (x) sum(x,na.rm = TRUE))
counts_fsusie <- tapply(methyl_snps_fsusie_nodup$pip,
                        cut(methyl_snps_fsusie_nodup$min_dist_to_tss,bins),
						function (x) sum(x,na.rm = TRUE))
```

Now we can plot the result:

```{r plot-weighted-distance-to-tss-methyl, fig.height=2.5, fig.width=3.5, eval=FALSE}
n <- length(bins)
i <- seq(2,n-2)
bin_centers   <- bins[i] + bin_size/2
counts_assoc  <- counts_assoc[i]
counts_susie  <- counts_susie[i]
counts_fsusie <- counts_fsusie[i]
counts_assoc  <- counts_assoc/sum(counts_assoc)
counts_susie  <- counts_susie/sum(counts_susie)
counts_fsusie <- counts_fsusie/sum(counts_fsusie)
pdat <- data.frame(method = rep(c("assoc","susie","fsusie"),
                                each = length(bin_centers)),
                   dist   = rep(bin_centers/1000,times = 3),
                   freq   = c(counts_assoc,counts_susie,counts_fsusie),
				   stringsAsFactors = TRUE)
p <- ggplot(pdat,aes(x = dist,y = freq,color = method)) +
  geom_line(linewidth = 0.5) +
  geom_point(size = 1) +
  scale_x_continuous(breaks = seq(-200,200,50)) +
  scale_y_continuous(breaks = seq(0,1,0.05)) +
  scale_color_manual(values = c("darkblue","darkorange","dodgerblue")) +
  labs(x = "distance to TSS (kb)",y = "proportion of SNPs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r ggsave-weighted-distance-to-tss-methyl, echo=FALSE, warning=FALSE, eval=FALSE}
ggsave("dist_to_tss_methyl.pdf",p,height = 2.5,width = 3.5)
```

Now let's turn to the recovery of affected CpGs.

Counting the number of CpGs per TAD is quite simple from the way the
fSuSiE results were compiled. It involves counting the number of unique
CpGs in each TAD:

```{r get-cpgs-per-tad-fsusie, eval=FALSE}
cpgs_per_tad_fsusie <-
  with(methyl_cpg_fsusie,tapply(ID,region,function (x) length(unique(x))))
```

Counting the number of CpGs per TAD from the SNP-CpG association tests is a
little more complicated because the CpGs were not assigned to TADs in the
results.

```{r, eval=FALSE}
tads <- get_tad_info(levels(methyl_cpg_fsusie$region))
cpgs_per_tad_assoc <- count_features_per_tad(methyl_cpg_assoc,tads)
```

These plots summarize the number of affected CpGs per TAD:

```{r plot-cpgs-per-tad, fig.height=2, fig.width=5, warning=FALSE, eval=FALSE}
cpgs_per_tad_assoc[cpgs_per_tad_assoc == 0] <- NA
cpgs_per_tad_fsusie[cpgs_per_tad_fsusie == 0] <- NA
pdat1 <- data.frame(x = cpgs_per_tad_assoc)
pdat2 <- data.frame(x = cpgs_per_tad_fsusie)
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 64) +
  xlim(0,2000) +
  labs(x = "number of CpGs",y = "number of TADs",
       title = "SNP-CpG association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 64) +
  xlim(0,2000) +
  labs(x = "number of CpGs",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

A small number of TADs have more than 2,000 affected CpGs:

```{r cpgs-per-tad-more, eval=FALSE}
sum(cpgs_per_tad_assoc > 2000,na.rm = TRUE)
sum(cpgs_per_tad_fsusie > 2000,na.rm = TRUE)
```

This plot compares the number of affected CpGs identified by fSuSiE
and the SNP-CpG association tests:

```{r compare-cpgs-per-tad, fig.height=2.5, fig.width=2.5, warning=FALSE, eval=FALSE}
pdat <- data.frame(assoc  = cpgs_per_tad_assoc,
                   fsusie = cpgs_per_tad_fsusie)
p <- ggplot(pdat,aes(x = assoc,y = fsusie)) +
  geom_point(color = "darkblue") +
  geom_abline(intercept = 0,slope = 1,color = "magenta",linetype = "dotted") +
  labs(x = "SNP-CpG association tests",y = "fSusiE") +
  xlim(0,4100) + 
  ylim(0,4100) + 
  theme_cowplot(font_size = 10)
print(p)
```

```{r compare-cpgs-per-tad-ggsave, echo=FALSE, warning=FALSE, eval=FALSE}
ggsave("affected_cpgs_per_tad.pdf",p,height = 2.5,width = 2.5)
```

One advantage of fSuSiE is that it is able to tell us which molecular
features are affected by which SNPs, and therefore it provides a more
coherent summary of how the SNPs affect methylation levels at a locus.
To examine this quantitatively, we compare the number of affected CpGs
per CS for fSuSiE to the number of affected CpGs per SNP from the
association tests. The result is much fewer CSs and many more
affected CpGs per CS:

```{r count-cpgs-per-snp, fig.height=2, fig.width=4.5, eval=FALSE}
x <- factor(methyl_cpg_assoc$variant_id)
cpgs_per_snp_assoc <- tapply(methyl_cpg_assoc$molecular_trait_id,x,
                             function (x) length(unique(x)))
rm(x)
nodup_cs <- levels(methyl_snps_fsusie_nodup$cs)
methyl_cpg_fsusie_nodup <- subset(methyl_cpg_fsusie,is.element(cs,nodup_cs))
methyl_cpg_fsusie_nodup <- transform(methyl_cpg_fsusie_nodup,cs = factor(cs))
cpgs_per_snp_fsusie <-
  with(methyl_cpg_fsusie_nodup,
       tapply(ID,cs,function (x) length(unique(x))))
pdat1 <- data.frame(x = cpgs_per_snp_assoc)
pdat2 <- data.frame(x = cpgs_per_snp_fsusie)
pdat1 <- subset(pdat1,x <= 25)
pdat2 <- subset(pdat2,x <= 150)
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 25) +
  labs(x = "number of CpGs",y = "number of SNPs",
       title = "SNP-CpG association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 25) +
  labs(x = "number of CpGs",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

```{r count-cpgs-per-snp-ggave, echo=FALSE, warning=FALSE, eval=FALSE}
ggsave("affected_cpgs_per_snp.pdf",
       plot_grid(p1,p2,nrow = 1,ncol = 2),
       height = 2,width = 4.5)
```

A small proportion of the SNPs and CSs are not plotted because they
have an unusually large number of CpGs:

```{r count-cpgs-per-snp-more, eval=FALSE}
mean(cpgs_per_snp_assoc > 20)
mean(cpgs_per_snp_fsusie > 150)
```

Now let's look closely at the 1-SNP CSs for methylation.

```{r get-1-snp-cs, eval=FALSE}
methyl_cs1snp_susie  <- table(methyl_snps_susie_nodup$cs)
methyl_cs1snp_susie  <- names(methyl_cs1snp_susie)[methyl_cs1snp_susie == 1]
methyl_cs1snp_susie  <- subset(methyl_snps_susie,
                               is.element(cs,methyl_cs1snp_susie))
methyl_cs1snp_fsusie <- table(methyl_snps_fsusie_nodup$cs)
methyl_cs1snp_fsusie <- names(methyl_cs1snp_fsusie)[methyl_cs1snp_fsusie == 1]
methyl_cs1snp_fsusie <- subset(methyl_snps_fsusie,
                              is.element(cs,methyl_cs1snp_fsusie))
```

TO DO: Add plots showing distribution of PIPs for SNPs in 1-SNP CSs.

Finally, this plot shows the distribution of the distances between the
mSNP and the nearest CpG affected by that SNP:

```{r dist-msnp-cpg-cs1snp, fig.height=2.5, fig.width=2.5, warning=FALSE, eval=FALSE}
methyl_cpg_fsusie_cs1snp <- subset(methyl_cpg_fsusie,
                                   is.element(cs,methyl_cs1snp_fsusie$cs))
rows <- match(methyl_cpg_fsusie_cs1snp$cs,methyl_cs1snp_fsusie$cs)
methyl_cpg_fsusie_cs1snp$variant_pos <- methyl_cs1snp_fsusie[rows,"pos"]
methyl_cpg_fsusie_cs1snp <- transform(methyl_cpg_fsusie_cs1snp,
                                      cs = factor(cs),
                                      dist = peak_start - variant_pos)
pdat <- tapply(methyl_cpg_fsusie_cs1snp$dist,methyl_cpg_fsusie_cs1snp$cs,
               function (x) {
			     i <- which.min(abs(x))
				 return(x[i])
    		   })
pdat <- data.frame(dist = pdat)
pdat <- transform(pdat,
  dist = cut(dist,c(-Inf,-1e6,-1e5,-1e4,-1e3,-100,
                    100,1e3,1e4,1e5,1e6,+Inf)))
p <- ggplot(pdat,aes(x = dist)) +
  geom_bar(color = "white",fill = "darkblue",bins = 64,width = 0.5) +
  labs(x = "CpG - mSNP distance (base-pairs)",
       y = "number of mSNPs") +
  theme_cowplot(font_size = 10) +
  theme(axis.text.x = element_text(angle = 45,hjust = 1)) 
print(p)
```

As we can see, the nearest affected CpG is very likely to be nearby,
although there are some putative long-range effects (which may or
may not be real).

[box-download-link]: https://uchicago.box.com/s/tt1vgg7vqayfthg0vsbl8gw0sdi0f1uo
