---
title: "Analysis of some susie and fsusie results on the ROSMAP data"
author: William Denault, Hao Sun, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

TO DO: GIVE OVERVIEW HERE.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages as well as some additional custom functions used in
the analysis below,

```{r load-pkgs, message=FALSE}
library(data.table)
source("code/rosmap_functions.R")
setDTthreads(1)
```

Load the susie fine-mapping results on the "Inh_mega_eQTL" RNA-seq
data.

```{r load-susie-results}
datadir <- file.path("/project2/mstephens/fungen_xqtl/ftp_fgc_xqtl",
                     "analysis_result/finemapping_twas/prepared_results")
load(file.path(datadir,"susie_Inh_mega_eQTL.RData"))					 
susie <- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)
```

Load the fsusie fine-mapping results on the DLPFC methylation data.

```{r load-fsusie-results}
load(file.path(datadir,"fsusie_ROSMAP_DLPFC_mQTL.RData"))
fsusie <- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)
```

Load the gene annotations. Specifically I extract here only the
annotated gene transcripts for protein-coding genes as defined in the
Ensembl/Havana database.

```{r load-gene-annotations}
gene_file <-
  file.path("data/genome_annotations",
    "Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz")
genes <- get_gene_annotations(gene_file)
```

