---
title: "Examine susie and fsusie results on the ROSMAP data"
author: William Denault, Hao Sun, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

Here we take a close look at a selection of the susie and fsusie
fine-mapping results for the ROSMAP data.

To build the workflowr page, I ran these commands on midway2:

```
sinteractive -c 4 --mem=24G --time=20:00:00 -p mstephens
module load R/4.1.0-no-openblas
module load pandoc/3.0.1
R
> .libPaths()[1]
# [1] "/home/pcarbo/R_libs_4_10_no_openblas"
> getwd()
# [1] "../analysis"
> workflowr::wflow_build("rosmap.Rmd",local=TRUE,view=FALSE,verbose=TRUE)
```

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages as well as some additional custom functions used in
the analysis below.

```{r load-pkgs, message=FALSE}
library(data.table)
library(ggplot2)
library(cowplot)
source("../code/rosmap_functions.R")
setDTthreads(1)
```

Load the susie fine-mapping results on the "Inh mega eQTL" RNA-seq
data.

```{r load-susie-results}
datadir <- file.path("/project2/mstephens/fungen_xqtl/ftp_fgc_xqtl",
                     "analysis_result/finemapping_twas/prepared_results")
load(file.path(datadir,"susie_Inh_mega_eQTL.RData"))					 
susie <- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)
```

Load the fsusie fine-mapping results on the DLPFC methylation data.

```{r load-fsusie-mqtl-results}
load(file.path(datadir,"fsusie_ROSMAP_DLPFC_mQTL.RData"))
fsusie_mqtl <- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)
```

Load the fsusie fine-mappign results on the DLPFC histone acetylation
(H3K9ac ChIP-seq) data.


```{r load-fsusie-haqtl-results}
load(file.path(datadir,"fsusie_ROSMAP_DLPFC_haQTL.RData"))
fsusie_haqtl <- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)
```

Load the gene annotations. Specifically I extract here only the
annotated gene transcripts for protein-coding genes as defined in the
Ensembl/Havana database.

```{r load-gene-annotations}
gene_file <-
  file.path("../data/genome_annotations",
    "Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz")
genes <- get_gene_annotations(gene_file)
```

SuSiE fine-mapping of RNA-seq
-----------------------------

Region sizes in base-pairs and SNPs:

```{r susie-region-sizes, fig.height=3, fig.width=6.25}
p1 <- region_sizes_histogram(susie$regions,susie$pips)
p2 <- num_snps_histogram(susie$regions)
plot_grid(p1,p2)
```

Number of CSs per region:

```{r susie-num-cs}
table(CSs = susie$regions$num_cs)
```

CS sizes:

```{r susie-cs-sizes, fig.height=3, fig.width=3}
susie_cs_sizes <- get_cs_sizes_by_region(susie$cs)
cs_sizes_histogram(susie_cs_sizes,x_breaks = c(1:5,10,100,1000))
```

NEXT: Create a plot showing the distance between the susie SNP and the
gene's TSS. First get the TSS for each gene/region. Note: this code is
based on function "gtf_to_tss_bed"
[here](https://github.com/broadinstitute/pyqtl/blob/master/qtl/io.py).

```{r get-gene-tss}
region_names <- susie$regions$region_name
rownames(susie$regions) <- region_names
susie$regions$tss       <- as.numeric(NA)
susie$regions$strand    <- as.character(NA)
for (i in region_names) {
  j <- which(genes$ensembl == i)
  if (length(j) == 1) {
    susie$regions[i,"strand"] <- as.character(genes[j,"strand"])
    if (genes[j,"strand"] == "+")
      susie$regions[i,"tss"] <- genes[j,"start"]
    else
	  susie$regions[i,"tss"] <- genes[j,"end"]
  }
}
susie$regions <- transform(susie$regions,strand = factor(strand))
```

Now compute the distance to the TSS weighted by the PIPs:

```{r dist-to-tss}
n      <- length(susie$pips)
bins   <- c(-Inf,seq(-1e6,1e6,5e4),Inf)
counts <- rep(0,length(bins) - 1)
for (i in 1:n) {
  if (!is.na(susie$regions[i,"tss"])) {
    pips <- susie$pips[[i]]
    dist_to_tss <- susie$regions[i,"tss"] - pips$pos
    if (susie$regions[i,"strand"] == "-") 
	  dist_to_tss <- -dist_to_tss
    dist_to_tss <- cut(dist_to_tss,bins)
    res <- tapply(pips$pip,dist_to_tss,sum)
    res[is.na(res)] <- 0
    counts <- counts + res
  }
}
```

Plot the result:

```{r plot-dist-to-tss, fig.height=3, fig.width=3.25}
n      <- length(counts)
bins   <- bins[seq(2,n-1)]
counts <- counts[seq(2,n-1)]
pdat <- data.frame(pos = bins + 2.5e4,count = counts)
ggplot(pdat,aes(x = pos,y = count)) +
  geom_point(color = "darkblue") +
  geom_line(color = "darkblue") +
  scale_y_continuous(limits = c(0,2000)) +
  labs(x = "distance from TSS",y = "SNPs weighted by PIPs") +
  theme_cowplot(font_size = 10)
```


fSuSiE fine-mapping of haQTLs and overlap with expression SNPs
--------------------------------------------------------------

Region sizes in base-pairs and SNPs:

```{r fsusie-haqtl-region-sizes, fig.height=3, fig.width=6.25}
p1 <- region_sizes_histogram(fsusie_haqtl$regions,fsusie_haqtl$pips)
p2 <- num_snps_histogram(fsusie_haqtl$regions)
plot_grid(p1,p2)
```

Number of CSs per region:

```{r fsusie-haqtl-num-cs}
table(CSs = fsusie_haqtl$regions$num_cs)
```

CS sizes:

```{r fsusie-haqtl-cs-sizes, fig.height=3, fig.width=3, warning=FALSE}
fsusie_haqtl_cs_sizes <- get_cs_sizes_by_region(fsusie_haqtl$cs)
cs_sizes_histogram(fsusie_haqtl_cs_sizes,x_breaks = c(1:5,10,100,1000)) +
  scale_y_continuous(trans = "log10")
```

Steps to compute overlap with expression SNPs:

1. Find 1-SNP CSs in fsusie results and the SNPs contained in those CSs.

2. Sum up the susie PIPs for the SNPs identified in Step 1.

```{r fsusie-haqtl-overlap-averaged}
fsusie_haqtl_cs1snp <- get_cs1snp(fsusie_haqtl$cs)
x <- get_pip_sum_overlap(susie$pips,fsusie_haqtl_cs1snp$id)
cat("Number of expression SNPs:",get_pip_sum(susie$pips),"\n")
cat("Number of acetylation SNPs:",nrow(fsusie_haqtl_cs1snp),"\n")
cat("Number of expression SNPs overlapping with acetylation SNPs:",x,"\n")
```

```{r temp}
x <- unlist(lapply(susie$pips,function (x) x$id))
y <- unlist(lapply(fsusie_haqtl$pips,function (x) x$id))
print(length(intersect(x,y)))
```

Separately, identify the SNPs in 1-SNP CSs that overlap with "high
confidence" expression SNPs (say, SNPs with PIP > 0.75). 

```{r fsusie-haqtl-overlap-high-conf}
susie_highconf_snps <- get_highconf_snps(susie$pips,level = 0.75)
ids <- intersect(susie_highconf_snps$id,fsusie_haqtl_cs1snp$id)
print(ids)
```

fSuSiE fine-mapping of mQTLs and overlap with expression SNPs
-------------------------------------------------------------

Region sizes in base-pairs and SNPs:

```{r fsusie-mqtl-region-sizes, fig.height=3, fig.width=6.25}
p1 <- region_sizes_histogram(fsusie_mqtl$regions,fsusie_mqtl$pips)
p2 <- num_snps_histogram(fsusie_mqtl$regions)
plot_grid(p1,p2)
```

Number of CSs per region:

```{r fsusie-mqtl-num-cs}
table(CSs = fsusie_mqtl$regions$num_cs)
```

CS sizes:

```{r fsusie-mqtl-cs-sizes, fig.height=3, fig.width=3, warning=FALSE}
fsusie_mqtl_cs_sizes <- get_cs_sizes_by_region(fsusie_mqtl$cs)
cs_sizes_histogram(fsusie_mqtl_cs_sizes,x_breaks = c(1:5,10,100,1000)) +
  scale_y_continuous(trans = "log10")
```

Get CSs containing exactly 1 SNP in susie results:

```{r cs1snp-susie}
n <- nlevels(susie$cs$region)
cs1snp_susie <- vector("list",n)
region_names <- levels(susie$cs$region)
names(cs1snp_susie) <- region_names
for (i in region_names) {
  dat <- subset(susie$cs,region == i)
  x <- table(dat$cs)
  x <- as.numeric(names(x)[x == 1])
  if (length(x) > 0)
   cs1snp_susie[[i]] <- subset(dat,is.element(cs,x))
}
cs1snp_susie <- do.call(rbind,cs1snp_susie)
rownames(cs1snp_susie) <- NULL
```

Get CSs containing exactly 1 SNP in fsusie mQTL results:

```{r cs1snp-fsusie-mqtl}
n <- nlevels(fsusie_mqtl$cs$region)
cs1snp_fsusie_mqtl <- vector("list",n)
region_names <- levels(fsusie_mqtl$cs$region)
names(cs1snp_fsusie_mqtl) <- region_names
for (i in region_names) {
  dat <- subset(fsusie_mqtl$cs,region == i)
  x <- table(dat$cs)
  x <- as.numeric(names(x)[x == 1])
  if (length(x) > 0)
   cs1snp_fsusie_mqtl[[i]] <- subset(dat,is.element(cs,x))
}
cs1snp_fsusie_mqtl <- do.call(rbind,cs1snp_fsusie_mqtl)
```


```{r}
intersect(cs1snp_susie$id,cs1snp_fsusie_mqtl$id)
```

+ Examine fsusie mQTL results.

+ Examine fsusie haQTL results.

+ Examine overlap between susie eQTL and fsusie mQTL results.

+ Examine overlap between susie eQTL and fsusie mQTL results.
