---
title: Examine ROSMAP H3K27ac fine-mapping results
author: William Denault, Hao Sun, Angjing Liu, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

**Note:** If you would like to run this analysis on your computer, you
will first need to download the fine-mapping outputs. They can be
downloaded from [here][box-download-link]. Once you have downloaded
the files, copy each file to the "data" or "outputs" subdirectory.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load some packages and custom functions used in the code below:

```{r load-pkgs, message=FALSE}
library(data.table)
library(dplyr)
library(ggplot2)
library(cowplot)
source("../code/rosmap_functions_more.R")
```

Load the gene annotations used in some of the analyses below. 

```{r load-gene-annotations}
gene_file <-
  file.path("../data/genome_annotations",
    "Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz")
genes <- get_gene_annotations(gene_file)
```

Load the allele frequencies computed by PLINK:

```{r read-allele-freqs}
load("../data/afreq.RData")
```

Load the H3K27ac SNP results generated by SuSiE-topPC, fSuSiE and the
SNP-peak association testing:

```{r read-snp-results}
peak_assoc_file <- "../outputs/ROSMAP_haQTL_qtl_snp_qval0.05.tsv.gz"
snps_susie_file <- "../outputs/ROSMAP_haQTL_cs_snp_toppc1_annotation.tsv.gz"
snps_fsusie_file <- "../outputs/ROSMAP_haQTL_cs_snp_annotation.tsv.gz"
peak_assoc  <- read_enrichment_results(peak_assoc_file,n = 8)
snps_susie  <- read_enrichment_results(snps_susie_file,n = 6)
snps_fsusie <- read_enrichment_results(snps_fsusie_file,n = 7)
snps_susie  <- snps_susie[1:6]
snps_fsusie <- snps_fsusie[1:7]
snps_susie$region <-
  sapply(strsplit(snps_susie$cs,":",fixed = TRUE),"[[",2)
snps_susie  <- transform(snps_susie,
                         region = factor(region),
                         cs     = factor(cs),
			 pc     = factor(pc))
snps_fsusie <- transform(snps_fsusie,
                         cs     = factor(cs),
			 region = factor(region),
			 study  = factor(study))
```

Add the allele frequencies to the H3K27ac fine-mapping results:

```{r add-allele-freqs, eval=FALSE}
ids  <- with(snps_susie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
snps_susie$maf <- afreq[rows,"maf"]
ids  <- with(snps_fsusie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
snps_fsusie$maf <- afreq[rows,"maf"]
```

There is no need to look at the TAD sizes here because the same TADs
were analyzed for both of the molecular traits (methylation and
H3K27ac).

[box-download-link]: https://uchicago.box.com/s/tt1vgg7vqayfthg0vsbl8gw0sdi0f1uo
