---
title: "Analysis of some susie and fsusie results on the ROSMAP data"
author: William Denault, Hao Sun, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

TO DO: GIVE OVERVIEW HERE.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages as well as some additional custom functions used in
the analysis below,

```{r load-pkgs, message=FALSE}
library(data.table)
library(ggplot2)
library(cowplot)
source("code/rosmap_functions.R")
setDTthreads(1)
```

Load the susie fine-mapping results on the "Inh_mega_eQTL" RNA-seq
data.

```{r load-susie-results}
datadir <- file.path("/project2/mstephens/fungen_xqtl/ftp_fgc_xqtl",
                     "analysis_result/finemapping_twas/prepared_results")
load(file.path(datadir,"susie_Inh_mega_eQTL.RData"))					 
susie <- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)
```

Load the fsusie fine-mapping results on the DLPFC methylation data.

```{r load-fsusie-results}
load(file.path(datadir,"fsusie_ROSMAP_DLPFC_mQTL.RData"))
fsusie <- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)
```

Load the gene annotations. Specifically I extract here only the
annotated gene transcripts for protein-coding genes as defined in the
Ensembl/Havana database.

```{r load-gene-annotations}
gene_file <-
  file.path("data/genome_annotations",
    "Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz")
genes <- get_gene_annotations(gene_file)
```

SuSiE fine-mapping of RNA-seq
-----------------------------

Size of regions in base-pairs and SNPs:

```{r susie-region-sizes, fig.height=3, fig.width=6}
susie$regions$pos_min <- sapply(susie$pips,function (x) min(x$pos))
susie$regions$pos_max <- sapply(susie$pips,function (x) max(x$pos))
susie$regions <- transform(susie$regions,size_bp = pos_max - pos_min)
p1 <- ggplot(susie$regions,aes(size_bp/1e6)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 64) +
  scale_x_continuous(breaks = seq(0,50,5)) +
  labs(x = "size of region (Mb)",
       y = "number of regions") +
  theme_cowplot(font_size = 10)
p2 <- ggplot(susie$regions,aes(num_snps)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 64) +
  scale_x_continuous(breaks = seq(0,1e5,1e4)) +
  labs(x = "number of SNPs",
       y = "number of regions") +
  theme_cowplot(font_size = 10)
plot_grid(p1,p2)
```

Number of CSs per region:

```{r susie-num-cs}
table(CSs = susie$regions$num_cs)
```

CS sizes:

```{r susie-cs-sizes, fig.height=3, fig.width=3}
n <- nlevels(susie$cs$region)
cs_sizes <- vector("list",n)
region_names <- levels(susie$cs$region)
names(cs_sizes) <- region_names
for (i in region_names)
  cs_sizes[[i]] <- as.vector(table(factor(subset(susie$cs,region == i)$cs)))
cs_sizes <- unlist(cs_sizes)
pdat <- data.frame(cs_size = cs_sizes)
ggplot(pdat,aes(cs_size)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 64) +
  scale_x_continuous(trans = "log10",breaks = c(1:5,10,100,1000)) +
  labs(x = "number of SNPs",
       y = "number of CSs") +
  theme_cowplot(font_size = 10)
```

NEXT: Create a plot showing the distance between the susie SNP and the
gene's TSS. First get the TSS for each gene/region.

```{r get-gene-tss}
region_names <- susie$regions$region_name
rownames(susie$regions) <- region_names
susie$regions$tss <- as.numeric(NA)
for (i in region_names) {
  j <- which(genes$ensembl == i)
  if (length(j) == 1)
    susie$regions[i,"tss"] <- genes[j,"start"]
}
```

Now compute the distance to the TSS weighted by the PIPs:

```{r dist-to-tss}
n      <- length(susie$pips)
bins   <- c(-Inf,seq(-1e6,1e6,5e4),Inf)
counts <- rep(0,length(bins) - 1)
for (i in 1:n) {
  pips <- susie$pips[[i]]
  dist_to_tss <- susie$regions[i,"tss"] - pips$pos
  dist_to_tss <- cut(dist_to_tss,bins)
  res <- tapply(pips$pip,dist_to_tss,sum)
  res[is.na(res)] <- 0
  counts <- counts + res
} 
```

Plot the result:

```{r plot-dist-to-tss}
n      <- length(counts)
bins   <- bins[seq(2,n-1)]
counts <- counts[seq(2,n-1)]
pdat <- data.frame(pos = bins + 2.5e4,count = counts)
ggplot(pdat,aes(x = pos,y = count)) +
  geom_col(color = "white",fill = "darkblue") +
  labs(x = "distance from TSS",y = "SNPs weighted by PIPs") +
  theme_cowplot(font_size = 12)
```
