---
title: Examine ROSMAP H3K27ac fine-mapping results
author: William Denault, Hao Sun, Angjing Liu, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

**Note:** If you would like to run this analysis on your computer, you
will first need to download the fine-mapping outputs. They can be
downloaded from [here][box-download-link]. Once you have downloaded
the files, copy each file to the "data" or "outputs" subdirectory.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load some packages and custom functions used in the code below:

```{r load-pkgs, message=FALSE}
library(data.table)
library(dplyr)
library(ggplot2)
library(cowplot)
source("../code/rosmap_functions_more.R")
```

Load the gene annotations used in some of the analyses below. 

```{r load-gene-annotations}
gene_file <-
  file.path("../data/genome_annotations",
    "Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz")
genes <- get_gene_annotations(gene_file)
```

Load the allele frequencies computed by PLINK:

```{r read-allele-freqs}
load("../data/afreq.RData")
```

Load the H3K27ac SNP results generated by SuSiE-topPC, fSuSiE and the
SNP-peak association testing:

```{r read-snp-results}
assoc_file       <- "../outputs/ROSMAP_haQTL_qtl_snp_qval0.05.tsv.gz"
snps_susie_file  <- "../outputs/ROSMAP_haQTL_cs_snp_toppc1_annotation.tsv.gz"
snps_fsusie_file <- "../outputs/ROSMAP_haQTL_cs_snp_annotation.tsv.gz"
assoc       <- read_enrichment_results(assoc_file,n = 8)
snps_susie  <- read_enrichment_results(snps_susie_file,n = 6)
snps_fsusie <- read_enrichment_results(snps_fsusie_file,n = 7)
snps_susie  <- snps_susie[1:6]
snps_fsusie <- snps_fsusie[1:7]
snps_susie$region <-
  sapply(strsplit(snps_susie$cs,":",fixed = TRUE),"[[",2)
snps_susie  <- transform(snps_susie,
                         region = factor(region),
                         cs     = factor(cs),
                         pc     = factor(pc))
snps_fsusie <- transform(snps_fsusie,
                         cs     = factor(cs),
                         region = factor(region),
                         study  = factor(study))
```

Add the allele frequencies to the H3K27ac fine-mapping results:

```{r add-allele-freqs}
ids  <- with(snps_susie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
snps_susie$maf <- afreq[rows,"maf"]
ids  <- with(snps_fsusie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
snps_fsusie$maf <- afreq[rows,"maf"]
```

Also load the peak-level results generated by fSuSiE:

```{r read-peak-results}
peaks_fsusie_file <-
  "../outputs/ROSMAP_haQTL_cs_effect_ha_peak_annotation.tsv.gz"
peaks_fsusie <- read_enrichment_results(peaks_fsusie_file,n = 9)
peaks_fsusie <- peaks_fsusie[1:9]
peaks_fsusie$region <-
  sapply(strsplit(peaks_fsusie$cs,":",fixed = TRUE),"[[",2)
peaks_fsusie <- transform(peaks_fsusie,
                          cs      = factor(cs), 
 						  region  = factor(region),
                          context = factor(context))
```

Keep only CSs if the MAF of the sentinel SNP is >5%:

```{r filter-by-maf}
keep        <- tapply(snps_susie[c("pip","maf")],snps_susie$cs,
                      function (x) x[which.max(x$pip),"maf"] >= 0.05)
keep_cs     <- names(which(keep))
snps_susie  <- subset(snps_susie,is.element(cs,keep_cs))
snps_susie  <- transform(snps_susie,
                         region = factor(region),
                         cs     = factor(cs))
keep        <- tapply(snps_fsusie[c("pip","maf")],snps_fsusie$cs,
                      function (x) x[which.max(x$pip),"maf"] >= 0.05)
keep_cs     <- names(which(keep))
snps_fsusie <- subset(snps_fsusie,is.element(cs,keep_cs))
snps_fsusie <- transform(snps_fsusie,
                         region = factor(region),
                         cs     = factor(cs))
```

Apply this same MAF filter to the SNP-peak association tests and the
fSuSiE peak-level results:

```{r filter-by-maf-more}
ids          <- with(assoc,paste(chr,pos,sep = "_"))
rows         <- match(ids,afreq$id)
assoc$maf    <- afreq[rows,"maf"]
assoc        <- subset(assoc,maf >= 0.05)
peaks_fsusie <- subset(peaks_fsusie,is.element(cs,keep_cs))
peaks_fsusie <- transform(peaks_fsusie,
                          region = factor(region),
                          cs     = factor(cs))
```

There is no need to look at the TAD sizes here because the same TADs
were analyzed for both of the molecular traits (methylation and
H3K27ac).

These histograms summarize the number of CSs per TAD:

```{r num-cs-per-tad, fig.height=2, fig.width=6, warning=FALSE}
pdat1 <- get_cs_vs_tad_size(snps_susie)
pdat2 <- get_cs_vs_tad_size(snps_fsusie)
pdat1 <- transform(pdat1,num_cs = factor(num_cs,1:20))
pdat2 <- transform(pdat2,num_cs = factor(num_cs,1:20))
p1 <- ggplot(pdat1,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "tomato") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "tomato") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

Compare discovery of causal SNPs (number of CSs) in the SuSiE-topPC
and fSuSiE analyses:

```{r compare-num-cs-per-tad, fig.height=2.75, fig.width=3.75, warning=FALSE}
dat1 <- get_cs_vs_tad_size(snps_susie)
dat2 <- get_cs_vs_tad_size(snps_fsusie)
dat1 <- dat1[c(1,3)]
dat2 <- dat2[c(1,3)]
names(dat1) <- c("tad","num_cs_susie")
names(dat2) <- c("tad","num_cs_fsusie")
dat <- merge(dat1,dat2,all = TRUE)
rows <- which(is.na(dat$num_cs_susie))
dat[rows,"num_cs_susie"] <- 0
pdat <- melt(with(dat,table(num_cs_susie,num_cs_fsusie)))
rows <- which(pdat$value == 0)
pdat[rows,"value"] <- NA
p <- ggplot(pdat,aes(x = num_cs_susie,y = num_cs_fsusie,size = value)) +
  geom_point(color = "white",fill = "tomato",shape = 21) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_x_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_y_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_size(breaks = c(1,10,100)) +
  labs(x = "SuSiE-topPC",y = "fSuSiE",size = "number of TADs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r compare-num-cs-per-tad-ggsave, echo=FALSE, warning=FALSE}
ggsave("cs_per_tad_scatterplot_h3k27ac.pdf",p,height = 3,width = 4)
```

Flag the "duplicate" CSs:

```{r flag-duplicated-cs, message=FALSE, cache=TRUE}
root_cs_susie  <- create_cs_maps(snps_susie)$root
root_cs_fsusie <- create_cs_maps(snps_fsusie)$root
```

Then remove the duplicate CSs:

```{r remove-duplicated-cs}
snps_susie  <- subset(snps_susie,is.element(cs,root_cs_susie))
snps_fsusie <- subset(snps_fsusie,is.element(cs,root_cs_fsusie))
snps_susie  <- transform(snps_susie,cs = factor(cs))
snps_fsusie <- transform(snps_fsusie,cs = factor(cs))
```

Also remove the duplicate CSs from the fSuSiE peak-level results:

```{r remove-duplicated-cs-2}
nodup_cs     <- levels(snps_fsusie$cs)
peaks_fsusie <- subset(peaks_fsusie,is.element(cs,nodup_cs))
peaks_fsusie <- transform(peaks_fsusie,cs = factor(cs))
```

Compare the sizes of the CSs in the SuSiE-topPC and fSuSiE analyses:

```{r cs-sizes, fig.height=2, fig.width=4, warning=FALSE}
bins <- c(0,1,2,5,10,20,Inf)
cs_size_susie  <- as.numeric(table(snps_susie$cs))
cs_size_fsusie <- as.numeric(table(snps_fsusie$cs))
cs_size_susie  <- cut(cs_size_susie,bins)
cs_size_fsusie <- cut(cs_size_fsusie,bins)
levels(cs_size_susie) <- bins[-1]
levels(cs_size_fsusie) <- bins[-1]
p1 <- ggplot(data.frame(cs_size = cs_size_susie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "tomato",
                 width = 0.65) +
  labs(x = "CS size",y = "number of CSs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(data.frame(cs_size = cs_size_fsusie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "tomato",
                 width = 0.65) +
  scale_y_continuous(breaks = seq(0,2000,250)) + 
  labs(x = "CS size",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

Here are the exact numbers:

```{r cs-sizes-table}
table(cs_size_susie)
table(cs_size_fsusie)
```

```{r cs-sizes-ggsave, echo=FALSE, warning=FALSE}
ggsave("cs_sizes_h3k27ac.pdf",
       plot_grid(p1,p2,nrow = 1,ncol = 2),
	   height = 2,width = 4)
```

We expect that the vast majority of the causal SNPs will be very close
to the TSS. Let's verify this empirically.

First, add a "min_dist_to_TSS" column to each set of results:

```{r add-min-dist-to-tss}
snps_assoc  <- get_top_snp_per_location(assoc)
snps_assoc  <- add_min_dist_to_tss(snps_assoc,genes)
snps_susie  <- add_min_dist_to_tss(snps_susie,genes)
snps_fsusie <- add_min_dist_to_tss(snps_fsusie,genes)
```

This next code chunk computes the histogram for the TSS plot:

```{r compute-weighted-distance-to-tss}
bin_size <- 5e4
bins <- c(-Inf,seq(-5e5,5.5e5,bin_size),Inf)
bins <- bins - bin_size/2
counts_assoc <- as.numeric(table(cut(snps_assoc$min_dist_to_tss,bins)))
counts_susie <- tapply(snps_susie$pip,
                       cut(snps_susie$min_dist_to_tss,bins),
					   function (x) sum(x,na.rm = TRUE))
counts_fsusie <- tapply(snps_fsusie$pip,
                        cut(snps_fsusie$min_dist_to_tss,bins),
						function (x) sum(x,na.rm = TRUE))
```

And now we can plot the result:

```{r plot-weighted-distance-to-tss, fig.height=2.5, fig.width=3.5}
n <- length(bins)
i <- seq(2,n-2)
bin_centers   <- bins[i] + bin_size/2
counts_assoc  <- counts_assoc[i]
counts_susie  <- counts_susie[i]
counts_fsusie <- counts_fsusie[i]
counts_assoc  <- counts_assoc/sum(counts_assoc)
counts_susie  <- counts_susie/sum(counts_susie)
counts_fsusie <- counts_fsusie/sum(counts_fsusie)
pdat <- data.frame(method = rep(c("assoc","susie","fsusie"),
                                each = length(bin_centers)),
                   dist   = rep(bin_centers/1000,times = 3),
                   freq   = c(counts_assoc,counts_susie,counts_fsusie),
				   stringsAsFactors = TRUE)
p <- ggplot(pdat,aes(x = dist,y = freq,color = method)) +
  geom_line(linewidth = 0.5) +
  geom_point(size = 1) +
  scale_x_continuous(breaks = seq(-500,500,100),limits = c(-500,500)) +
  scale_y_continuous(breaks = seq(0,1,0.1)) +
  scale_color_manual(values = c("darkblue","darkorange","dodgerblue")) +
  labs(x = "distance to TSS (kb)",y = "proportion of SNPs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r ggsave-weighted-distance-to-tss, echo=FALSE, warning=FALSE}
ggsave("dist_to_tss_h3k27ac.pdf",p,height = 2.5,width = 3.5)
```

Now let's look closely at the 1-SNP CSs.

```{r get-1-snp-cs}
cs1snp_susie  <- table(snps_susie$cs)
cs1snp_susie  <- names(cs1snp_susie)[cs1snp_susie == 1]
cs1snp_susie  <- subset(snps_susie,is.element(cs,cs1snp_susie))
cs1snp_fsusie <- table(snps_fsusie$cs)
cs1snp_fsusie <- names(cs1snp_fsusie)[cs1snp_fsusie == 1]
cs1snp_fsusie <- subset(snps_fsusie,is.element(cs,cs1snp_fsusie))
```

Many of the PIPs in the 1-SNP CSs are 1 or very close to 1:

```{r pip-1-snp-cs}
table(cut(1 - cs1snp_susie$pip,c(0,0.0001,0.001,0.01,1)))
table(cut(1 - cs1snp_fsusie$pip,c(0,0.0001,0.001,0.01,1)))
```

This plot shows the distribution of the distances between the haSNP
(in the 1-SNP CS) and the nearest H3K27ac peak affected by that SNP:

Count the number of affected peaks per TAD from the fSuSiE results:

```{r  get-peaks-per-tad-fsusie}
peaks_per_tad_fsusie <-
  with(peaks_fsusie,tapply(ID,region,function (x) length(unique(x))))
```

Counting the number of H3K27ac peaks per TAD from the SNP-peak
association tests is a little more complicated because the peaks were
not assigned to TADs in the results.

```{r get-peaks-per-tad-assoc}
tads <- get_tad_info(levels(peaks_fsusie$region))
peaks_per_tad_assoc <- count_features_per_tad(assoc,tads)
```

These two plots summarize the number of affected peaks per TAD:

```{r plot-peaks-per-tad, fig.height=2, fig.width=5, warning=FALSE}
peaks_per_tad_assoc[peaks_per_tad_assoc == 0] <- NA
peaks_per_tad_fsusie[peaks_per_tad_fsusie == 0] <- NA
pdat1 <- data.frame(x = peaks_per_tad_assoc)
pdat2 <- data.frame(x = peaks_per_tad_fsusie)
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = "white",fill = "tomato",bins = 64) +
  xlim(0,100) +
  labs(x = "number of H3K27ac peaks",y = "number of TADs",
       title = "SNP-peak association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "tomato",bins = 64) +
  xlim(0,100) +
  labs(x = "number of H3K27ac peaks",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

A very small number of TADs have more than 100 affected peaks:

```{r peaks-per-tad-more}
mean(peaks_per_tad_assoc > 100,na.rm = TRUE)
mean(peaks_per_tad_fsusie > 100,na.rm = TRUE)
```

This plot compares the number of affected H3K27ac peaks per TAD
identified by fSuSiE and by the SNP-peak association tests:

```{r compare-peaks-per-tad, fig.height=2.5, fig.width=2.5, warning=FALSE}
pdat <- data.frame(assoc  = peaks_per_tad_assoc,
                   fsusie = peaks_per_tad_fsusie)
p <- ggplot(pdat,aes(x = assoc,y = fsusie)) +
  geom_point(color = "tomato") +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  labs(x = "SNP-peak association tests",y = "fSusiE") +
  xlim(0,200) + 
  ylim(0,200) + 
  theme_cowplot(font_size = 10)
print(p)
```

```{r compare-peaks-per-tad-ggsave, echo=FALSE, warning=FALSE}
ggsave("affected_h3k27ac_peaks_per_tad.pdf",p,height=2.5,width=2.5)
```

These next two plots compare the number of affected peaks per CS for
fSuSiE to the number of affected peaks per SNP from the association
tests:

```{r count-peaks-per-snp, fig.height=2, fig.width=4.5}
x <- factor(assoc$variant_id)
peaks_per_snp_assoc <- tapply(assoc$molecular_trait_id,x,
                              function (x) length(unique(x)))
rm(x)
peaks_per_snp_fsusie <-
  with(peaks_fsusie,tapply(ID,cs,function (x) length(unique(x))))
pdat1 <- data.frame(x = peaks_per_snp_assoc)
pdat2 <- data.frame(x = peaks_per_snp_fsusie)
pdat1 <- subset(pdat1,x <= 10)
pdat2 <- subset(pdat2,x <= 40)
pdat1 <- transform(pdat1,x = factor(x))
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_bar(color = "white",fill = "tomato") +
  labs(x = "number of H3K27ac peaks",y = "number of SNPs",
       title = "SNP-peak association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "tomato",bins = 25) +
  labs(x = "number of H3K27ac peaks",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

```{r count-peaks-per-snp-ggsave, echo=FALSE, warning=FALSE}
ggsave("affected_h3k27ac_peaks_per_snp.pdf",
       plot_grid(p1,p2,nrow = 1,ncol = 2),
	   height = 2,width = 4.5)
```

A small proportion of the SNPs and CSs were not plotted because they
had an unusually large number of affected peaks:

```{r count-peaks-per-snp-more}
mean(peaks_per_snp_assoc > 10)
mean(peaks_per_snp_fsusie > 40)
```

Finally, this plot shows the distribution of the distances between the
haSNP (in a 1-SNP CS) and the *nearest peak* affected by that SNP:

```{r dist-hasnp-peak-cs1snp, fig.height=2, fig.width=2.5, warning=FALSE}
peaks_fsusie_cs1snp <- subset(peaks_fsusie,
                              is.element(cs,cs1snp_fsusie$cs))
rows <- match(peaks_fsusie_cs1snp$cs,cs1snp_fsusie$cs)
peaks_fsusie_cs1snp$variant_pos <- cs1snp_fsusie[rows,"pos"]
peaks_fsusie_cs1snp <-
  transform(peaks_fsusie_cs1snp,
            cs   = factor(cs),
            dist = (peak_start + peak_end)/2 - variant_pos)
pdat <- tapply(peaks_fsusie_cs1snp$dist,peaks_fsusie_cs1snp$cs,
               function (x) {
			     i <- which.min(abs(x))
				 return(x[i])
    		   })
pdat <- data.frame(dist = pdat/1e6)
p <- ggplot(subset(pdat,abs(dist) < 2),aes(x = dist)) +
  geom_histogram(bins = 128,color = "tomato",fill = "tomato") +
  scale_y_continuous(breaks = seq(0,500,100)) +
  labs(x = "H3K27ac peak - haSNP pos. (Mb)",
       y = "number of haSNPs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r dist-hasnp-peak-cs1np-more}
table(abs(pdat$dist) < 2)
```

```{r dist-hasnp-peak-cs1snp-ggsave, echo=FALSE, warning=FALSE}
ggsave("dist_hasnp_peak_cs1snp.pdf",p,height = 2.25,width = 2.5)
```

[box-download-link]: https://uchicago.box.com/s/tt1vgg7vqayfthg0vsbl8gw0sdi0f1uo
