---
title: Examine ROSMAP methylation fine-mapping results
author: William Denault, Hao Sun, Angjing Liu, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

**Note:** If you would like to run this analysis on your computer, you
will first need to download the fine-mapping outputs. They can be
downloaded from [here][box-download-link]. Once you have downloaded
the files, copy each file to the "data" or "outputs" subdirectory.

Load some packages and custom functions used in the code below:

```{r load-pkgs, message=FALSE}
library(data.table)
library(dplyr)
library(ggplot2)
library(cowplot)
source("../code/rosmap_functions_more.R")
```

Load the gene annotations used in some of the analyses below. 

```{r load-gene-annotations}
gene_file <-
  file.path("../data/genome_annotations",
    "Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz")
genes <- get_gene_annotations(gene_file)
```

Load the allele frequencies computed by PLINK:

```{r read-allele-freqs}
load("../data/afreq.RData")
```

Next I load methylation SNP results generated by SuSiE-topPC, fSuSiE
and the SNP-CpG association testing:

```{r read-snp-results}
assoc_file       <- "../outputs/ROSMAP_mQTL_qtl_snp_qval0.05.tsv.gz"
snps_susie_file  <- "../outputs/ROSMAP_mQTL_cs_snp_toppc1_annotation.tsv.gz"
snps_fsusie_file <- "../outputs/ROSMAP_mQTL_cs_snp_annotation.tsv.gz"
snps_susie  <- read_enrichment_results(snps_susie_file,n = 6)
snps_fsusie <- read_enrichment_results(snps_fsusie_file,n = 7)
assoc       <- read_enrichment_results(assoc_file,n = 8)
snps_susie  <- snps_susie[1:6]
snps_fsusie <- snps_fsusie[1:7]
snps_susie$region <-
  sapply(strsplit(snps_susie$cs,":",fixed = TRUE),"[[",2)
snps_susie  <- transform(snps_susie,
                         region = factor(region),
                         cs     = factor(cs),
			  			 pc     = factor(pc))
snps_fsusie <- transform(snps_fsusie,
                         cs     = factor(cs),
						 region = factor(region),
						 study  = factor(study))
```

Add the allele frequencies to the methylation fine-mapping results:

```{r add-allele-freqs}
ids  <- with(snps_susie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
snps_susie$maf <- afreq[rows,"maf"]
ids  <- with(snps_fsusie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
snps_fsusie$maf <- afreq[rows,"maf"]
```

Also load the CpG-level results generated by fSuSiE:

```{r read-cpg-results}
cpg_fsusie_file    <- "../outputs/ROSMAP_mQTL_cs_effect_cpg_annotation.tsv.gz"
cpgs_fsusie        <- read_enrichment_results(cpg_fsusie_file,n = 9)
cpgs_fsusie        <- cpgs_fsusie[1:9]
cpgs_fsusie$region <- sapply(strsplit(cpgs_fsusie$cs,":",fixed = TRUE),"[[",2)
cpgs_fsusie <- transform(cpgs_fsusie,
                         cs      = factor(cs), 
                         region  = factor(region),
                         context = factor(context))
```

Keep only CSs if the MAF of the sentinel SNP is >5%:

```{r filter-by-maf}
keep        <- tapply(snps_susie[c("pip","maf")],snps_susie$cs,
                      function (x) x[which.max(x$pip),"maf"] >= 0.05)
keep_cs     <- names(which(keep))
snps_susie  <- subset(snps_susie,is.element(cs,keep_cs))
snps_susie  <- transform(snps_susie,
                         region = factor(region),
                         cs     = factor(cs))
keep        <- tapply(snps_fsusie[c("pip","maf")],snps_fsusie$cs,
                      function (x) x[which.max(x$pip),"maf"] >= 0.05)
keep_cs     <- names(which(keep))
snps_fsusie <- subset(snps_fsusie,is.element(cs,keep_cs))
snps_fsusie <- transform(snps_fsusie,
                         region = factor(region),
                         cs     = factor(cs))
```

Apply this same MAF filter to the SNP-CpG association tests and the
fSuSiE CpG-level results:

```{r filter-by-maf-more}
ids         <- with(assoc,paste(chr,pos,sep = "_"))
rows        <- match(ids,afreq$id)
assoc$maf   <- afreq[rows,"maf"]
assoc       <- subset(assoc,maf >= 0.05)
cpgs_fsusie <- subset(cpgs_fsusie,is.element(cs,keep_cs))
cpgs_fsusie <- transform(cpgs_fsusie,
                         region = factor(region),
                         cs     = factor(cs))
```

Let's now get an overview of the TADs. This is the number of
fine-mapping regions (TADs) that contained at least one CS in each of
the analyses:

```{r tad-counts}
nlevels(snps_susie$region)
nlevels(snps_fsusie$region)
```

This plot summarizes the sizes of the TADs that were analyzed by
SuSiE-topPC and fSuSiE:

```{r plot-tad-sizes, fig.height=2.75, fig.width=2.75, warning=FALSE}
plot_tad_sizes <- function (tads) {
  tad_size <- get_tad_sizes(tads)
  pdat <- data.frame(tad_size = tad_size)
  return(ggplot(pdat,aes(x = tad_size)) +
         geom_histogram(color = "white",fill = "black",bins = 48) +
         labs(x = "size (Mb)",y = "number of TADs") +
         theme_cowplot(font_size = 10))
}
tads <- levels(snps_fsusie$region)
p <- plot_tad_sizes(tads) +
  scale_x_continuous(limits = c(2,9),breaks = 1:10) +
  scale_y_continuous(breaks = seq(0,100,10))
print(p)
```

```{r save-plot-tad-sizes, echo=FALSE, warning=FALSE}
ggsave("tad_sizes_histogram.pdf",p,height = 3,width = 3)
```

Some other useful statistics on the TAD sizes:

```{r summarize-tad-sizes}
tad_size <- get_tad_sizes(tads)
length(tad_size)
range(tad_size)
mean(tad_size)
median(tad_size)
sum(tad_size > 9)
```

There are some technical concerns about fine-mapping mSNPs that
overlap with CpGs, so these SNPs (and the CSs that include these SNPs)
should be removed from the results.

For the SNP-CpG association tests, it turns out that there are no
SNPs to "blacklist":

```{r blacklist-assoc}
sum(with(assoc,pos == molecular_trait_id_pos))
```

Now identify the blacklisted mSNPs as the mSNPs that have the same
position as a CpG (I will use both "peak start" and "peak end"):

```{r get-blacklisted-snps}
ids_snp <- c(with(snps_susie,paste(chr,pos,sep = "_")),
             with(snps_fsusie,paste(chr,pos,sep = "_")))
ids_cpg <- with(cpgs_fsusie,
                c(paste(chr,peak_start,sep = "_"),
				  paste(chr,peak_end,sep = "_")))
blacklisted_ids <- intersect(ids_snp,ids_cpg)
```

Now remove all SuSiE and fSuSiE CSs containing one or more blacklisted
SNPs:

```{r remove-blacklisted-cs}
ids            <- with(snps_susie,paste(chr,pos,sep = "_"))
rows           <- which(is.element(ids,blacklisted_ids))
blacklisted_cs <- unique(snps_susie[rows,]$cs)
snps_susie     <- subset(snps_susie,!is.element(cs,blacklisted_cs))
ids            <- with(snps_fsusie,paste(chr,pos,sep = "_"))
rows           <- which(is.element(ids,blacklisted_ids))
blacklisted_cs <- unique(snps_fsusie[rows,]$cs)
snps_fsusie    <- subset(snps_fsusie,!is.element(cs,blacklisted_cs))
snps_susie     <- transform(snps_susie,cs = factor(cs))
snps_fsusie    <- transform(snps_fsusie,cs = factor(cs))
```

These histograms summarize the number of CSs per TAD:

```{r num-cs-per-tad, fig.height=2, fig.width=5, warning=FALSE}
pdat1 <- get_cs_vs_tad_size(snps_susie)
pdat2 <- get_cs_vs_tad_size(snps_fsusie)
pdat1 <- transform(pdat1,num_cs = factor(num_cs,1:20))
pdat2 <- transform(pdat2,num_cs = factor(num_cs,1:20))
p1 <- ggplot(pdat1,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

Compare discovery of causal SNPs (number of CSs) in the SuSiE-topPC
and fSuSiE analyses:

```{r compare-num-cs-per-tad, fig.height=2.75, fig.width=3.75, warning=FALSE}
dat1 <- get_cs_vs_tad_size(snps_susie)
dat2 <- get_cs_vs_tad_size(snps_fsusie)
dat1 <- dat1[c(1,3)]
dat2 <- dat2[c(1,3)]
names(dat1) <- c("tad","num_cs_susie")
names(dat2) <- c("tad","num_cs_fsusie")
dat <- merge(dat1,dat2,all = TRUE)
rows <- which(is.na(dat$num_cs_susie))
dat[rows,"num_cs_susie"] <- 0
pdat <- melt(with(dat,table(num_cs_susie,num_cs_fsusie)))
rows <- which(pdat$value == 0)
pdat[rows,"value"] <- NA
p <- ggplot(pdat,aes(x = num_cs_susie,y = num_cs_fsusie,size = value)) +
  geom_point(color = "white",fill = "darkblue",shape = 21) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_x_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_y_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_size(breaks = c(1,5,10,50,100)) +
  labs(x = "SuSiE-topPC",y = "fSuSiE",size = "number of TADs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r compare-num-cs-per-tad-ggsave, echo=FALSE, warning=FALSE}
ggsave("cs_per_tad_scatterplot_methyl.pdf",p,height = 3,width = 4)
```

Flag the "duplicate" CSs:

```{r flag-duplicated-cs, message=FALSE, cache=TRUE}
root_cs_susie  <- create_cs_maps(snps_susie)$root
root_cs_fsusie <- create_cs_maps(snps_fsusie)$root
```

Then remove the duplicate CSs:

```{r remove-duplicated-cs}
snps_susie  <- subset(snps_susie,is.element(cs,root_cs_susie))
snps_fsusie <- subset(snps_fsusie,is.element(cs,root_cs_fsusie))
snps_susie  <- transform(snps_susie,cs = factor(cs))
snps_fsusie <- transform(snps_fsusie,cs = factor(cs))
```

Also remove the duplicate CSs from the fSuSiE CpG-level results:

```{r remove-duplicated-cs-2}
nodup_cs    <- levels(snps_fsusie$cs)
cpgs_fsusie <- subset(cpgs_fsusie,is.element(cs,nodup_cs))
cpgs_fsusie <- transform(cpgs_fsusie,cs = factor(cs))
```

Compare the sizes of the CSs in the SuSiE-topPC and fSuSiE analyses:

```{r cs-sizes, fig.height=2, fig.width=4, warning=FALSE}
bins <- c(0,1,2,5,10,20,Inf)
snps_susie     <- transform(snps_susie,cs = factor(cs))
snps_fsusie    <- transform(snps_fsusie,cs = factor(cs))
cs_size_susie  <- as.numeric(table(snps_susie$cs))
cs_size_fsusie <- as.numeric(table(snps_fsusie$cs))
cs_size_susie  <- cut(cs_size_susie,bins)
cs_size_fsusie <- cut(cs_size_fsusie,bins)
levels(cs_size_susie) <- bins[-1]
levels(cs_size_fsusie) <- bins[-1]
p1 <- ggplot(data.frame(cs_size = cs_size_susie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue",
                 width = 0.65) +
  labs(x = "CS size",y = "number of CSs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(data.frame(cs_size = cs_size_fsusie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue",
                 width = 0.65) +
  scale_y_continuous(breaks = seq(0,1e4,2000)) +			 
  labs(x = "CS size",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

Here are the exact numbers:

```{r cs-sizes-table}
table(cs_size_susie)
table(cs_size_fsusie)
```

Here's another visualization of the same result:

```{r cs-sizes-another-plot, fig.height=2, fig.width=3.2}
pdat <- rbind(data.frame(method = "SuSiE-topPC",cs_size = cs_size_susie),
              data.frame(method = "fSuSiE",cs_size = cs_size_fsusie))
pdat <- transform(pdat,method = factor(method))
p <- ggplot(pdat,aes(x = cs_size,fill = method)) +
  geom_histogram(stat = "count",color = "white",width = 0.65,
                 position = "dodge") +
  labs(x = "CS size",y = "number of CSs") +
  scale_fill_manual(values = c("darkorange","dodgerblue")) +
  scale_y_continuous(breaks = seq(0,10000,1000)) +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
print(p)
```

```{r cs-sizesggsave, echo=FALSE, warning=FALSE}
ggsave("cs_sizes_methyl.pdf",p,height = 2,width = 3.2)
```

We expect that most of the causal SNPs to be very close to the nearest
TSS. Let's check this. This will involve a couple of intermediate
calculations.

```{r add-min-dist-to-tss}
snps_assoc  <- get_top_snp_per_location(assoc)
snps_assoc  <- add_min_dist_to_tss(snps_assoc,genes)
snps_susie  <- add_min_dist_to_tss(snps_susie,genes)
snps_fsusie <- add_min_dist_to_tss(snps_fsusie,genes)
```

For SuSiE-topPC and fSuSiE, we calculate "weighted" counts of SNPs,
in which the weights are given by the PIPs.

```{r compute-weighted-distance-to-tss}
bin_size <- 10000
bins <- c(-Inf,seq(-2e5,2e5,bin_size),Inf)
bins <- bins + bin_size/2
counts_assoc <- as.numeric(table(cut(snps_assoc$min_dist_to_tss,bins)))
counts_susie <- tapply(snps_susie$pip,
                       cut(snps_susie$min_dist_to_tss,bins),
					   function (x) sum(x,na.rm = TRUE))
counts_fsusie <- tapply(snps_fsusie$pip,
                        cut(snps_fsusie$min_dist_to_tss,bins),
						function (x) sum(x,na.rm = TRUE))
```

Now we can plot the result:

```{r plot-weighted-distance-to-tss, fig.height=2.5, fig.width=3.5}
n <- length(bins)
i <- seq(2,n-2)
bin_centers   <- bins[i] + bin_size/2
counts_assoc  <- counts_assoc[i]
counts_susie  <- counts_susie[i]
counts_fsusie <- counts_fsusie[i]
counts_assoc  <- counts_assoc/sum(counts_assoc)
counts_susie  <- counts_susie/sum(counts_susie)
counts_fsusie <- counts_fsusie/sum(counts_fsusie)
pdat <- data.frame(method = rep(c("assoc","susie","fsusie"),
                                each = length(bin_centers)),
                   dist   = rep(bin_centers/1000,times = 3),
                   freq   = c(counts_assoc,counts_susie,counts_fsusie),
				   stringsAsFactors = TRUE)
p <- ggplot(pdat,aes(x = dist,y = freq,color = method)) +
  geom_line(linewidth = 0.5) +
  geom_point(size = 1) +
  scale_x_continuous(breaks = seq(-200,200,50)) +
  scale_y_continuous(breaks = seq(0,1,0.05)) +
  scale_color_manual(values = c("darkblue","darkorange","dodgerblue")) +
  labs(x = "distance to TSS (kb)",y = "proportion of SNPs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r ggsave-weighted-distance-to-tss, echo=FALSE, warning=FALSE}
ggsave("dist_to_tss_methyl.pdf",p,height = 2.5,width = 3.5)
```

Now let's look closely at the 1-SNP CSs.

```{r get-1-snp-cs}
cs1snp_susie  <- table(snps_susie$cs)
cs1snp_susie  <- names(cs1snp_susie)[cs1snp_susie == 1]
cs1snp_susie  <- subset(snps_susie,is.element(cs,cs1snp_susie))
cs1snp_fsusie <- table(snps_fsusie$cs)
cs1snp_fsusie <- names(cs1snp_fsusie)[cs1snp_fsusie == 1]
cs1snp_fsusie <- subset(snps_fsusie,is.element(cs,cs1snp_fsusie))
```

Many of the PIPs in the 1-SNP CSs are 1 or very close to 1:

```{r pip-1-snp-cs}
table(cut(1 - cs1snp_susie$pip,c(0,0.0001,0.001,0.01,1)))
table(cut(1 - cs1snp_fsusie$pip,c(0,0.0001,0.001,0.01,1)))
```

Now let's turn to the recovery of affected CpGs.

Counting the number of CpGs per TAD is quite simple from the way the
fSuSiE results were compiled. It involves counting the number of unique
CpGs in each TAD:

```{r get-cpgs-per-tad-fsusie}
cpgs_per_tad_fsusie <-
  with(cpgs_fsusie,tapply(ID,region,function (x) length(unique(x))))
```

Counting the number of CpGs per TAD from the SNP-CpG association tests
is a little more complicated because the CpGs were not assigned to
TADs in the results.

```{r get-cpgs-per-tad-assoc}
tads <- get_tad_info(levels(cpgs_fsusie$region))
cpgs_per_tad_assoc <- count_features_per_tad(assoc,tads)
```

These plots summarize the number of affected CpGs per TAD:

```{r plot-cpgs-per-tad, fig.height=2, fig.width=5, warning=FALSE}
cpgs_per_tad_assoc[cpgs_per_tad_assoc == 0] <- NA
cpgs_per_tad_fsusie[cpgs_per_tad_fsusie == 0] <- NA
pdat1 <- data.frame(x = cpgs_per_tad_assoc)
pdat2 <- data.frame(x = cpgs_per_tad_fsusie)
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 64) +
  xlim(0,2000) +
  labs(x = "number of CpGs",y = "number of TADs",
       title = "SNP-CpG association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 64) +
  xlim(0,2000) +
  labs(x = "number of CpGs",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

A small number of TADs have more than 2,000 affected CpGs:

```{r cpgs-per-tad-more}
sum(cpgs_per_tad_assoc > 2000,na.rm = TRUE)
sum(cpgs_per_tad_fsusie > 2000,na.rm = TRUE)
```

This plot compares the number of affected CpGs identified by fSuSiE
and the SNP-CpG association tests:

```{r compare-cpgs-per-tad, fig.height=2.5, fig.width=2.5, warning=FALSE}
pdat <- data.frame(assoc  = cpgs_per_tad_assoc,
                   fsusie = cpgs_per_tad_fsusie)
p <- ggplot(pdat,aes(x = assoc,y = fsusie)) +
  geom_point(color = "darkblue") +
  geom_abline(intercept = 0,slope = 1,color = "magenta",linetype = "dotted") +
  labs(x = "SNP-CpG association tests",y = "fSusiE") +
  xlim(0,4100) + 
  ylim(0,4100) + 
  theme_cowplot(font_size = 10)
print(p)
```

Note that none of the TADs have more than 4,100 affected CpGs:

```{r compare-cpgs-per-tad-more}
sum(pdat$assoc > 4100,na.rm = TRUE)
sum(pdat$fsusie > 4100,na.rm = TRUE)
```

```{r compare-cpgs-per-tad-ggsave, echo=FALSE, warning=FALSE}
ggsave("affected_cpgs_per_tad.pdf",p,height = 2.5,width = 2.5)
```

One advantage of fSuSiE is that it is able to tell us which molecular
features are affected by which SNPs, and therefore it provides a more
coherent summary of how the SNPs affect methylation levels at a locus.
To examine this quantitatively, we compare the number of affected CpGs
per CS for fSuSiE to the number of affected CpGs per SNP from the
association tests. The result is much fewer CSs and many more
affected CpGs per CS:

```{r count-cpgs-per-snp, fig.height=2, fig.width=4.5}
x <- factor(assoc$variant_id)
cpgs_per_snp_assoc <- tapply(assoc$molecular_trait_id,x,
                             function (x) length(unique(x)))
rm(x)
cpgs_per_snp_fsusie <-
  with(cpgs_fsusie,tapply(ID,cs,function (x) length(unique(x))))
pdat1 <- data.frame(x = cpgs_per_snp_assoc)
pdat2 <- data.frame(x = cpgs_per_snp_fsusie)
pdat1 <- subset(pdat1,x <= 25)
pdat2 <- subset(pdat2,x <= 150)
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 25) +
  labs(x = "number of CpGs",y = "number of SNPs",
       title = "SNP-CpG association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 25) +
  labs(x = "number of CpGs",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

```{r count-cpgs-per-snp-ggave, echo=FALSE, warning=FALSE}
ggsave("affected_cpgs_per_snp.pdf",
       plot_grid(p1,p2,nrow = 1,ncol = 2),
       height = 2,width = 4.5)
```

A small proportion of the SNPs and CSs are not plotted because they
have an unusually large number of CpGs:

```{r count-cpgs-per-snp-more}
mean(cpgs_per_snp_assoc > 25)
mean(cpgs_per_snp_fsusie > 150)
```

Finally, this plot shows the distribution of the distances between the
mSNP and the *nearest CpG* affected by that SNP:

```{r dist-msnp-cpg-cs1snp, fig.height=2, fig.width=2.5, warning=FALSE}
cpgs_fsusie_cs1snp <- subset(cpgs_fsusie,
                             is.element(cs,cs1snp_fsusie$cs))
rows <- match(cpgs_fsusie_cs1snp$cs,cs1snp_fsusie$cs)
cpgs_fsusie_cs1snp$variant_pos <- cs1snp_fsusie[rows,"pos"]
cpgs_fsusie_cs1snp <- transform(cpgs_fsusie_cs1snp,
                                cs = factor(cs),
                                dist = peak_start - variant_pos)
pdat <- tapply(cpgs_fsusie_cs1snp$dist,cpgs_fsusie_cs1snp$cs,
               function (x) {
			     i <- which.min(abs(x))
				 return(x[i])
    		   })
pdat <- data.frame(dist = pdat/1000)
p <- ggplot(subset(pdat,abs(dist) < 100),aes(x = dist)) +
  geom_histogram(bins = 128,color = "darkblue",fill = "darkblue") +
  labs(x = "CpG - mSNP pos. (kb)",
       y = "number of mSNPs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r dist-msnp-cpg-cs1np-more}
table(abs(pdat$dist) < 100)
```

```{r dist-msnp-peak-cs1snp-ggsave, echo=FALSE, warning=FALSE}
ggsave("dist_msnp_cpg_cs1snp.pdf",p,height = 2.25,width = 2.5)
```

[box-download-link]: https://uchicago.box.com/s/tt1vgg7vqayfthg0vsbl8gw0sdi0f1uo
