---
title: Examine results on the ROSMAP data
author: William Denault, Hao Sun, Angjing Liu, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

*Add some text here giving an overview of this analysis.*

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

**Note:** If you would like to run this analysis on your computer, you
will first need to download the fine-mapping outputs. They can be
downloaded from [here][box-download-link]. Once you have downloaded
the files, copy them to the "outputs" subdirectory.

Load some packges used in the code below:

```{r load-pkgs, message=FALSE}
library(data.table)
library(ggplot2)
library(cowplot)
```

## Methylation fine-mapping

First I define a helper function for loading the enrichment results:

```{r read-enrichment-results}
# The "n" argument specifies the number of "meta data" columns.
# Columns after that are treated as the enrichment results. These
# columns contain only binary data (0 or 1) indicating whether or not
# the genomic feature (genetic variant or molecular trait location)
# is assigned that specific annotation.
read_enrichment_results <- function (filename, n) {
  out <- fread(filename,sep = "\t",stringsAsFactors = FALSE,header = TRUE)
  class(out) <- "data.frame"
  out <- transform(out,chr = factor(chr))
  cols <- seq(n + 1,ncol(out))
  for (i in cols)
    out[[i]] <- factor(out[[i]])
  return(out)
}
```

Next I load methylation SNP results generated by SuSiE-topPC and fSuSiE:

```{r read-methyl-snp-results}
methyl_snps_susie_file <-
  "../outputs/ROSMAP_mQTL_cs_snp_toppc1_annotation.tsv.gz"
methyl_snps_fsusie_file <- "../outputs/ROSMAP_mQTL_cs_snp_annotation.tsv.gz"
methyl_snps_susie  <- read_enrichment_results(methyl_snps_susie_file,n = 6)
methyl_snps_fsusie <- read_enrichment_results(methyl_snps_fsusie_file,n = 7)
methyl_snps_susie$region <-
  sapply(strsplit(methyl_snps_susie$cs,":",fixed = TRUE),"[[",2)
methyl_snps_susie  <- transform(methyl_snps_susie,
                                region = factor(region),
                                cs     = factor(cs),
			  				    pc     = factor(pc))
methyl_snps_fsusie <- transform(methyl_snps_fsusie,
                                cs     = factor(cs),
								region = factor(region),
								study  = factor(study))
```

Add text here.

```{r}
nlevels(methyl_snps_susie$region)
nlevels(methyl_snps_fsusie$region)
```

Get the TAD sizes.

```{r}
get_tad_sizes <- function (tads) {
  tads <- strsplit(tads,"_",fixed = TRUE)
  pos0 <- as.numeric(sapply(tads,"[[",2))
  pos1 <- as.numeric(sapply(tads,"[[",3))
  return((pos1 - pos0)/1e6)
}
```

Plot the TAD sizes.

```{r, fig.height=2.75, fig.width=2.75, warning=FALSE}
plot_tad_sizes <- function (tads) {
  tad_size <- get_tad_sizes(tads)
  pdat <- data.frame(tad_size = tad_size)
  return(ggplot(pdat,aes(x = tad_size)) +
         geom_histogram(color = "white",fill = "darkblue",bins = 48) +
         labs(x = "size (Mb)",y = "number of TADs") +
         theme_cowplot(font_size = 10))
}
tads <- levels(methyl_snps_fsusie$region)
plot_tad_sizes(tads) +
  scale_x_continuous(limits = c(2,9),breaks = 1:10) +
  scale_y_continuous(breaks = seq(0,100,10))
```

More on TAD sizes:

```{r}
tad_size <- get_tad_sizes(tads)
sum(tad_size > 9)
```

Create plots showing number of CSs by TAD.

```{r, fig.height=2.5, fig.width=2.5, warning=FALSE}
get_cs_vs_tad_size <- function (dat) {
  tads <- levels(dat$region)
  out <- data.frame(tad      = tads,
                    tad_size = get_tad_sizes(tads),
                    num_cs   = tapply(dat$cs,dat$region,
					                  function (x) length(unique(x))))
  rownames(out) <- NULL
  return(out)
}
pdat <- get_cs_vs_tad_size(methyl_snps_susie)
pdat <- transform(pdat,num_cs = factor(num_cs,1:16))
ggplot(pdat,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue") +
  scale_x_discrete(drop = FALSE) +
  theme_cowplot(font_size = 10)
```

```{r, fig.height=2.5, fig.width=2.5, warning=FALSE}
pdat <- get_cs_vs_tad_size(methyl_snps_fsusie)
pdat <- transform(pdat,num_cs = factor(num_cs,1:20))
ggplot(pdat,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue") +
  scale_x_discrete(drop = FALSE) +
  theme_cowplot(font_size = 10)
```

```{r}
```

## H3K27ac fine-mapping

TO DO.

[box-download-link]: https://uchicago.box.com/s/tt1vgg7vqayfthg0vsbl8gw0sdi0f1uo
