---
title: Examine methylation and H3K27ac fine-mapping results from the ROSMAP data
author: William Denault, Hao Sun, Angjing Liu, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

**Note:** If you would like to run this analysis on your computer, you
will first need to download the fine-mapping outputs. They can be
downloaded from [here][box-download-link]. Once you have downloaded
the files, copy each file to the "data" or "outputs" subdirectory.

Load some packges and custom functions used in the code below:

```{r load-pkgs, message=FALSE}
library(data.table)
library(dplyr)
library(ggplot2)
library(cowplot)
library(hexbin)
source("../code/merge_cs_sets.R")
```

This is a function I use to Extract the gene annotations from the GTF
("gene transfer format") file. Here we keep only the annotated gene
transcripts for protein-coding genes as defined in the Ensembl/Havana
database.

```{r get-gene-annotations}
get_gene_annotations <- function (gene_file) {
  out <- fread(file = gene_file,sep = "\t",header = FALSE,skip = 1)
  class(out) <- "data.frame"
  names(out) <- c("chromosome","source","feature","start","end","score",
                    "strand","frame","attributes")
  out <- out[c("chromosome","source","feature","start","end","strand",
               "attributes")]
  out <- transform(out,
                   chromosome = factor(chromosome),
                   source     = factor(source),
                   feature    = factor(feature),
                   strand     = factor(strand))
  out <- subset(out,
                source == "ensembl_havana" &
                feature == "transcript")
  out <-
    transform(out,
      ensembl   = sapply(strsplit(attributes,";"),
                         function (x) substr(x[[1]],10,24)),
      gene_type = sapply(strsplit(attributes,";"),
                         function (x) substr(x[[3]],13,nchar(x[[3]]) - 1)),
      gene_name = sapply(strsplit(attributes,";"),
                         function (x) substr(x[[4]],13,nchar(x[[4]]) - 1)))
  out <- out[-7]
  out <- transform(out,gene_type = factor(gene_type))
  out <- subset(out,gene_type == "protein_coding")
  rownames(out) <- NULL
  return(out)
}
```

Load the gene annotations which will be used in some of the analyses
below. Specifically, I extract here only the annotated gene transcripts
for protein-coding genes as defined in the Ensembl/Havana database.

```{r load-gene-annotations}
gene_file <-
  file.path("../data/genome_annotations",
    "Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz")
genes <- get_gene_annotations(gene_file)
```

## Methylation SNPs

First, I define a helper function for loading the enrichment results:

```{r read-enrichment-results}
# The "n" argument specifies the number of "meta data" columns.
# Columns after that are treated as the enrichment results. These
# columns contain only binary data (0 or 1) indicating whether or not
# the genomic feature (genetic variant or molecular trait location)
# is assigned that specific annotation.
read_enrichment_results <- function (filename, n) {
  out <- fread(filename,sep = "\t",stringsAsFactors = FALSE,header = TRUE)
  class(out) <- "data.frame"
  out <- transform(out,chr = factor(chr))
  if (ncol(out) > n) {
    cols <- seq(n + 1,ncol(out))
    for (i in cols)
      out[[i]] <- factor(out[[i]])
  }
  return(out)
}
```

Next I load methylation SNP results generated by SuSiE-topPC, fSuSiE
and the SNP-CpG association testing:

```{r read-methyl-snp-results}
methyl_cpg_assoc_file  <- "../outputs/ROSMAP_mQTL_qtl_snp_qval0.05.tsv.gz"
methyl_snps_susie_file <-
  "../outputs/ROSMAP_mQTL_cs_snp_toppc1_annotation.tsv.gz"
methyl_snps_fsusie_file <- "../outputs/ROSMAP_mQTL_cs_snp_annotation.tsv.gz"
methyl_snps_susie  <- read_enrichment_results(methyl_snps_susie_file,n = 6)
methyl_snps_fsusie <- read_enrichment_results(methyl_snps_fsusie_file,n = 7)
methyl_cpg_assoc   <- read_enrichment_results(methyl_cpg_assoc_file,n = 8)
methyl_snps_susie$region <-
  sapply(strsplit(methyl_snps_susie$cs,":",fixed = TRUE),"[[",2)
methyl_snps_susie  <- transform(methyl_snps_susie,
                                region = factor(region),
                                cs     = factor(cs),
			  				    pc     = factor(pc))
methyl_snps_fsusie <- transform(methyl_snps_fsusie,
                                cs     = factor(cs),
								region = factor(region),
								study  = factor(study))
```

Also, I load the allele frequencies computed by PLINK:

```{r read-allele-freqs}
afreq_file <- "../data/ROSMAP_NIA_WGS.leftnorm.bcftools_qc.plink_qc.afreq.gz"
system(paste("gzcat",afreq_file,"| cut -f 1,6 > temp1.txt"))
system(paste("gzcat",afreq_file,"| cut -f 2 | cut -d ':' -f 2 |",
             "cut -d '_' -f 1 > temp2.txt"))
system(paste("paste temp1.txt temp2.txt > rosmap.afreq"))			 
afreq <- fread("rosmap.afreq",sep = "\t",stringsAsFactors = FALSE,
               header = TRUE,nrows = Inf)
class(afreq) <- "data.frame"
names(afreq) <- c("chr","maf","pos")
afreq <- transform(afreq,
                   chr = factor(chr),
				   maf = pmin(maf,1 - maf))
afreq <- subset(afreq,maf > 0.001)
levels(afreq$chr) <- paste0("chr",1:22)
afreq$id <- sprintf("%s_%d",as.character(afreq$chr),afreq$pos)
```

Add the allele frequencies to the methylation fine-mapping results:

```{r add-allele-freqs-methyl}
ids  <- with(methyl_snps_susie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
methyl_snps_susie$maf <- afreq[rows,"maf"]
ids  <- with(methyl_snps_fsusie,paste(chr,pos,sep = "_"))
rows <- match(ids,afreq$id)
methyl_snps_fsusie$maf <- afreq[rows,"maf"]
```

### TAD statistics

This is the number of fine-mapping regions (TADs) that contained at
least one CS in each of the analyses:

```{r methyl-tad-counts}
nlevels(methyl_snps_susie$region)
nlevels(methyl_snps_fsusie$region)
```

This is a function we will use below to get the sizes of the TADs
(in Mb):

```{r get-tad-sizes}
get_tad_sizes <- function (tads) {
  tads <- strsplit(tads,"_",fixed = TRUE)
  pos0 <- as.numeric(sapply(tads,"[[",2))
  pos1 <- as.numeric(sapply(tads,"[[",3))
  return((pos1 - pos0)/1e6)
}
```

This plot summarizes the sizes of the TADs that were analyzed by
SuSiE-topPC and fSuSiE:

```{r plot-methyl-tad-sizes, fig.height=2.75, fig.width=2.75, warning=FALSE}
plot_tad_sizes <- function (tads) {
  tad_size <- get_tad_sizes(tads)
  pdat <- data.frame(tad_size = tad_size)
  return(ggplot(pdat,aes(x = tad_size)) +
         geom_histogram(color = "white",fill = "black",bins = 48) +
         labs(x = "size (Mb)",y = "number of TADs") +
         theme_cowplot(font_size = 10))
}
tads <- levels(methyl_snps_fsusie$region)
p <- plot_tad_sizes(tads) +
  scale_x_continuous(limits = c(2,9),breaks = 1:10) +
  scale_y_continuous(breaks = seq(0,100,10))
print(p)
```

```{r save-plot-methyl-tad-sizes, echo=FALSE, warning=FALSE}
ggsave("tad_sizes_histogram.pdf",p,height = 3,width = 3)
```

Some other useful statistics on the TAD sizes:

```{r summarize-methyl-tad-sizes}
tad_size <- get_tad_sizes(tads)
length(tad_size)
range(tad_size)
mean(tad_size)
median(tad_size)
sum(tad_size > 9)
```

### Identify and remove "blacklisted" mSNPs

There are some technical concerns about fine-mapping mSNPs that are
also CpGs, so these SNPs (and the CSs that include these SNPs)
should be removed from the results.

For the SNP-CpG association tests, finding the "blacklisted" is 
quite straightforward because they are just the mSNPs that have the
same positions as the CpGs:

```{r blacklist-assoc}
rows <- which(with(methyl_cpg_assoc,pos == molecular_trait_id_pos))
length(rows)
ids  <- with(methyl_cpg_assoc[rows,],paste(chr,pos,sep = "_"))
i <- match(ids,afreq$id)
table(cut(afreq[i,"maf"],c(0,0.01,0.1,1)))
rows <- which(with(methyl_cpg_assoc,pos != molecular_trait_id_pos))
methyl_cpg_assoc <- methyl_cpg_assoc[rows,]
```

For fSuSiE, it is a bit more complicated because we need to first load
the CpG results:

```{r read-methyl-cpg-results}
methyl_cpg_fsusie_file <-
  "../outputs/ROSMAP_mQTL_cs_effect_cpg_annotation.tsv.gz"
methyl_cpg_fsusie <- read_enrichment_results(methyl_cpg_fsusie_file,n = 9)
methyl_cpg_fsusie$region <-
  sapply(strsplit(methyl_cpg_fsusie$cs,":",fixed = TRUE),"[[",2)
methyl_cpg_fsusie <- transform(methyl_cpg_fsusie,
                               cs      = factor(cs), 
 						       region  = factor(region),
						       context = factor(context))
```

Now identify the blacklisted mSNPs as the mSNPs that have the same
position as a CpG (I will  use both "peak start" and "peak end"):

```{r get-blacklisted-snps}
ids_snp <- c(with(methyl_snps_susie,paste(chr,pos,sep = "_")),
             with(methyl_snps_fsusie,paste(chr,pos,sep = "_")))
ids_cpg <- with(methyl_cpg_fsusie,
                c(paste(chr,peak_start,sep = "_"),
				  paste(chr,peak_end,sep = "_")))
blacklisted_ids <- intersect(ids_snp,ids_cpg)
length(blacklisted_ids)
```

Before removing the blacklisted SNPs from the results, let's
take a quick look at them:

```{r examine-blacklisted-snps}
ids <- with(methyl_snps_fsusie,paste(chr,pos,sep = "_"))
rows <- which(is.element(ids,blacklisted_ids))
methyl_snps_fsusie_blacklisted <- methyl_snps_fsusie[rows,]
nrow(methyl_snps_fsusie_blacklisted)
sum(methyl_snps_fsusie_blacklisted$pip > 0.5)
```

It is interesting that a large fraction of the blacklisted SNPs have a
high PIPs. Let's also look at the MAFs of these "blacklisted" SNPs:

```{r blacklisted-snps-maf-hist, fig.height=2.5, fig.width=2.5}
ggplot(methyl_snps_fsusie_blacklisted,aes(x = log10(maf))) +
  geom_histogram(fill = "darkblue",color = "white",bins = 32) +
  labs(x = "log10(MAF)",y = "number of SNPs") +
  theme_cowplot(font_size = 10)
```

Before moving forward with the analysis, remove all SuSiE and
fSuSiE CSs containing the blacklisted SNPs:

```{r remove-blacklisted-cs}
ids                <- with(methyl_snps_susie,paste(chr,pos,sep = "_"))
rows               <- which(is.element(ids,blacklisted_ids))
blacklisted_cs     <- unique(methyl_snps_susie[rows,]$cs)
methyl_snps_susie  <- subset(methyl_snps_susie,!is.element(cs,blacklisted_cs))
ids                <- with(methyl_snps_fsusie,paste(chr,pos,sep = "_"))
rows               <- which(is.element(ids,blacklisted_ids))
blacklisted_cs     <- unique(methyl_snps_fsusie[rows,]$cs)
methyl_snps_fsusie <- subset(methyl_snps_fsusie,!is.element(cs,blacklisted_cs))
methyl_snps_susie  <- transform(methyl_snps_susie,cs = factor(cs))
methyl_snps_fsusie <- transform(methyl_snps_fsusie,cs = factor(cs))
```

### Compare SuSiE-topPC and fSuSiE methylation CSs

These histograms summarize the number of CSs per TAD:

```{r num-cs-per-tad-methyl, fig.height=2, fig.width=5, warning=FALSE}
get_cs_vs_tad_size <- function (dat) {
  tads <- levels(dat$region)
  out <- data.frame(tad      = tads,
                    tad_size = get_tad_sizes(tads),
                    num_cs   = tapply(dat$cs,dat$region,
					                  function (x) length(unique(x))))
  rownames(out) <- NULL
  return(out)
}
pdat1 <- get_cs_vs_tad_size(methyl_snps_susie)
pdat2 <- get_cs_vs_tad_size(methyl_snps_fsusie)
pdat1 <- transform(pdat1,num_cs = factor(num_cs,1:20))
pdat2 <- transform(pdat2,num_cs = factor(num_cs,1:20))
p1 <- ggplot(pdat1,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

Compare discovery of causal SNPs (number of CSs) in the SuSiE-topPC
and fSuSiE analyses:

```{r compare-num-cs-per-tad-methyl, fig.height=2.75, fig.width=3.75, warning=FALSE}
dat1 <- get_cs_vs_tad_size(methyl_snps_susie)
dat2 <- get_cs_vs_tad_size(methyl_snps_fsusie)
dat1 <- dat1[c(1,3)]
dat2 <- dat2[c(1,3)]
names(dat1) <- c("tad","num_cs_susie")
names(dat2) <- c("tad","num_cs_fsusie")
dat <- merge(dat1,dat2,all = TRUE)
rows <- which(is.na(dat$num_cs_susie))
dat[rows,"num_cs_susie"] <- 0
pdat <- melt(with(dat,table(num_cs_susie,num_cs_fsusie)))
rows <- which(pdat$value == 0)
pdat[rows,"value"] <- NA
p <- ggplot(pdat,aes(x = num_cs_susie,y = num_cs_fsusie,size = value)) +
  geom_point(color = "white",fill = "darkblue",shape = 21) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_x_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_y_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_size(breaks = c(1,5,10,50,100)) +
  labs(x = "SuSiE-topPC",y = "fSuSiE",size = "number of TADs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r compare-num-cs-per-tad-methyl-ggsave, echo=FALSE, warning=FALSE}
ggsave("cs_per_tad_scatterplot_methyl.pdf",p,height = 3,width = 4)
```

For plots counting CSs across the genome, I need to address the issue
that the same CSs are sometimes discovered in overlapping TADs. To
this end, this function removes CSs so that no two CSs share the same
SNP:

```{r create-cs-maps}
create_cs_maps <- function (info_df) {
  info_df %>% 
    arrange(cs) %>%
    group_by(variant_id) %>%
    mutate(cs_set = list(sort(unique(cs))),
           cs_set = sapply(cs_set,paste,collapse = ",")) %>%
    ungroup %>%
    count(cs,cs_set) %>%
    group_by(cs) %>%
    summarize(
	  cs_set = paste(sort(unique(unlist(strsplit(cs_set,",")))),
	                 collapse = ","),.groups = "drop") %>%
    merge_cs_sets
}
```

Now let's use this function to flag and remove the "duplicate" CSs.

```{r remove-duplicated-cs-methyl, message=FALSE, cache=TRUE}
root_cs_susie  <- create_cs_maps(methyl_snps_susie)$root
root_cs_fsusie <- create_cs_maps(methyl_snps_fsusie)$root
methyl_snps_susie_nodup  <- subset(methyl_snps_susie,
                                   is.element(cs,root_cs_susie))
methyl_snps_fsusie_nodup <- subset(methyl_snps_fsusie,
                                   is.element(cs,root_cs_fsusie))
```

Compare the sizes of the CSs in the SuSiE-topPC and fSuSiE analyses:

```{r cs-sizes-methyl, fig.height=2, fig.width=4, warning=FALSE}
bins <- c(0,1,2,5,10,20,Inf)
methyl_snps_susie_nodup <- transform(methyl_snps_susie_nodup,cs = factor(cs))
methyl_snps_fsusie_nodup <- transform(methyl_snps_fsusie_nodup,cs = factor(cs))
cs_size_susie  <- as.numeric(table(methyl_snps_susie_nodup$cs))
cs_size_fsusie <- as.numeric(table(methyl_snps_fsusie_nodup$cs))
cs_size_susie  <- cut(cs_size_susie,bins)
cs_size_fsusie <- cut(cs_size_fsusie,bins)
levels(cs_size_susie) <- bins[-1]
levels(cs_size_fsusie) <- bins[-1]
p1 <- ggplot(data.frame(cs_size = cs_size_susie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue",
                 width = 0.65) +
  labs(x = "CS size",y = "number of CSs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(data.frame(cs_size = cs_size_fsusie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "darkblue",
                 width = 0.65) +
  scale_y_continuous(breaks = seq(0,1e4,2000)) +			 
  labs(x = "CS size",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

Here are the exact numbers:

```{r cs-sizes-table-methyl}
table(cs_size_susie)
table(cs_size_fsusie)
```

### Distance-to-TSS plot

We expect that most of the causal SNPs to be very close to the nearest
TSS. Let's check this. This will involve a couple of intermediate
calculations.

```{r cs-sizes-methyl-ggsave, echo=FALSE, warning=FALSE}
ggsave("cs_sizes_methyl.pdf",
       plot_grid(p1,p2,nrow = 1,ncol = 2),
	   height = 2,width = 4)
```

This function adds a column to the SNP results containing the minimum
distance to the nearest TSS.

```{r add-min-dist-to-tss}
add_min_dist_to_tss <- function (dat, genes) {
  n <- nrow(dat)
  dat$min_dist_to_tss <- rep(Inf,n)
  n <- nrow(genes)
  for (i in 1:n) {
    rows <- which(as.character(genes[i,"chromosome"]) == as.character(dat$chr))
	if (length(rows) > 0) {
      if (genes[i,"strand"] == "+")
        d <- genes[i,"start"] - dat[rows,"pos"]
      else
        d <- dat[rows,"pos"] - genes[i,"end"]
	  i <- which(abs(d) < abs(dat[rows,"min_dist_to_tss"]))
  	  if (length(i) > 0) {
	    d    <- d[i]
	    rows <- rows[i]
	    dat[rows,"min_dist_to_tss"] <- d
      }
    }
  }
  return(dat)
}
```

This function is used to extract the top SNP per location (e.g., CpG)
from the association tests.

```{r get-top-snp-per-location}
get_top_snp_per_location <- function (dat) {
  x <- factor(dat$molecular_trait_id)
  qval <- dat$qvalue
  names(qval) <- with(dat,paste(chr,pos,sep = "_"))
  res <- tapply(qval,x,function (x) names(which.min(x)))
  res <- strsplit(res,"_",fixed = TRUE)
  out <- data.frame(chr = factor(sapply(res,"[[",1)),
                    pos = as.numeric(sapply(res,"[[",2)))
  out <- transform(out,chr = factor(chr))
  rownames(out) <- levels(x)
  return(out)
}
```

With these functions, we can compile the data needed for this plot.

```{r add-min-dist-to-tss-methyl}
methyl_snps_assoc        <- get_top_snp_per_location(methyl_cpg_assoc)
methyl_snps_assoc        <- add_min_dist_to_tss(methyl_snps_assoc,genes)
methyl_snps_susie_nodup  <- add_min_dist_to_tss(methyl_snps_susie_nodup,genes)
methyl_snps_fsusie_nodup <- add_min_dist_to_tss(methyl_snps_fsusie_nodup,genes)
```

For SuSiE-topPC and fSuSiE, we calculate "weighted" counts of SNPs,
in which the weights are given by the PIPs.

```{r compute-weighted-distance-to-tss-methyl}
bin_size <- 10000
bins <- c(-Inf,seq(-2e5,2e5,bin_size),Inf)
bins <- bins + bin_size/2
counts_assoc <- as.numeric(table(cut(methyl_snps_assoc$min_dist_to_tss,bins)))
counts_susie <- tapply(methyl_snps_susie_nodup$pip,
                       cut(methyl_snps_susie_nodup$min_dist_to_tss,bins),
					   function (x) sum(x,na.rm = TRUE))
counts_fsusie <- tapply(methyl_snps_fsusie_nodup$pip,
                        cut(methyl_snps_fsusie_nodup$min_dist_to_tss,bins),
						function (x) sum(x,na.rm = TRUE))
```

Now we can plot the result:

```{r plot-weighted-distance-to-tss-methyl, fig.height=2.5, fig.width=3.5}
n <- length(bins)
i <- seq(2,n-2)
bin_centers   <- bins[i] + bin_size/2
counts_assoc  <- counts_assoc[i]
counts_susie  <- counts_susie[i]
counts_fsusie <- counts_fsusie[i]
counts_assoc  <- counts_assoc/sum(counts_assoc)
counts_susie  <- counts_susie/sum(counts_susie)
counts_fsusie <- counts_fsusie/sum(counts_fsusie)
pdat <- data.frame(method = rep(c("assoc","susie","fsusie"),
                                each = length(bin_centers)),
                   dist   = rep(bin_centers/1000,times = 3),
                   freq   = c(counts_assoc,counts_susie,counts_fsusie),
				   stringsAsFactors = TRUE)
p <- ggplot(pdat,aes(x = dist,y = freq,color = method)) +
  geom_line(linewidth = 0.5) +
  geom_point(size = 1) +
  scale_x_continuous(breaks = seq(-200,200,50)) +
  scale_y_continuous(breaks = seq(0,1,0.05)) +
  scale_color_manual(values = c("darkblue","darkorange","dodgerblue")) +
  labs(x = "distance to TSS (kb)",y = "proportion of SNPs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r ggsave-weighted-distance-to-tss-methyl, echo=FALSE, warning=FALSE}
ggsave("dist_to_tss_methyl.pdf",p,height = 2.5,width = 3.5)
```

### Affected CpGs

Now let's turn to the recovery of affected CpGs.

Counting the number of CpGs per TAD is quite simple from the way the
fSuSiE results were compiled. It involves counting the number of unique
CpGs in each TAD:

```{r get-cpgs-per-tad-fsusie}
cpgs_per_tad_fsusie <-
  with(methyl_cpg_fsusie,tapply(ID,region,function (x) length(unique(x))))
```

Counting the number of CpGs per TAD from the SNP-CpG association tests is a
little more complicated because the CpGs were not assigned to TADs in the
results.

```{r get-cpgs-per-tad-assoc, cache=TRUE}
# Extract the information about the TADs from the TAD labels.
get_tad_info <- function (tads) {
  res <- strsplit(tads,"_")
  return(data.frame(tad   = tads,
                    chr   = factor(sapply(res,"[[",1)),
                    start = as.numeric(sapply(res,"[[",2)),
  			        end   = as.numeric(sapply(res,"[[",3))))
}

# Count the number of CpGs in each TAD.
count_features_per_tad <- function (features, tads) {
  tads$chr <- as.character(tads$chr)
  features$chr <- as.character(features$chr)
  n <- nrow(tads)
  out <- rep(0,n)
  names(out) <- tads$tad
  for (i in 1:n) {
    rows <- which(features$chr == tads[i,"chr"] &
                  features$pos >= tads[i,"start"] &
	  		  	  features$pos <= tads[i,"end"])
    out[i] <- length(unique(features[rows,"molecular_trait_id"]))

  }
  return(out)
}

tads <- get_tad_info(levels(methyl_cpg_fsusie$region))
cpgs_per_tad_assoc <- count_features_per_tad(methyl_cpg_assoc,tads)
```

These plots summarize the number of affected CpGs per TAD:

```{r plot-cpgs-per-tad, fig.height=2, fig.width=5, warning=FALSE}
cpgs_per_tad_assoc[cpgs_per_tad_assoc == 0] <- NA
cpgs_per_tad_fsusie[cpgs_per_tad_fsusie == 0] <- NA
pdat1 <- data.frame(x = cpgs_per_tad_assoc)
pdat2 <- data.frame(x = cpgs_per_tad_fsusie)
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 64) +
  xlim(0,2000) +
  labs(x = "number of CpGs",y = "number of TADs",
       title = "SNP-CpG association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 64) +
  xlim(0,2000) +
  labs(x = "number of CpGs",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

A small number of TADs have more than 2,000 affected CpGs:

```{r cpgs-per-tad-more}
sum(cpgs_per_tad_assoc > 2000,na.rm = TRUE)
sum(cpgs_per_tad_fsusie > 2000,na.rm = TRUE)
```

This plot compares the number of affected CpGs identified by fSuSiE
and the SNP-CpG association tests:

```{r compare-cpgs-per-tad, fig.height=2.5, fig.width=2.5, warning=FALSE}
pdat <- data.frame(assoc  = cpgs_per_tad_assoc,
                   fsusie = cpgs_per_tad_fsusie)
p <- ggplot(pdat,aes(x = assoc,y = fsusie)) +
  geom_point(color = "darkblue") +
  geom_abline(intercept = 0,slope = 1,color = "magenta",linetype = "dotted") +
  labs(x = "SNP-CpG association tests",y = "fSusiE") +
  xlim(0,4100) + 
  ylim(0,4100) + 
  theme_cowplot(font_size = 10)
print(p)
```

```{r compare-cpgs-per-tad-ggsave, echo=FALSE, warning=FALSE}
ggsave("affected_cpgs_per_tad.pdf",p,height = 2.5,width = 2.5)
```

One advantage of fSuSiE is that it is able to tell us which molecular
features are affected by which SNPs, and therefore it provides a more
coherent summary of how the SNPs affect methylation levels at a locus.
To examine this quantitatively, we compare the number of affected CpGs
per CS for fSuSiE to the number of affected CpGs per SNP from the
association tests. The result is much fewer CSs and many more
affected CpGs per CS:

```{r count-cpgs-per-snp, fig.height=2, fig.width=4.5}
x <- factor(methyl_cpg_assoc$variant_id)
cpgs_per_snp_assoc <- tapply(methyl_cpg_assoc$molecular_trait_id,x,
                             function (x) length(unique(x)))
rm(x)
nodup_cs <- levels(methyl_snps_fsusie_nodup$cs)
methyl_cpg_fsusie_nodup <- subset(methyl_cpg_fsusie,is.element(cs,nodup_cs))
methyl_cpg_fsusie_nodup <- transform(methyl_cpg_fsusie_nodup,cs = factor(cs))
cpgs_per_snp_fsusie <-
  with(methyl_cpg_fsusie_nodup,
       tapply(ID,cs,function (x) length(unique(x))))
pdat1 <- data.frame(x = cpgs_per_snp_assoc)
pdat2 <- data.frame(x = cpgs_per_snp_fsusie)
pdat1 <- subset(pdat1,x <= 25)
pdat2 <- subset(pdat2,x <= 150)
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 25) +
  labs(x = "number of CpGs",y = "number of SNPs",
       title = "SNP-CpG association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "darkblue",bins = 25) +
  labs(x = "number of CpGs",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

```{r, count-cpgs-per-snp-ggave, echo=FALSE, warning=FALSE}
ggsave("affected_cpgs_per_snp.pdf",
       plot_grid(p1,p2,nrow = 1,ncol = 2),
       height = 2,width = 4.5)
```

A small proportion of the SNPs and CSs are not plotted because they
have an unusually large number of CpGs:

```{r count-cpgs-per-snp-more}
mean(cpgs_per_snp_assoc > 20)
mean(cpgs_per_snp_fsusie > 150)
```

### Properties of mSNPs in 1-SNP CSs

Now let's look closely at the 1-SNP CSs for methylation.

```{r get-1-snp-cs}
methyl_cs1snp_susie  <- table(methyl_snps_susie_nodup$cs)
methyl_cs1snp_susie  <- names(methyl_cs1snp_susie)[methyl_cs1snp_susie == 1]
methyl_cs1snp_susie  <- subset(methyl_snps_susie,
                               is.element(cs,methyl_cs1snp_susie))
methyl_cs1snp_fsusie <- table(methyl_snps_fsusie_nodup$cs)
methyl_cs1snp_fsusie <- names(methyl_cs1snp_fsusie)[methyl_cs1snp_fsusie == 1]
methyl_cs1snp_fsusie <- subset(methyl_snps_fsusie,
                              is.element(cs,methyl_cs1snp_fsusie))
```

Compare the MAFs of the 1-SNP CSs to the MAFs across SNPs in CSs:

```{r one-snp-cs-mafs, fig.height=5, fig.width=5}
p1 <- ggplot(methyl_cs1snp_susie,aes(x = log10(maf))) +
  geom_histogram(color = "white",fill = "darkblue",bins = 36) +
  labs(x = "log10(MAF)",y = "number of SNPs",
       title = "SuSiE-topPC, 1-SNP CSs") +
  theme_cowplot(font_size = 10)
p2 <- ggplot(methyl_cs1snp_fsusie,aes(x = log10(maf))) +
  geom_histogram(color = "white",fill = "darkblue",bins = 36) +
  labs(x = "log10(MAF)",y = "number of SNPs",
       title = "fSuSiE, 1-SNP CSs") +
  theme_cowplot(font_size = 10)
p3 <- ggplot(methyl_snps_susie_nodup,aes(x = log10(maf))) +
  geom_histogram(color = "white",fill = "darkblue",bins = 36) +
  labs(x = "log10(MAF)",y = "number of SNPs",
       title = "SuSiE-topPC, all CSs") +
  theme_cowplot(font_size = 10)
p4 <- ggplot(methyl_snps_fsusie_nodup,aes(x = log10(maf))) +
  geom_histogram(color = "white",fill = "darkblue",bins = 36) +
  labs(x = "log10(MAF)",y = "number of SNPs",
       title = "fSuSiE, all CSs") +
  theme_cowplot(font_size = 10)
plot_grid(p1,p2,p3,p4,nrow = 2,ncol = 2)
```

What is most striking is the excess of rarer SNPs in the 1-SNP CSs,
particularly in the 1-SNP CSs identified by SuSiE-topPC.

TO DO: Add plots showing distribution of PIPs for SNPs in 1-SNP CSs.

Finally, this plot shows the distribution of the distances between the
mSNP and the nearest CpG affected by that SNP:

```{r dist-msnp-cpg-cs1snp, fig.height=2.5, fig.width=2.5, warning=FALSE}
methyl_cpg_fsusie_cs1snp <- subset(methyl_cpg_fsusie,
                                   is.element(cs,methyl_cs1snp_fsusie$cs))
rows <- match(methyl_cpg_fsusie_cs1snp$cs,methyl_cs1snp_fsusie$cs)
methyl_cpg_fsusie_cs1snp$variant_pos <- methyl_cs1snp_fsusie[rows,"pos"]
methyl_cpg_fsusie_cs1snp <- transform(methyl_cpg_fsusie_cs1snp,
                                      cs = factor(cs),
                                      dist = peak_start - variant_pos)
pdat <- tapply(methyl_cpg_fsusie_cs1snp$dist,methyl_cpg_fsusie_cs1snp$cs,
               function (x) {
			     i <- which.min(abs(x))
				 return(x[i])
    		   })
pdat <- data.frame(dist = pdat)
pdat <- transform(pdat,
  dist = cut(dist,c(-Inf,-1e6,-1e5,-1e4,-1e3,-100,
                    100,1e3,1e4,1e5,1e6,+Inf)))
p <- ggplot(pdat,aes(x = dist)) +
  geom_bar(color = "white",fill = "darkblue",bins = 64,width = 0.5) +
  labs(x = "CpG - mSNP distance (base-pairs)",
       y = "number of mSNPs") +
  theme_cowplot(font_size = 10) +
  theme(axis.text.x = element_text(angle = 45,hjust = 1)) 
print(p)
```

As we can see, the nearest affected CpG is very likely to be nearby,
although there are some putative long-range effects (which may or
may not be real).

## H3K27ac SNPs

Load the H3K27ac SNP results generated by SuSiE-topPC, fSuSiE and the
SNP-peak association testing:

```{r read-ha-snp-results}
ha_peak_assoc_file <- "../outputs/ROSMAP_haQTL_qtl_snp_qval0.05.tsv.gz"
ha_snps_susie_file <- "../outputs/ROSMAP_haQTL_cs_snp_toppc1_annotation.tsv.gz"
ha_snps_fsusie_file <- "../outputs/ROSMAP_haQTL_cs_snp_annotation.tsv.gz"
ha_peak_assoc  <- read_enrichment_results(ha_peak_assoc_file,n = 8)
ha_snps_susie  <- read_enrichment_results(ha_snps_susie_file,n = 6)
ha_snps_fsusie <- read_enrichment_results(ha_snps_fsusie_file,n = 7)
ha_snps_susie$region <-
  sapply(strsplit(ha_snps_susie$cs,":",fixed = TRUE),"[[",2)
ha_snps_susie  <- transform(ha_snps_susie,
                            region = factor(region),
                            cs     = factor(cs),
			  			    pc     = factor(pc))
ha_snps_fsusie <- transform(ha_snps_fsusie,
                            cs     = factor(cs),
						    region = factor(region),
						    study  = factor(study))
```

There is no need to look at the TAD sizes because the same TADs were
analyzed for both molecular traits, methylation and H3K27ac.

These histograms summarize the number of CSs per TAD:

```{r num-cs-per-tad-ha, fig.height=2, fig.width=5, warning=FALSE}
pdat1 <- get_cs_vs_tad_size(ha_snps_susie)
pdat2 <- get_cs_vs_tad_size(ha_snps_fsusie)
pdat1 <- transform(pdat1,num_cs = factor(num_cs,1:20))
pdat2 <- transform(pdat2,num_cs = factor(num_cs,1:20))
p1 <- ggplot(pdat1,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "tomato") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = num_cs)) +
  geom_histogram(stat = "count",color = "white",fill = "tomato") +
  scale_x_discrete(drop = FALSE) +
  labs(x = "number of CSs",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

Compare discovery of causal SNPs (number of CSs) in the SuSiE-topPC
and fSuSiE analyses:

```{r compare-num-cs-per-tad-ha, fig.height=2.75, fig.width=3.75, warning=FALSE}
dat1 <- get_cs_vs_tad_size(ha_snps_susie)
dat2 <- get_cs_vs_tad_size(ha_snps_fsusie)
dat1 <- dat1[c(1,3)]
dat2 <- dat2[c(1,3)]
names(dat1) <- c("tad","num_cs_susie")
names(dat2) <- c("tad","num_cs_fsusie")
dat <- merge(dat1,dat2,all = TRUE)
rows <- which(is.na(dat$num_cs_susie))
dat[rows,"num_cs_susie"] <- 0
pdat <- melt(with(dat,table(num_cs_susie,num_cs_fsusie)))
rows <- which(pdat$value == 0)
pdat[rows,"value"] <- NA
p <- ggplot(pdat,aes(x = num_cs_susie,y = num_cs_fsusie,size = value)) +
  geom_point(color = "white",fill = "tomato",shape = 21) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_x_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_y_continuous(breaks = seq(0,20,2),limits = c(0,20)) +
  scale_size(breaks = c(1,10,100)) +
  labs(x = "SuSiE-topPC",y = "fSuSiE",size = "number of TADs") +
  theme_cowplot(font_size = 10)
```

```{r compare-num-cs-per-tad-ha-ggsave, echo=FALSE, warning=FALSE}
ggsave("cs_per_tad_scatterplot_h3k27ac.pdf",p,height = 3,width = 4)
```

Flag and remove the "duplicate" CSs.

```{r remove-duplicated-cs-ha, message=FALSE, cache=TRUE}
root_cs_susie        <- create_cs_maps(ha_snps_susie)$root
root_cs_fsusie       <- create_cs_maps(ha_snps_fsusie)$root
ha_snps_susie_nodup  <- subset(ha_snps_susie,is.element(cs,root_cs_susie))
ha_snps_fsusie_nodup <- subset(ha_snps_fsusie,is.element(cs,root_cs_fsusie))
```

Compare the sizes of the CSs in the SuSiE-topPC and fSuSiE analyses:

```{r cs-sizes-ha, fig.height=2, fig.width=4, warning=FALSE}
bins <- c(0,1,2,5,10,20,Inf)
ha_snps_susie_nodup <- transform(ha_snps_susie_nodup,cs = factor(cs))
ha_snps_fsusie_nodup <- transform(ha_snps_fsusie_nodup,cs = factor(cs))
cs_size_susie  <- as.numeric(table(ha_snps_susie_nodup$cs))
cs_size_fsusie <- as.numeric(table(ha_snps_fsusie_nodup$cs))
cs_size_susie  <- cut(cs_size_susie,bins)
cs_size_fsusie <- cut(cs_size_fsusie,bins)
levels(cs_size_susie) <- bins[-1]
levels(cs_size_fsusie) <- bins[-1]
p1 <- ggplot(data.frame(cs_size = cs_size_susie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "tomato",
                 width = 0.65) +
  labs(x = "CS size",y = "number of CSs",title = "SuSiE-topPC") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(data.frame(cs_size = cs_size_fsusie),aes(x = cs_size)) +
  geom_histogram(stat = "count",color = "white",fill = "tomato",
                 width = 0.65) +
  labs(x = "CS size",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

We expect that the vast majority of the causal SNPs will be very close
to the TSS. Let's verify this empirically.

```{r cs-sizes-h3k27ac, echo=FALSE, warning=FALSE}
ggsave("cs_sizes_h3k27ac.pdf",
       plot_grid(p1,p2,nrow = 1,ncol = 2),
	   height = 2,width = 4)
```

First, add a "min_dist_to_TSS" column to each set of results:

```{r add-min-dist-to-tss-ha}
ha_snps_assoc  <- get_top_snp_per_location(ha_peak_assoc)
ha_snps_assoc  <- add_min_dist_to_tss(ha_snps_assoc,genes)
ha_snps_susie  <- add_min_dist_to_tss(ha_snps_susie,genes)
ha_snps_fsusie <- add_min_dist_to_tss(ha_snps_fsusie,genes)
```

This next code chunk computes the histogram for the TSS plot:

```{r compute-weighted-distance-to-tss-ha}
bin_size <- 5e4
bins <- c(-Inf,seq(-5e5,5.5e5,bin_size),Inf)
bins <- bins - bin_size/2
counts_assoc <- as.numeric(table(cut(ha_snps_assoc$min_dist_to_tss,bins)))
counts_susie <- tapply(ha_snps_susie$pip,
                       cut(ha_snps_susie$min_dist_to_tss,bins),
					   function (x) sum(x,na.rm = TRUE))
counts_fsusie <- tapply(ha_snps_fsusie$pip,
                        cut(ha_snps_fsusie$min_dist_to_tss,bins),
						function (x) sum(x,na.rm = TRUE))
```

And now we can plot the result:

```{r plot-weighted-distance-to-tss-ha, fig.height=2.5, fig.width=3.5}
n <- length(bins)
i <- seq(2,n-2)
bin_centers   <- bins[i] + bin_size/2
counts_assoc  <- counts_assoc[i]
counts_susie  <- counts_susie[i]
counts_fsusie <- counts_fsusie[i]
counts_assoc  <- counts_assoc/sum(counts_assoc)
counts_susie  <- counts_susie/sum(counts_susie)
counts_fsusie <- counts_fsusie/sum(counts_fsusie)
pdat <- data.frame(method = rep(c("assoc","susie","fsusie"),
                                each = length(bin_centers)),
                   dist   = rep(bin_centers/1000,times = 3),
                   freq   = c(counts_assoc,counts_susie,counts_fsusie),
				   stringsAsFactors = TRUE)
p <- ggplot(pdat,aes(x = dist,y = freq,color = method)) +
  geom_line(linewidth = 0.5) +
  geom_point(size = 1) +
  scale_x_continuous(breaks = seq(-500,500,100),limits = c(-500,500)) +
  scale_y_continuous(breaks = seq(0,1,0.1)) +
  scale_color_manual(values = c("darkblue","darkorange","dodgerblue")) +
  labs(x = "distance to TSS (kb)",y = "proportion of SNPs") +
  theme_cowplot(font_size = 10)
print(p)
```

```{r ggsave-weighted-distance-to-tss-ha, echo=FALSE, warning=FALSE}
ggsave("dist_to_tss_h3k27ac.pdf",p,height = 2.5,width = 3.5)
```

## Affected H3K27ac peaks

Now let's turn to the recovery of affected H3K27ac peaks. First load
the results generated by fSuSiE:

```{r read-ha-peak-results}
ha_peak_fsusie_file <-
  "../outputs/ROSMAP_haQTL_cs_effect_ha_peak_annotation.tsv.gz"
ha_peak_fsusie <- read_enrichment_results(ha_peak_fsusie_file,n = 9)
ha_peak_fsusie$region <-
  sapply(strsplit(ha_peak_fsusie$cs,":",fixed = TRUE),"[[",2)
ha_peak_fsusie <- transform(ha_peak_fsusie,
                            cs      = factor(cs), 
 						    region  = factor(region),
						    context = factor(context))
```

Count the number of affected peaks per TAD from the fSuSiE results:

```{r  get-ha-peaks-per-tad-fsusie}
ha_peaks_per_tad_fsusie <-
  with(ha_peak_fsusie,tapply(ID,region,function (x) length(unique(x))))
```

Next count the number of affected peaks per TAD from the SNP-peak
association tests, using the functions defined above:

```{r get-ha-peaks-per-tad-assoc}
tads <- get_tad_info(levels(ha_peak_fsusie$region))
ha_peaks_per_tad_assoc <- count_features_per_tad(ha_peak_assoc,tads)
```

These two plots summarize the number of affected peaks per TAD:

```{r plot-ha-peaks-per-tad, fig.height=2, fig.width=5, warning=FALSE}
ha_peaks_per_tad_assoc[ha_peaks_per_tad_assoc == 0] <- NA
ha_peaks_per_tad_fsusie[ha_peaks_per_tad_fsusie == 0] <- NA
pdat1 <- data.frame(x = ha_peaks_per_tad_assoc)
pdat2 <- data.frame(x = ha_peaks_per_tad_fsusie)
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(color = "white",fill = "tomato",bins = 64) +
  xlim(0,200) +
  labs(x = "number of H3K27ac peaks",y = "number of TADs",
       title = "SNP-peak association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "tomato",bins = 64) +
  xlim(0,200) +
  labs(x = "number of H3K27ac peaks",y = "number of TADs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

A very small number of TADs have more than 200 affected peaks:

```{r ha-peaks-per-tad-more}
mean(ha_peaks_per_tad_assoc > 200,na.rm = TRUE)
mean(ha_peaks_per_tad_fsusie > 200,na.rm = TRUE)
```

This plot compares the number of affected H3K27ac peaks identified by
fSuSiE and the SNP-peak association tests:

```{r compare-ha-peaks-per-tad, fig.height=2.5, fig.width=2.5, warning=FALSE}
pdat <- data.frame(assoc  = ha_peaks_per_tad_assoc,
                   fsusie = ha_peaks_per_tad_fsusie)
p <- ggplot(pdat,aes(x = assoc,y = fsusie)) +
  geom_point(color = "tomato") +
  geom_abline(intercept = 0,slope = 1,color = "magenta",linetype = "dotted") +
  labs(x = "SNP-peak association tests",y = "fSusiE") +
  xlim(0,200) + 
  ylim(0,200) + 
  theme_cowplot(font_size = 10)
print(p)
```

```{r compare-ha-peaks-per-tad-ggsave, echo=FALSE, warning=FALSE}
ggsave("affected_peaks_per_tad.pdf",p,height=2.5,width=2.5)
```

These next two plots compare the number of affected peaks per CS for
fSuSiE to the number of affected peaks per SNP from the association
tests:

```{r count-ha-peaks-per-snp, fig.height=2, fig.width=4.5}
x <- factor(ha_peak_assoc$variant_id)
ha_peaks_per_snp_assoc <- tapply(ha_peak_assoc$molecular_trait_id,x,
                                 function (x) length(unique(x)))
rm(x)
nodup_cs <- levels(ha_snps_fsusie_nodup$cs)
ha_peak_fsusie_nodup <- subset(ha_peak_fsusie,is.element(cs,nodup_cs))
ha_peak_fsusie_nodup <- transform(ha_peak_fsusie_nodup,cs = factor(cs))
ha_peaks_per_snp_fsusie <-
  with(ha_peak_fsusie_nodup,tapply(ID,cs,function (x) length(unique(x))))
pdat1 <- data.frame(x = ha_peaks_per_snp_assoc)
pdat2 <- data.frame(x = ha_peaks_per_snp_fsusie)
pdat1 <- subset(pdat1,x <= 10)
pdat2 <- subset(pdat2,x <= 40)
pdat1 <- transform(pdat1,x = factor(x))
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_bar(color = "white",fill = "tomato") +
  labs(x = "number of H3K27ac peaks",y = "number of SNPs",
       title = "SNP-peak association tests") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
p2 <- ggplot(pdat2,aes(x = x)) +
  geom_histogram(color = "white",fill = "tomato",bins = 25) +
  labs(x = "number of H3K27ac peaks",y = "number of CSs",title = "fSuSiE") +
  theme_cowplot(font_size = 10) +
  theme(plot.title = element_text(size = 10,face = "plain"))
plot_grid(p1,p2,nrow = 1,ncol = 2)
```

```{r count-ha-peaks-per-snp-ggsave, echo=FALSE, warning=FALSE}
ggsave("affected_peaks_per_snp.pdf",
       plot_grid(p1,p2,nrow = 1,ncol = 2),
	   height = 2,width = 4.5)
```

A small proportion of the SNPs and CSs were not plotted because they
had an unusually large number of affected peaks:

```{r count-ha-peaks-per-snp-more}
mean(ha_peaks_per_snp_assoc > 10)
mean(ha_peaks_per_snp_fsusie > 40)
```

[box-download-link]: https://uchicago.box.com/s/tt1vgg7vqayfthg0vsbl8gw0sdi0f1uo
