---
title: "Examine susie and fsusie results on the ROSMAP data"
author: William Denault, Hao Sun, Peter Carbonetto, Gao Wang
output: workflowr::wflow_html
---

Here we take a close look at a selection of the susie and fsusie
fine-mapping results for the ROSMAP data.

To build the workflowr page, I ran these commands on midway2:

```
sinteractive -c 4 --mem=24G --time=20:00:00
module load R/4.1.0-no-openblas
module load pandoc/3.0.1
R
> .libPaths()[1]
# [1] "/home/pcarbo/R_libs_4_10_no_openblas"
> getwd()
# [1] "../analysis"
> workflowr::wflow_build("rosmap.Rmd",local=TRUE,view=FALSE,verbose=TRUE)
```

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages as well as some additional custom functions used in
the analysis below.

```{r load-pkgs, message=FALSE}
library(data.table)
library(ggplot2)
library(cowplot)
source("../code/rosmap_functions.R")
setDTthreads(1)
```

Load the susie fine-mapping results on the DLPFC bulk RNA-seq data.

```{r load-susie-results}
datadir <- file.path("/project2/mstephens/fungen_xqtl/ftp_fgc_xqtl",
                     "analysis_result/finemapping_twas/prepared_results")
load(file.path(datadir,"susie_dlpfc_dejager_eQTL.RData"))					 
susie <- list(regions = regions,cs = cs,pips = pips)
susie$cs <- transform(susie$cs,pos = get_pos_from_id(id))
rm(regions,cs,pips)
```

Load the fsusie fine-mapping results on the DLPFC methylation data.

```{r load-fsusie-mqtl-results}
load(file.path(datadir,"fsusie_ROSMAP_DLPFC_mQTL.RData"))
fsusie_mqtl <- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)
```

Load the fsusie fine-mapping results on the DLPFC histone acetylation
(H3K9ac ChIP-seq) data.

```{r load-fsusie-haqtl-results}
load(file.path(datadir,"fsusie_ROSMAP_DLPFC_haQTL.RData"))
fsusie_haqtl <- list(regions = regions,cs = cs,pips = pips)
rm(regions,cs,pips)
```

Load the gene annotations. Specifically I extract here only the
annotated gene transcripts for protein-coding genes as defined in the
Ensembl/Havana database.

```{r load-gene-annotations}
gene_file <-
  file.path("../data/genome_annotations",
    "Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.gtf.gz")
genes <- get_gene_annotations(gene_file)
```

SuSiE fine-mapping of RNA-seq
-----------------------------

Region sizes in base-pairs and SNPs:

```{r susie-region-sizes, fig.height=3, fig.width=6.25}
p1 <- region_sizes_histogram(susie$regions,susie$pips,max_mb = 12)
p2 <- num_snps_histogram(susie$regions)
plot_grid(p1,p2)
```

Number of CSs per region:

```{r susie-num-cs}
table(CSs = susie$regions$num_cs)
```

CS sizes:

```{r susie-cs-sizes, fig.height=3, fig.width=3}
susie_cs_sizes <- get_cs_sizes_by_region(susie$cs)
cs_sizes_histogram(susie_cs_sizes,x_breaks = c(1:5,10,100,1000))
```

Plot showing distance between the susie SNP and the gene's TSS, in
which the distances are weighted by the PIPs (for SNPs in CSs only):

```{r dist-to-tss}
susie$regions <- add_tss_to_regions(susie$regions,genes)
bin_size <- 25000
bins   <- c(-Inf,seq(-1e6,1e6,bin_size),Inf)
counts <- compute_weighted_distance_to_tss(susie$regions,susie$cs,bins)
n      <- length(bins)
bins   <- bins[seq(2,n-2)]
counts <- counts[seq(2,n-2)]
pdat <- data.frame(pos = (bins + bin_size/2)/1e6,count = counts)
ggplot(pdat,aes(x = pos,y = count)) +
  geom_point(color = "darkblue") +
  geom_line(color = "darkblue") +
  labs(x = "distance to TSS (Mb)",
       y = "SNPs weighted by PIP") +
  theme_cowplot(font_size = 10)
```

fSuSiE fine-mapping of haQTLs
-----------------------------

Region sizes in base-pairs and SNPs:

```{r fsusie-haqtl-region-sizes, fig.height=3, fig.width=6.25}
p1 <- region_sizes_histogram(fsusie_haqtl$regions,fsusie_haqtl$pips,
                             max_mb = 10)
p2 <- num_snps_histogram(fsusie_haqtl$regions,max_snps = 50000)
plot_grid(p1,p2)
```

Number of CSs per region:

```{r fsusie-haqtl-num-cs}
table(CSs = fsusie_haqtl$regions$num_cs)
```

CS sizes:

```{r fsusie-haqtl-cs-sizes, fig.height=3, fig.width=3, warning=FALSE}
fsusie_haqtl_cs_sizes <- get_cs_sizes_by_region(fsusie_haqtl$cs)
cs_sizes_histogram(fsusie_haqtl_cs_sizes,x_breaks = c(1:5,10,100,1000)) +
  scale_y_continuous(trans = "log10")
```

fSuSiE fine-mapping of mQTLs
----------------------------

Region sizes in base-pairs and SNPs:

```{r fsusie-mqtl-region-sizes, fig.height=3, fig.width=6.25}
p1 <- region_sizes_histogram(fsusie_mqtl$regions,fsusie_mqtl$pips,
                             max_mb = 10)
p2 <- num_snps_histogram(fsusie_mqtl$regions,max_snps = 50000)
plot_grid(p1,p2)
```

Number of CSs per region:

```{r fsusie-mqtl-num-cs}
table(CSs = fsusie_mqtl$regions$num_cs)
```

CS sizes:

```{r fsusie-mqtl-cs-sizes, fig.height=3, fig.width=3, warning=FALSE}
fsusie_mqtl_cs_sizes <- get_cs_sizes_by_region(fsusie_mqtl$cs)
cs_sizes_histogram(fsusie_mqtl_cs_sizes,x_breaks = c(1:5,10,100,1000)) +
  scale_y_continuous(trans = "log10")
```

Examine overlap among expression, methylation and histone acetylation SNPs
--------------------------------------------------------------------------

TO DO: Determine overlap between expression, methylation and histone
acetylation SNPs simply by extracting the SNPs with PIP > 0.75 that
are also in CSs.

```{r high-conf-snps-overlap}
highconf_susie <- get_highconf_snps(susie$pips,level = 0.6)
highconf_mqtl  <- get_highconf_snps(fsusie_mqtl$pips,level = 0.6)
highconf_haqtl <- get_highconf_snps(fsusie_haqtl$pips,level = 0.6)
ids <- unique(c(highconf_susie$id,highconf_mqtl$id,highconf_haqtl$id))
dat <- data.frame(id = ids,
                  eqtl  = is.element(ids,highconf_susie$id),
				  mqtl  = is.element(ids,highconf_mqtl$id),
				  haqtl = is.element(ids,highconf_haqtl$id))
table(dat[c("eqtl","mqtl")])
table(dat[c("eqtl","haqtl")])
table(dat[c("mqtl","haqtl")])
```

TO DO NEXT: Put together a data frame containing more information
about these high-confidence, overlapping SNPs.
